
```{r, warning=F, message=F}
library(readxl)
library(DESeq2)
library(ggfortify)
library(ComplexHeatmap)
library(cluster)
library(data.table)
library(fgsea)
library(DT)
library(tibble)
library(dplyr)
library(tidyverse)
library(pathview)
library(clusterProfiler)
library(enrichplot)
library(ggplot2)
library(enrichR)
library(pheatmap)
library(gProfileR)
library(gprofiler2)
library(grid)
library(gridExtra)
library(ggpubr)
```

# 1) Metadata 

# 1.1) Metadata RNA ADRN vs. MES (General, Normoxia and Hypoxia)

* Used 5-hmC as template

```{r, warning=F, message=F}
"~/Desktop/Gepoliano/DE_RNA-Seq_Cells/DE_RNA-Seq_Cells_ALIGNMENT_HG38/" -> directory6
all_Files_RNA_Seq <- grep(".txt",list.files(directory6),value=TRUE)

cell_line_pheno_RNA_Seq <- read_excel(
  "~/Desktop/Gepoliano/DE_RNA-Seq_Cells/DE_RNA-Seq_Cells_ALIGNMENT_HG38/pheno_CellLine_RNASeq_Realignment.xlsx")

##_______________________________________________________##
# 1) ADRN and MES DFs: Not Subsetted (General)

combined_table_RNA_ADRN_vs_MES <- data.frame(
  sampleName = cell_line_pheno_RNA_Seq$FILE_NAME,
  fileName = cell_line_pheno_RNA_Seq$FILE_NAME,
  condition = cell_line_pheno_RNA_Seq)

##_______________________________________________________##
# 2) Convert Vector to Factor

combined_table_RNA_ADRN_vs_MES$condition.Identity <- as.factor(combined_table_RNA_ADRN_vs_MES$condition.Identity)

##_______________________________________________________##
# 3) Subset Normoxia and Hypoxia

combined_table_RNA_Normoxia <- combined_table_RNA_ADRN_vs_MES[ which(
    combined_table_RNA_ADRN_vs_MES$condition.Pheno == 'N'  ), ]

combined_table_RNA_Hypoxia <- combined_table_RNA_ADRN_vs_MES[ which(
    combined_table_RNA_ADRN_vs_MES$condition.Pheno == 'H'  ), ]

##_______________________________________________________##
# 4) Subset ADRN and MES

combined_table_RNA_ADRN_All <- combined_table_RNA_ADRN_vs_MES[ which(
     combined_table_RNA_ADRN_vs_MES$condition.Identity == 'ADRN' ), ]

combined_table_RNA_MES_All <- combined_table_RNA_ADRN_vs_MES[ which(
     combined_table_RNA_ADRN_vs_MES$condition.Identity == 'MES' ), ]

##_______________________________________________________##
# 5) Individual Isogenic Pairs Effect of Hypoxia

combined_table_RNA_WN <- combined_table_RNA_ADRN_vs_MES[ which(
     combined_table_RNA_ADRN_vs_MES$condition.Cell_line == 'WN' ), ]

combined_table_RNA_WS <- combined_table_RNA_ADRN_vs_MES[ which(
     combined_table_RNA_ADRN_vs_MES$condition.Cell_line == 'WS' ), ]

combined_table_RNA_SY5Y <- combined_table_RNA_ADRN_vs_MES[ which(
     combined_table_RNA_ADRN_vs_MES$condition.Cell_line == 'SY5Y' ), ]

combined_table_RNA_SHEP <- combined_table_RNA_ADRN_vs_MES[ which(
     combined_table_RNA_ADRN_vs_MES$condition.Cell_line == 'SHEP' ), ]

combined_table_RNA_55N <- combined_table_RNA_ADRN_vs_MES[ which(
     combined_table_RNA_ADRN_vs_MES$condition.Cell_line == '55N' ), ]

combined_table_RNA_5S <- combined_table_RNA_ADRN_vs_MES[ which(
     combined_table_RNA_ADRN_vs_MES$condition.Cell_line == '5S' ), ]

combined_table_RNA_BE2 <- combined_table_RNA_ADRN_vs_MES[ which(
     combined_table_RNA_ADRN_vs_MES$condition.Cell_line == 'BE2' ), ]

combined_table_RNA_NBLS <- combined_table_RNA_ADRN_vs_MES[ which(
     combined_table_RNA_ADRN_vs_MES$condition.Cell_line == 'NBLS' ), ]

```

# 1.2) RNA-Seq Prepare Metadata Isogenic Combined

```{r, message=F, warning=F}

"~/Desktop/Gepoliano/DE_RNA-Seq_Cells/DE_RNA-Seq_Cells_ALIGNMENT_HG38/" -> directory6
all_Files_RNA_Seq <- grep("FL_counts.txt",list.files(directory6),value=TRUE)


cell_line_pheno_RNA_Seq_Realignment <- read_excel(
  "~/Desktop/Gepoliano/DE_RNA-Seq_Cells/DE_RNA-Seq_Cells_ALIGNMENT_HG38/pheno_CellLine_RNASeq_Realignment.xlsx")


##_______________________________________________________##
# 1) Isogenic

cell_line_pheno_RNA_Isogenic <- cell_line_pheno_RNA_Seq_Realignment[ which(
  cell_line_pheno_RNA_Seq_Realignment$Isogenic_Type == 'Isogenic'), ]

##_______________________________________________________##
# 1.1) Isogenic DFs

combined_table_RNA_Isogenic <- data.frame(
  sampleName = cell_line_pheno_RNA_Isogenic$FILE_NAME,
  fileName = cell_line_pheno_RNA_Isogenic$FILE_NAME,
  condition = cell_line_pheno_RNA_Isogenic)

##_______________________________________________________##
# 1.2) Convert Vector to Factor

combined_table_RNA_Isogenic$condition.Pheno <- as.factor(combined_table_RNA_Isogenic$condition.Pheno)

##_______________________________________________________##
# 1.3) Subset ADRN and MES

cell_line_pheno_RNA_Isogenic_ADRN <- combined_table_RNA_Isogenic[ which(
    combined_table_RNA_Isogenic$condition.Identity == 'ADRN'  ), ]

cell_line_pheno_RNA_Isogenic_MES <- combined_table_RNA_Isogenic[ which(
    combined_table_RNA_Isogenic$condition.Identity == 'MES'  ), ]

##_______________________________________________________##
# 2) Subset ADRN and MES not Isogenic

##_______________________________________________________##
# 2.1) non-Isogenic DFs

combined_table_RNA_non_Isogenic <- data.frame(
  sampleName = cell_line_pheno_RNA_Seq_Realignment$FILE_NAME,
  fileName = cell_line_pheno_RNA_Seq_Realignment$FILE_NAME,
  condition = cell_line_pheno_RNA_Seq_Realignment)

##_______________________________________________________##
# 2.2) Convert Vector to Factor

combined_table_RNA_non_Isogenic$condition.Pheno <- as.factor(combined_table_RNA_non_Isogenic$condition.Pheno)

##_______________________________________________________##
# 2.3) Subset ADRN and MES non-Isogenic

cell_line_pheno_RNA_non_Isogenic_ADRN <- combined_table_RNA_non_Isogenic[ which(
    combined_table_RNA_non_Isogenic$condition.Identity == 'ADRN'  ), ]

cell_line_pheno_RNA_non_Isogenic_MES <- combined_table_RNA_non_Isogenic[ which(
    combined_table_RNA_non_Isogenic$condition.Identity == 'MES'  ), ]
```

# 1.3) RNA-Seq Normoxia and Hypoxia; ADRN MES combined_table_RNA DFs

```{r, warning=F, message=F}
combined_table_RNA_ADRN <- data.frame(
  sampleName = cell_line_pheno_RNA_ADRN$FILE_NAME,
  fileName = cell_line_pheno_RNA_ADRN$FILE_NAME,
  condition = cell_line_pheno_RNA_ADRN)

combined_table_RNA_MES <- data.frame(
  sampleName = cell_line_pheno_RNA_MES$FILE_NAME,
  fileName = cell_line_pheno_RNA_MES$FILE_NAME,
  condition = cell_line_pheno_RNA_MES)
```

# 1.4) Metadata 5-hmC ADRN vs. MES

# 1.4.1) ADRN vs. MES (General, Normoxia and Hypoxia)

```{r, warning=F, message=F}
"/Users/gepolianochaves/Desktop/Gepoliano/DE_Chicago_Cells/5HMC_counts" -> directory4
all_Files <- grep(".txt",list.files(directory4),value=TRUE)

cell_line_pheno <- read_excel(
  "/Users/gepolianochaves/Desktop/Gepoliano/DE_5HMC/cell_line_pheno.xlsx")

##_______________________________________________________##
# 1) ADRN and MES DFs: Not Subsetted (General)

combined_table_5hmC_ADRN_vs_MES <- data.frame(
  sampleName = cell_line_pheno$FILE_NAME,
  fileName = cell_line_pheno$FILE_NAME,
  condition = cell_line_pheno)

##_______________________________________________________##
# 2) Convert Vector to Factor

combined_table_5hmC_ADRN_vs_MES$condition.Identity <- as.factor(combined_table_5hmC_ADRN_vs_MES$condition.Identity)

##_______________________________________________________##
# 3) Subset Normoxia and Hypoxia

combined_table_5hmC_Normoxia <- combined_table_5hmC_ADRN_vs_MES[ which(
    combined_table_5hmC_ADRN_vs_MES$condition.Pheno == 'N'  ), ]

combined_table_5hmC_Hypoxia <- combined_table_5hmC_ADRN_vs_MES[ which(
    combined_table_5hmC_ADRN_vs_MES$condition.Pheno == 'H'  ), ]

##_______________________________________________________##
# 4) Subset ADRN

combined_table_5hmC_ADRN <- combined_table_5hmC_ADRN_vs_MES[ which(
    combined_table_5hmC_ADRN_vs_MES$condition.Identity == 'ADRN'  ), ]

##_______________________________________________________##
# 5) Subset ADRN 

# Isogenic
combined_table_5hmC_ADRN_Isogenic <- combined_table_5hmC_ADRN[ which(
    combined_table_5hmC_ADRN$condition.Cell_line == 'WN' |
      combined_table_5hmC_ADRN$condition.Cell_line == 'SY5Y' |
      combined_table_5hmC_ADRN$condition.Cell_line == '55N' 
      ), ]

combined_table_5hmC_ADRN_WN <- combined_table_5hmC_ADRN[ which(
    combined_table_5hmC_ADRN$condition.Cell_line == 'WN'
      ), ]

combined_table_5hmC_ADRN_SY5Y <- combined_table_5hmC_ADRN[ which(
    combined_table_5hmC_ADRN$condition.Cell_line == 'SY5Y'
      ), ]

combined_table_5hmC_ADRN_55N <- combined_table_5hmC_ADRN[ which(
    combined_table_5hmC_ADRN$condition.Cell_line == '55N'
      ), ]

# Non-Isogenic

combined_table_5hmC_ADRN_non_Isogenic <- combined_table_5hmC_ADRN[ which(
      combined_table_5hmC_ADRN$condition.Cell_line == 'BE2' |
      combined_table_5hmC_ADRN$condition.Cell_line == 'NBLS' 
      ), ]

combined_table_5hmC_ADRN_BE2 <- combined_table_5hmC_ADRN[ which(
      combined_table_5hmC_ADRN$condition.Cell_line == 'BE2'
      ), ]

combined_table_5hmC_ADRN_NBLS <- combined_table_5hmC_ADRN[ which(
      combined_table_5hmC_ADRN$condition.Cell_line == 'NBLS'
      ), ]
```

# 1.5) Metadata 5-hmC Load Sample Table 5-hmC (Isogenic Pairs)

```{r, warning=F, message=F}
"/Users/gepolianochaves/Desktop/Gepoliano/DE_Chicago_Cells/5HMC_counts" -> directory4
all_Files <- grep(".txt",list.files(directory4),value=TRUE)

# cell_line_pheno <- read_excel(
#   "/Users/gepolianochaves/Desktop/Gepoliano/DE_5HMC/cell_line_pheno.xlsx")

cell_line_pheno_5hmC <- read_excel(
  "/Users/gepolianochaves/Desktop/Gepoliano/DE_5HMC/cell_line_pheno.xlsx")

##_______________________________________________________##
# 1) Remove NBLS and BE2
cell_line_pheno_5hmC_Seq_Realignment <- cell_line_pheno[ which(
  cell_line_pheno$Cell_line != 'BE2' &
    cell_line_pheno$Cell_line != 'NBLS'  ), ]

cell_line_pheno_5hmC_ADRN <- cell_line_pheno_5hmC_Seq_Realignment[which(
  cell_line_pheno_5hmC_Seq_Realignment$Cell_line_type == "N" 
),]

cell_line_pheno_5hmC_MES <- cell_line_pheno_5hmC_Seq_Realignment[which(
  cell_line_pheno_5hmC_Seq_Realignment$Cell_line_type == "S" 
),]

##_______________________________________________________##
# 2) ADRN and MES DFs

combined_table_5hmC_ADRN <- data.frame(
  sampleName = cell_line_pheno_5hmC_ADRN$FILE_NAME,
  fileName = cell_line_pheno_5hmC_ADRN$FILE_NAME,
  condition = cell_line_pheno_5hmC_ADRN)

combined_table_5hmC_MES <- data.frame(
  sampleName = cell_line_pheno_5hmC_MES$FILE_NAME,
  fileName = cell_line_pheno_5hmC_MES$FILE_NAME,
  condition = cell_line_pheno_5hmC_MES)
```



# 2) PCA

# 2.1) Construct DDS RNA (General, Normoxia and Hypoxia ADRN vs. MES)

* Used 5-hmC as a template

```{r}
dds.RNA_General_ADRN_vs_MES <- DESeqDataSetFromHTSeqCount(
  sampleTable = combined_table_RNA_ADRN_vs_MES,
  directory = directory6,
  design = ~ condition.Identity
)

dds.RNA_Normoxia_ADRN_vs_MES <- DESeqDataSetFromHTSeqCount(
  sampleTable = combined_table_RNA_Normoxia,
  directory = directory6,
  design = ~ condition.Identity
)

dds.RNA_Hypoxia_ADRN_vs_MES <- DESeqDataSetFromHTSeqCount(
  sampleTable = combined_table_RNA_Hypoxia,
  directory = directory6,
  design = ~ condition.Identity
)


dds.RNA_General_ADRN_vs_MES <- DESeq(dds.RNA_General_ADRN_vs_MES)
dds.RNA_Normoxia_ADRN_vs_MES <- DESeq(dds.RNA_Normoxia_ADRN_vs_MES)
dds.RNA_Hypoxia_ADRN_vs_MES <- DESeq(dds.RNA_Hypoxia_ADRN_vs_MES)


## Lists the comparisons (coefficients)
resultsNames(
  dds.RNA_General_ADRN_vs_MES)
resultsNames(
  dds.RNA_Normoxia_ADRN_vs_MES)
resultsNames(
  dds.RNA_Hypoxia_ADRN_vs_MES)

dds.RNA_General_ADRN_vs_MES.res <- results(
  dds.RNA_General_ADRN_vs_MES,
  contrast = c("condition.Identity", "ADRN", "MES"))

dds.RNA_Normoxia_ADRN_vs_MES.res <- results(
  dds.RNA_Normoxia_ADRN_vs_MES,
  contrast = c("condition.Identity", "ADRN", "MES"))

dds.RNA_Hypoxia_ADRN_vs_MES.res <- results(
  dds.RNA_Hypoxia_ADRN_vs_MES,
  contrast = c("condition.Identity", "ADRN", "MES"))

```

# 2.2) Construct DDS RNA (Isogenic ADRN and MES Normoxia vs. Hypoxia)

# 2.2.1) Construct DDS

* Used 5-hmC as a template

```{r}
dds.RNA_Isogenic_ADRN_Norm_vs_Hypo <- DESeqDataSetFromHTSeqCount(
  sampleTable = cell_line_pheno_RNA_Isogenic_ADRN,
  directory = directory6,
  design = ~ condition.Pheno
)

dds.RNA_non_Isogenic_ADRN_Norm_vs_Hypo <- DESeqDataSetFromHTSeqCount(
  sampleTable = cell_line_pheno_RNA_non_Isogenic_ADRN,
  directory = directory6,
  design = ~ condition.Pheno
)

dds.RNA_Isogenic_MES_Norm_vs_Hypo <- DESeqDataSetFromHTSeqCount(
  sampleTable = cell_line_pheno_RNA_Isogenic_MES,
  directory = directory6,
  design = ~ condition.Pheno
)


dds.RNA_Isogenic_ADRN_Norm_vs_Hypo <- DESeq(dds.RNA_Isogenic_ADRN_Norm_vs_Hypo)
dds.RNA_non_Isogenic_ADRN_Norm_vs_Hypo <- DESeq(dds.RNA_non_Isogenic_ADRN_Norm_vs_Hypo)
dds.RNA_Isogenic_MES_Norm_vs_Hypo <- DESeq(dds.RNA_Isogenic_MES_Norm_vs_Hypo)


## Lists the comparisons (coefficients)
resultsNames(
  dds.RNA_Isogenic_ADRN_Norm_vs_Hypo)

resultsNames(
  dds.RNA_non_Isogenic_ADRN_Norm_vs_Hypo)

resultsNames(
  dds.RNA_Isogenic_MES_Norm_vs_Hypo)


dds.RNA_Isogenic_ADRN_Norm_vs_Hypo.res <- results(
  dds.RNA_Isogenic_ADRN_Norm_vs_Hypo,
  contrast = c("condition.Pheno", "N", "H"))

dds.RNA_non_Isogenic_ADRN_Norm_vs_Hypo.res <- results(
  dds.RNA_non_Isogenic_ADRN_Norm_vs_Hypo,
  contrast = c("condition.Pheno", "H", "N"))

dds.RNA_Isogenic_MES_Norm_vs_Hypo.res <- results(
  dds.RNA_Isogenic_MES_Norm_vs_Hypo,
  contrast = c("condition.Pheno", "H", "N"))

##_______________________________________________________##
# X) Individual Isogenic Pairs Effect of Hypoxia

# 1) WN

dds.RNA_Isogenic_WN_Norm_vs_Hypo <- DESeqDataSetFromHTSeqCount(
  sampleTable = combined_table_RNA_WN,
  directory = directory6,
  design = ~ condition.Pheno
)

dds.RNA_Isogenic_WN_Norm_vs_Hypo <- DESeq(dds.RNA_Isogenic_WN_Norm_vs_Hypo)

dds.RNA_Isogenic_WN_Norm_vs_Hypo.res <- results(
  dds.RNA_Isogenic_WN_Norm_vs_Hypo,
  contrast = c("condition.Pheno", "H", "N"))

# 2) WS

dds.RNA_Isogenic_WS_Norm_vs_Hypo <- DESeqDataSetFromHTSeqCount(
  sampleTable = combined_table_RNA_WS,
  directory = directory6,
  design = ~ condition.Pheno
)

dds.RNA_Isogenic_WS_Norm_vs_Hypo <- DESeq(dds.RNA_Isogenic_WS_Norm_vs_Hypo)

dds.RNA_Isogenic_WS_Norm_vs_Hypo.res <- results(
  dds.RNA_Isogenic_WS_Norm_vs_Hypo,
  contrast = c("condition.Pheno", "H", "N"))


# 3) SY5Y

dds.RNA_Isogenic_SY5Y_Norm_vs_Hypo <- DESeqDataSetFromHTSeqCount(
  sampleTable = combined_table_RNA_SY5Y,
  directory = directory6,
  design = ~ condition.Pheno
)

dds.RNA_Isogenic_SY5Y_Norm_vs_Hypo <- DESeq(dds.RNA_Isogenic_SY5Y_Norm_vs_Hypo)

dds.RNA_Isogenic_SY5Y_Norm_vs_Hypo.res <- results(
  dds.RNA_Isogenic_SY5Y_Norm_vs_Hypo,
  contrast = c("condition.Pheno", "H", "N"))

# 4) SHEP

dds.RNA_Isogenic_SHEP_Norm_vs_Hypo <- DESeqDataSetFromHTSeqCount(
  sampleTable = combined_table_RNA_SHEP,
  directory = directory6,
  design = ~ condition.Pheno
)

dds.RNA_Isogenic_SHEP_Norm_vs_Hypo <- DESeq(dds.RNA_Isogenic_SHEP_Norm_vs_Hypo)

dds.RNA_Isogenic_SHEP_Norm_vs_Hypo.res <- results(
  dds.RNA_Isogenic_SHEP_Norm_vs_Hypo,
  contrast = c("condition.Pheno", "H", "N"))

# 5) 55N

dds.RNA_Isogenic_55N_Norm_vs_Hypo <- DESeqDataSetFromHTSeqCount(
  sampleTable = combined_table_RNA_55N,
  directory = directory6,
  design = ~ condition.Pheno
)

dds.RNA_Isogenic_55N_Norm_vs_Hypo <- DESeq(dds.RNA_Isogenic_55N_Norm_vs_Hypo)

dds.RNA_Isogenic_55N_Norm_vs_Hypo.res <- results(
  dds.RNA_Isogenic_55N_Norm_vs_Hypo,
  contrast = c("condition.Pheno", "H", "N"))

# 6) 5S

dds.RNA_Isogenic_5S_Norm_vs_Hypo <- DESeqDataSetFromHTSeqCount(
  sampleTable = combined_table_RNA_5S,
  directory = directory6,
  design = ~ condition.Pheno
)

dds.RNA_Isogenic_5S_Norm_vs_Hypo <- DESeq(dds.RNA_Isogenic_5S_Norm_vs_Hypo)

dds.RNA_Isogenic_5S_Norm_vs_Hypo.res <- results(
  dds.RNA_Isogenic_5S_Norm_vs_Hypo,
  contrast = c("condition.Pheno", "H", "N"))

# 7) BE2

dds.RNA_Isogenic_BE2_Norm_vs_Hypo <- DESeqDataSetFromHTSeqCount(
  sampleTable = combined_table_RNA_BE2,
  directory = directory6,
  design = ~ condition.Pheno
)

dds.RNA_Isogenic_BE2_Norm_vs_Hypo <- DESeq(dds.RNA_Isogenic_BE2_Norm_vs_Hypo)

dds.RNA_Isogenic_BE2_Norm_vs_Hypo.res <- results(
  dds.RNA_Isogenic_BE2_Norm_vs_Hypo,
  contrast = c("condition.Pheno", "H", "N"))

# 8) NBLS

dds.RNA_Isogenic_NBLS_Norm_vs_Hypo <- DESeqDataSetFromHTSeqCount(
  sampleTable = combined_table_RNA_NBLS,
  directory = directory6,
  design = ~ condition.Pheno
)

dds.RNA_Isogenic_NBLS_Norm_vs_Hypo <- DESeq(dds.RNA_Isogenic_NBLS_Norm_vs_Hypo)

dds.RNA_Isogenic_NBLS_Norm_vs_Hypo.res <- results(
  dds.RNA_Isogenic_NBLS_Norm_vs_Hypo,
  contrast = c("condition.Pheno", "H", "N"))

```

# 2.3) Construct DDS RNA (Normoxia and Hypoxia ADRN vs. MES)

```{r}
# Did not find combined_table_RNA_ADRN
# dds.RNA_ADRN <- DESeqDataSetFromHTSeqCount(
#   sampleTable = combined_table_RNA_ADRN,
#   directory = directory6,
#   design = ~ condition.Pheno
# )

# Non-Isogenic BE2
dds.RNA_ADRN <- DESeqDataSetFromHTSeqCount(
  sampleTable = cell_line_pheno_RNA_non_Isogenic_ADRN,
  directory = directory6,
  design = ~ condition.Pheno
)

# Non-Isogenic BE2
dds.RNA_ADRN <- DESeqDataSetFromHTSeqCount(
  sampleTable = combined_table_RNA_ADRN_All,
  directory = directory6,
  design = ~ condition.Pheno
)

# Similarly to the first command, did not find combined_table_RNA_ADRN
# So I had to prepare the combined_table_RNA_MES_All object
dds.RNA_MES <- DESeqDataSetFromHTSeqCount(
  sampleTable = combined_table_RNA_MES_All,
  directory = directory6,
  design = ~ condition.Pheno
)

dds.RNA_ADRN <- DESeq(dds.RNA_ADRN)
dds.RNA_MES <- DESeq(dds.RNA_MES)

## Lists the comparisons (coefficients)
resultsNames(
  dds.RNA_ADRN)
resultsNames(
  dds.RNA_MES)

# This definition has the final impact on the standpoint of the Hypoxia definition.
# Hypoxia has to be defined first, not Normoxia

dds.RNA_ADRN.res <- results(
  dds.RNA_ADRN,
  contrast = c("condition.Pheno", "H", "N"))

dds.RNA_MES.res <- results(
  dds.RNA_MES,
  contrast = c("condition.Pheno", "H", "N"))
```

# 2.4) Construct DDS 5hmC (ADRN vs. MES)

```{r}
dds.5hmC_General_ADRN_vs_MES <- DESeqDataSetFromHTSeqCount(
  sampleTable = combined_table_5hmC_ADRN_vs_MES,
  directory = directory4,
  design = ~ condition.Identity
)

dds.5hmC_Normoxia_ADRN_vs_MES <- DESeqDataSetFromHTSeqCount(
  sampleTable = combined_table_5hmC_Normoxia,
  directory = directory4,
  design = ~ condition.Identity
)

dds.5hmC_Hypoxia_ADRN_vs_MES <- DESeqDataSetFromHTSeqCount(
  sampleTable = combined_table_5hmC_Hypoxia,
  directory = directory4,
  design = ~ condition.Identity
)


dds.5hmC_General_ADRN_vs_MES <- DESeq(dds.5hmC_General_ADRN_vs_MES)
dds.5hmC_Normoxia_ADRN_vs_MES <- DESeq(dds.5hmC_Normoxia_ADRN_vs_MES)
dds.5hmC_Hypoxia_ADRN_vs_MES <- DESeq(dds.5hmC_Hypoxia_ADRN_vs_MES)


## Lists the comparisons (coefficients)
resultsNames(
  dds.5hmC_General_ADRN_vs_MES)
resultsNames(
  dds.5hmC_Normoxia_ADRN_vs_MES)
resultsNames(
  dds.5hmC_Hypoxia_ADRN_vs_MES)

dds.5hmC_General_ADRN_vs_MES.res <- results(
  dds.5hmC_General_ADRN_vs_MES,
  contrast = c("condition.Identity", "ADRN", "MES"))

dds.5hmC_Normoxia_ADRN_vs_MES.res <- results(
  dds.5hmC_Normoxia_ADRN_vs_MES,
  contrast = c("condition.Identity", "ADRN", "MES"))

dds.5hmC_Hypoxia_ADRN_vs_MES.res <- results(
  dds.5hmC_Hypoxia_ADRN_vs_MES,
  contrast = c("condition.Identity", "ADRN", "MES"))
```

# 2.5) Construct DDS 5hmC

```{r}
dds.5hmC_ADRN <- DESeqDataSetFromHTSeqCount(
  sampleTable = combined_table_5hmC_ADRN,
  directory = directory4,
  design = ~ condition.Pheno
)

dds.5hmC_ADRN_Isogenic <- DESeqDataSetFromHTSeqCount(
  sampleTable = combined_table_5hmC_ADRN_Isogenic,
  directory = directory4,
  design = ~ condition.Pheno
)

dds.5hmC_ADRN_non_Isogenic <- DESeqDataSetFromHTSeqCount(
  sampleTable = combined_table_5hmC_ADRN_non_Isogenic,
  directory = directory4,
  design = ~ condition.Pheno
)

dds.5hmC_ADRN_WN <- DESeqDataSetFromHTSeqCount(
  sampleTable = combined_table_5hmC_ADRN_WN,
  directory = directory4,
  design = ~ condition.Pheno
)

dds.5hmC_ADRN_SY5Y <- DESeqDataSetFromHTSeqCount(
  sampleTable = combined_table_5hmC_ADRN_SY5Y,
  directory = directory4,
  design = ~ condition.Pheno
)

dds.5hmC_ADRN_55N <- DESeqDataSetFromHTSeqCount(
  sampleTable = combined_table_5hmC_ADRN_55N,
  directory = directory4,
  design = ~ condition.Pheno
)

dds.5hmC_ADRN_BE2 <- DESeqDataSetFromHTSeqCount(
  sampleTable = combined_table_5hmC_ADRN_BE2,
  directory = directory4,
  design = ~ condition.Pheno
)

dds.5hmC_ADRN_NBLS <- DESeqDataSetFromHTSeqCount(
  sampleTable = combined_table_5hmC_ADRN_NBLS,
  directory = directory4,
  design = ~ condition.Pheno
)

dds.5hmC_MES <- DESeqDataSetFromHTSeqCount(
  sampleTable = combined_table_5hmC_MES,
  directory = directory4,
  design = ~ condition.Pheno
)

dds.5hmC_ADRN <- DESeq(dds.5hmC_ADRN)
dds.5hmC_ADRN_Isogenic <- DESeq(dds.5hmC_ADRN_Isogenic)
dds.5hmC_ADRN_non_Isogenic <- DESeq(dds.5hmC_ADRN_non_Isogenic)

dds.5hmC_ADRN_WN <- DESeq(dds.5hmC_ADRN_WN)
dds.5hmC_ADRN_SY5Y <- DESeq(dds.5hmC_ADRN_SY5Y)
dds.5hmC_ADRN_55N <- DESeq(dds.5hmC_ADRN_55N)
dds.5hmC_ADRN_BE2 <- DESeq(dds.5hmC_ADRN_BE2)
dds.5hmC_ADRN_NBLS <- DESeq(dds.5hmC_ADRN_NBLS)
dds.5hmC_MES <- DESeq(dds.5hmC_MES)

## Lists the comparisons (coefficients)
resultsNames(
  dds.5hmC_ADRN)
resultsNames(
  dds.5hmC_MES)

# This definition has the final impact on the standpoint of the Hypoxia definition.
# Hypoxia has to be defined first, not Normoxia

dds.5hmC_ADRN.res <- results(
  dds.5hmC_ADRN,
  contrast = c("condition.Pheno", "H", "N"))

dds.5hmC_ADRN_Isogenic.res <- results(
  dds.5hmC_ADRN_Isogenic,
  contrast = c("condition.Pheno", "H", "N"))

dds.5hmC_ADRN_non_Isogenic.res <- results(
  dds.5hmC_ADRN_non_Isogenic,
  contrast = c("condition.Pheno", "H", "N"))

dds.5hmC_ADRN_WN.res <- results(
  dds.5hmC_ADRN_WN,
  contrast = c("condition.Pheno", "H", "N"))

dds.5hmC_ADRN_SY5Y.res <- results(
  dds.5hmC_ADRN_SY5Y,
  contrast = c("condition.Pheno", "H", "N"))

dds.5hmC_ADRN_55N.res <- results(
  dds.5hmC_ADRN_55N,
  contrast = c("condition.Pheno", "H", "N"))

dds.5hmC_ADRN_BE2.res <- results(
  dds.5hmC_ADRN_BE2,
  contrast = c("condition.Pheno", "H", "N"))

dds.5hmC_ADRN_NBLS.res <- results(
  dds.5hmC_ADRN_NBLS,
  contrast = c("condition.Pheno", "H", "N"))

dds.5hmC_MES.res <- results(
  dds.5hmC_MES,
  contrast = c("condition.Pheno", "H", "N"))
```

# 2.6) Plot PCA RNA (MES)

# 2.6.1) ADRN and MES

```{r, warning=F, message=F}
vst_RNA_General_ADRN_vs_MES <- vst(dds.RNA_General_ADRN_vs_MES, blind=FALSE)
## Plot PCA
plotPCA(vst_RNA_General_ADRN_vs_MES, intgroup = c("condition.Identity"))
plotPCA(vst_RNA_General_ADRN_vs_MES, intgroup = c("condition.Cell_line_type"))
plotPCA(vst_RNA_General_ADRN_vs_MES, intgroup = c("condition.Cell_line"))

# vst objects in this chunk should be the same as the object bellow.
# In order for plot the PCA bellow, need to find and run this object.
# plotPCA(vsd_RNA_condition.Cell_line_type, intgroup = c("condition.Cell_line_type"))
```

# 2.6.2) ADRN and MES

```{r, warning=F, message=F}
vsd_rna.condition.Cell_line_type_ADRN <- vst(dds.RNA_ADRN, blind=FALSE)
## Plot PCA
plotPCA(vsd_rna.condition.Cell_line_type_ADRN, intgroup = c("condition.Cell_line_type"))
plotPCA(vsd_rna.condition.Cell_line_type_ADRN, intgroup = c("condition.Cell_line"))


vsd_rna.condition.Cell_line_type_MES <- vst(dds.RNA_MES, blind=FALSE)
## Plot PCA
plotPCA(vsd_rna.condition.Cell_line_type_MES, intgroup = c("condition.Cell_line_type"))
plotPCA(vsd_rna.condition.Cell_line_type_MES, intgroup = c("condition.Cell_line"))
```

# 2.7) Plot PCA 5hmC

# 2.7.1) ADRN vs. MES

```{r, warning=F, message=F}
vst_5hmC_General_ADRN_vs_MES <- vst(dds.5hmC_General_ADRN_vs_MES, blind=FALSE)

## Plot PCA
plotPCA(vst_5hmC_General_ADRN_vs_MES, intgroup = c("condition.Identity"))
plotPCA(vst_5hmC_General_ADRN_vs_MES, intgroup = c("condition.Cell_line_type"))
plotPCA(vst_5hmC_General_ADRN_vs_MES, intgroup = c("condition.Cell_line"))

```

# 2.7.2) MES

```{r, warning=F, message=F}
vsd_5hmC.condition.Cell_line_type_ADRN <- vst(dds.5hmC_ADRN, blind=FALSE)
## Plot PCA
plotPCA(vsd_5hmC.condition.Cell_line_type_ADRN, intgroup = c("condition.Cell_line_type"))
plotPCA(vsd_5hmC.condition.Cell_line_type_ADRN, intgroup = c("condition.Cell_line"))


vsd_5hmC.condition.Cell_line_type_MES <- vst(dds.5hmC_MES, blind=FALSE)
## Plot PCA
plotPCA(vsd_5hmC.condition.Cell_line_type_MES, intgroup = c("condition.Cell_line_type"))
plotPCA(vsd_5hmC.condition.Cell_line_type_MES, intgroup = c("condition.Cell_line"))
```


# 3) ADRN vs. MES DEs in MA Plots

# 3.1) ADRN vs MES

```{r}
##____________________________________________________________________________##
# 1) Normoxia

DF_RNA_Normoxia_ADRN_vs_MES <- as.data.frame(dds.RNA_Normoxia_ADRN_vs_MES.res)

# Split LFC
DF_select_ID_LFC_Pos <- DF_RNA_Normoxia_ADRN_vs_MES[ which(
  DF_RNA_Normoxia_ADRN_vs_MES$log2FoldChange > 5 & DF_RNA_Normoxia_ADRN_vs_MES$padj < 0.05), ]
DF_ordered_LFC_Pos <- DF_select_ID_LFC_Pos[order(-DF_select_ID_LFC_Pos$log2FoldChange),] 
DF_ordered_LFC_Top_Pos <- DF_ordered_LFC_Pos[1:20,]
DF_ordered_LFC_Top_Pos <- rownames(DF_ordered_LFC_Top_Pos)

DF_select_ID_LFC_Neg <- DF_RNA_Normoxia_ADRN_vs_MES[ which(
  DF_RNA_Normoxia_ADRN_vs_MES$log2FoldChange < -5 & DF_RNA_Normoxia_ADRN_vs_MES$padj < 0.05), ]
DF_ordered_LFC_Neg <- DF_select_ID_LFC_Neg[order(DF_select_ID_LFC_Neg$log2FoldChange),]
DF_ordered_LFC_Top_Neg <- DF_ordered_LFC_Neg[1:20,]
DF_ordered_LFC_Top_Neg <- rownames(DF_ordered_LFC_Top_Neg)

DF <- DF_RNA_Normoxia_ADRN_vs_MES

first_list_interest <- c("PRRX1", "VIM", "SPARC", "TGFB1", "PHOX2B", "DACH1", "ASCL1",
                        "AHSA1", "ASRGL1", "ATP6V0E2", "CSE1L", "FKBP4", "IGSF3", "ISL1", "LSM4", "
                        NET1", "NRSN1", "RBP1", "TAGLN3", "TFAP2B", "UBE2T")

# Select specific genes to show
# set top = 0, then specify genes using label.select argument
ggmaplot(DF, main = expression("ADRN" %->% "MES (Normoxia)"),
         fdr = 0.05, fc = 10, size = 0.4,
         genenames = as.vector(DF$name),
         label.rectangle = TRUE,
         ggtheme = ggplot2::theme_minimal(),
         top = 0, label.select = c("PRRX1", "VIM", "SPARC", "TGFB1", "PHOX2B", "DACH1", "ASCL1",
                        #"AHSA1", 
                        #"ASRGL1", 
                        #"ATP6V0E2", 
                        #"CSE1L", 
                        #"FKBP4", 
                        #"IGSF3", 
                        #"ISL1", 
                        "LSM4",
                        #"NET1", 
                        #"NRSN1", 
                        #"RBP1", 
                        #"TAGLN3", 
                        #"TFAP2B", 
                        "UBE2T"
                        )
)

##____________________________________________________________________________##
# 2) Hypoxia

DF_RNA_Hypoxia_ADRN_vs_MES <- as.data.frame(dds.RNA_Hypoxia_ADRN_vs_MES.res)

DF <- DF_RNA_Hypoxia_ADRN_vs_MES

first_list_interest <- c("PRRX1", "VIM", "SPARC", "TGFB1", "PHOX2B", "DACH1", "ASCL1",
                        "AHSA1", "ASRGL1", "ATP6V0E2", "CSE1L", "FKBP4", "IGSF3", "ISL1", "LSM4",
                        "NET1", "NRSN1", "RBP1", "TAGLN3", "TFAP2B", "UBE2T")

# Select specific genes to show
# set top = 0, then specify genes using label.select argument
ggmaplot(DF, main = expression("ADRN" %->% "MES (Hypoxia)"),
         fdr = 0.05, fc = 10, size = 0.4,
         genenames = as.vector(DF$name),
         label.rectangle = TRUE,
         ggtheme = ggplot2::theme_minimal(),
         top = 0, label.select = c("PRRX1", "VIM", "SPARC", "TGFB1", "PHOX2B", "DACH1", "ASCL1",
                        #"AHSA1", 
                        #"ASRGL1", 
                        #"ATP6V0E2", 
                        #"CSE1L", 
                        #"FKBP4", 
                        #"IGSF3", 
                        #"ISL1", 
                        "LSM4",
                        #"NET1", 
                        #"NRSN1", 
                        #"RBP1", 
                        #"TAGLN3", 
                        #"TFAP2B", 
                        "UBE2T"
                        )
)

##____________________________________________________________________________##
# 3) Plot Genes by LFC and baseMean
DF <- subset(DF_RNA_Normoxia_ADRN_vs_MES, rownames(DF) %in% c("PRRX1", 
                                                           "VIM", 
                                                           "SPARC", 
                                                           "TGFB1", 
                                                           "PHOX2B", 
                                                           "DACH1", 
                                                           "ASCL1",
                                                           #"AHSA1",
                                                           #"ASRGL1",
                                                           #"ATP6V0E2",
                                                           #"CSE1L",
                                                           #"FKBP4",
                                                           #"IGSF3",
                                                           #"ISL1",
                                                           "LSM4",
                                                           #"NET1",
                                                           #"NRSN1",
                                                           #"RBP1",
                                                           #"TAGLN3",
                                                           #"TFAP2B",
                                                           "UBE2T"
                                                           ))

DF["Gene"] <- rownames(DF)

ggplot(DF, aes(x = reorder(Gene, log2FoldChange), y = log2FoldChange)) + geom_bar(stat = "identity")

ggplot(DF, aes(x = reorder(Gene, baseMean), y = baseMean)) + geom_bar(stat = "identity")

##____________________________________________________________________________##
# 4) Plot Normoxia
# Change order manually Pheno

df_adrn_normoxia <- assay(dds.RNA_Normoxia_ADRN_vs_MES)
df_adrn_normoxia <- as.data.frame(df_adrn_normoxia)
df_adrn_normoxia <- t(df_adrn_normoxia)
Sample_ID = rownames(df_adrn_normoxia)
df_adrn_normoxia <- cbind(Sample_ID, df_adrn_normoxia)

dt1 <- data.table(cell_line_pheno_RNA_Seq_Realignment, key = "FILE_NAME") 
dt2 <- data.table(df_adrn_normoxia, key = "Sample_ID")
joined.dt1.dt.2 <- dt1[dt2]
df_adrn_normoxia <- dt1[dt2]

df_adrn_normoxia$Pheno <- factor(df_adrn_normoxia$Identity,          
                  levels = c("ADRN",  
                             "MES"))
ggplot(df_adrn_normoxia, aes(fill=Identity, y=VIM, x=reorder(Cell_line, VIM))) + 
    geom_bar(position="dodge", stat="identity")+
  theme(axis.text.x = element_text(angle=45, hjust=1))+
  scale_fill_manual(values = c("#ff0c0c","#00008b"), labels = c("ADRN","MES"))

# Change order manually Pheno
df_adrn_modified <- df_adrn_normoxia
df_adrn_modified$Pheno <- factor(df_adrn_modified$Pheno,          
                  levels = c("N",  
                             "H"))
ggplot(df_adrn_modified, aes(fill=Pheno, y=VIM, x=reorder(Cell_line, VIM))) + 
    geom_bar(position="dodge", stat="identity")+
  theme(axis.text.x = element_text(angle=45, hjust=1))+
  scale_fill_manual(values = c("#ff0c0c","#00008b"), labels = c("N","H"))


# ADRN from previously
df_adrn_modified$Pheno <- factor(df_adrn_modified$Identity,          
                  levels = c("ADRN",  
                             "MES"))
ggplot(df_adrn_modified, aes(fill=Pheno, y=VIM, x=reorder(Cell_line, VIM))) + 
    geom_bar(position="dodge", stat="identity")+
  theme(axis.text.x = element_text(angle=45, hjust=1))+
  scale_fill_manual(values = c("#ff0c0c","#00008b"), labels = c("N","H"))
```

# 3.2) All ADRN and All MES Normoxia vs. Hypoxia

```{r}
##____________________________________________________________________________##
# 1) All ADRN
DF_RNA_non_Isogenic_ADRN_Norm_vs_Hypo <- as.data.frame(dds.RNA_non_Isogenic_ADRN_Norm_vs_Hypo.res)

# Split LFC
DF_ADRN_select_LFC_Pos <- DF_RNA_non_Isogenic_ADRN_Norm_vs_Hypo[ which(DF_RNA_non_Isogenic_ADRN_Norm_vs_Hypo$log2FoldChange > 5), ]
DF_ADRN_select_LFC_Neg <- DF_RNA_non_Isogenic_ADRN_Norm_vs_Hypo[ which(DF_RNA_non_Isogenic_ADRN_Norm_vs_Hypo$log2FoldChange < -2
                                                                       ), ]

first_list_interest <- c("PRRX1", "VIM", "SPARC", "TGFB1", "PHOX2B", "DACH1", "ASCL1",
                        "AHSA1", "ASRGL1", "ATP6V0E2", "CSE1L", "FKBP4", "IGSF3", "ISL1", "LSM4",
                        "NET1", "NRSN1", "RBP1", "TAGLN3", "TFAP2B", "UBE2T")

# Select specific genes to show
# set top = 0, then specify genes using label.select argument
ggmaplot(DF_RNA_non_Isogenic_ADRN_Norm_vs_Hypo, main = expression("Normoxia" %->% "Hypoxia (ADRN)"),
         fdr = 0.05, fc = 2, size = 0.4,
         genenames = as.vector(DF_RNA_non_Isogenic_ADRN_Norm_vs_Hypo$name),
         label.rectangle = TRUE,
         ggtheme = ggplot2::theme_minimal(),
         top = 0, label.select = c("PRRX1", "VIM", "SPARC", "TGFB1", "PHOX2B", "DACH1", "ASCL1"
                        #"AHSA1", 
                        #"ASRGL1", 
                        #"ATP6V0E2", 
                        #"CSE1L", 
                        #"FKBP4", 
                        #"IGSF3", 
                        #"ISL1", 
                        #"LSM4", 
                        #"NET1", 
                        #"NRSN1", 
                        #"RBP1", 
                        #"TAGLN3", 
                        #"TFAP2B", 
                        #"UBE2T"
                        )
)

test <- DF_RNA_non_Isogenic_ADRN_Norm_vs_Hypo
test_order_ascending <- test[order(test$baseMean),] 
test_order_ascending$index <- 1:nrow(test_order_ascending)

library("hutils")
quantile_object <- mutate_ntile(
  test_order_ascending,
  baseMean,
  3,
  weights = NULL,
  by = NULL,
  keyby = NULL,
  new.col = NULL,
  character.only = FALSE,
  overwrite = TRUE,
  check.na = FALSE)

rownames(quantile_object) <- rownames(test_order_ascending)

df_GeneIndexes_select <- quantile_object %>% dplyr::select("baseMean", "index", "baseMeanTerciles")

GeneIndexes <- subset(df_GeneIndexes_select, rownames(test_order_ascending) %in% c("PRRX1", "VIM", "SPARC","PHOX2B", "DACH1","ACSL1"))

##____________________________________________________________________________##
# 2) All MES

DF_RNA_Isogenic_MES_Norm_vs_Hypo <- as.data.frame(dds.RNA_Isogenic_MES_Norm_vs_Hypo.res)

first_list_interest <- c("PRRX1", "VIM", "SPARC", "TGFB1", "PHOX2B", "DACH1", "ASCL1",
                        "AHSA1", "ASRGL1", "ATP6V0E2", "CSE1L", "FKBP4", "IGSF3", "ISL1", "LSM4",
                        "NET1", "NRSN1", "RBP1", "TAGLN3", "TFAP2B", "UBE2T")

# Select specific genes to show
# set top = 0, then specify genes using label.select argument
ggmaplot(DF_RNA_Isogenic_MES_Norm_vs_Hypo, main = expression("Normoxia" %->% "Hypoxia (MES)"),
         fdr = 0.05, fc = 2, size = 0.4,
         genenames = as.vector(DF_RNA_Isogenic_MES_Norm_vs_Hypo$name),
         label.rectangle = TRUE,
         ggtheme = ggplot2::theme_minimal(),
         top = 0, label.select = c("PRRX1", "VIM", "SPARC", "TGFB1", "PHOX2B", "DACH1", "ASCL1"
                        #"AHSA1", 
                        #"ASRGL1", 
                        #"ATP6V0E2", 
                        #"CSE1L", 
                        #"FKBP4", 
                        #"IGSF3", 
                        #"ISL1", 
                        #"LSM4", 
                        #"NET1", 
                        #"NRSN1", 
                        #"RBP1", 
                        #"TAGLN3", 
                        #"TFAP2B", 
                        #"UBE2T"
                        )
)



test <- DF_RNA_Isogenic_MES_Norm_vs_Hypo
test_order_ascending <- test[order(test$baseMean),] 
test_order_ascending$index <- 1:nrow(test_order_ascending)

library("hutils")
quantile_object <- mutate_ntile(
  test_order_ascending,
  baseMean,
  3,
  weights = NULL,
  by = NULL,
  keyby = NULL,
  new.col = NULL,
  character.only = FALSE,
  overwrite = TRUE,
  check.na = FALSE)

rownames(quantile_object) <- rownames(test_order_ascending)

df_GeneIndexes_select <- quantile_object %>% dplyr::select("baseMean", "index", "baseMeanTerciles")

GeneIndexes <- subset(df_GeneIndexes_select, rownames(test_order_ascending) %in% c("PRRX1", "VIM", "SPARC","PHOX2B", "DACH1", "ASCL1"))

```

# 4.7) Optimal Gene Set for ADRN/MES Prediction

```{r}
# my_cell_line <- readline(prompt="Enter Cell Line for GSVA Boxplots: ")

##_______________________________________________________##
# 1) Gene Expression Matrix from VST

vsd_RNA_cells_DF <- as.data.frame(assay(vst_RNA_General_ADRN_vs_MES))

library(GSVA)
library(GSVAdata)
library(GSEABase)

vsd_RNA_cells_MTX <- as.matrix(vsd_RNA_cells_DF)

##_______________________________________________________##
# 2) Construct GSVA Matrix

# The txt is the appropriate file here. Be careful not to load the xlx file.

gene_sets_list <- getGmt("~/Desktop/Gepoliano/Hypoxia/Cytoskeleton/N_S_Lists_GC_transposed.txt")

GSVA_vsd_RNA_cells_MTX <- gsva(vsd_RNA_cells_MTX, 
                          gene_sets_list, 
                          min.sz=1, max.sz=Inf, 
                          verbose=TRUE)

##_______________________________________________________##
# 3) Bind Gene Expression Table and GSVA Matrix by columns

GSVA_GeneExpression_vsd_RNA_cells_DF <- rbind(GSVA_vsd_RNA_cells_MTX, vsd_RNA_cells_DF)

##_______________________________________________________##
# 4) Choose Gene Set Terms of Interest

GSVA_GeneExpression_vsd_RNA_cells_DF["GeneSetTermID"] <- rownames(GSVA_GeneExpression_vsd_RNA_cells_DF)

##_______________________________________________________##
# 5) Transpose DF with all genes

GSVA_GeneExpression_Not_Selected_Transposed <- t(GSVA_GeneExpression_vsd_RNA_cells_DF)

##_______________________________________________________##
# 6) Remove Extra Line 49 and Create Sample_ID column

library(dplyr)

# Can not be sliced if not DF

GSVA_GeneExpression_Not_Selected_Transposed <- as.data.frame(GSVA_GeneExpression_Not_Selected_Transposed)

GSVA_GeneExpression_Not_Selected_Transposed <- GSVA_GeneExpression_Not_Selected_Transposed %>% slice(-c(49))

GSVA_GeneExpression_Not_Selected_Transposed["Sample_ID"] <- rownames(GSVA_GeneExpression_Not_Selected_Transposed)

##_______________________________________________________##
# 7) Select Gene Terms of Interest from DF

my_gsva_term <- c("A1", "A2", "A3", "A4", "A5", "A6", "A7", "A8", "A9", "A10",
                  "A11", "A12", "A13", "A14", "A15", "A16", "A17", "A18",
                  "Down_5hmC", "VIM", "PRRX1", "SPARC", "TGFB1",
                  "WWTR1",
                  "ANXA1","COL1A1", "CHGA", "CPLX2", # MES and ADRN From talk to Alex
                  "ITGA8","NDRG1","NGP_Up","Kelly_Up",
                  "KIT","PROM1","ID2", "POU5F1", 
                  "TH", "DBH",#Bhaskara:POU5F1=OCT4 
                  #"CD273", 
                  "ALDH1A1", "SOX2","NANOG",# Stemness genes Chen 2020
                  "CD274", "CSF1", "HIF1A","EPAS1", "TET1", "MYC", # HIF1A is MES in normoxia, ADRN in Hypoxia
                  # MYC is MES in Amp; ADRN in MYCN non-Amp
                  "MYCN", 
                  "CA9", "HK2", "ENO1", "PGK1", "BNIP3",#  ADRN in non-MYCN Amp cells; MES in MYCN Amp
                  "DACH1", "ASCL1","CKB", "ACTL6B",
                  "B4GALNT1","PHOX2A", "PHOX2B", "GATA3", "HAND1", "HAND2", "FAM163A", "FAM163B",
                  "FOXM1", "Intermediate_filament", "Actin_Filament",
                  "Tubulin_Complex", "Mielin_Sheath", "Type_III_Intermediate_Filament", 
                  "TUBA1A", "TUBB2B",
                  "SNAP25", "SNAP23", 
                  "UBE2T", 
                  "STC1", "BHLHE40",# MES
                  "SUCNR1","CXCR4",
                  "TFAP2B", "ISL1",# ADRN
                  "GAPDH",
                  "P4HA1",# Alex's MES genes involved in collagen formation and metabolism (SPARC)
                  "ADAMTS2",# Alex's MES genes involved in collagen formation and metabolism (SPARC)
                  "LDHA",
                  "EZHIP" # Evaluate the involvement of EZHIP in ADRN and MES in NB
                  )

##_______________________________________________________##
# 8) Make terms numeric in for Loop

# Test class before
class(GSVA_GeneExpression_Not_Selected_Transposed$A1)
class(GSVA_GeneExpression_Not_Selected_Transposed$NGP_Up)
class(GSVA_GeneExpression_Not_Selected_Transposed$Type_III_Intermediate_Filament)

# Need to assign object here so it can be used in the next command
GSVA_GeneExpression_Selected_Transposed <- GSVA_GeneExpression_Not_Selected_Transposed

# A lot of time there's an error for case matching between name in excel and name in my_gsva_term (above)
for (gsva_gene_set in my_gsva_term){
GSVA_GeneExpression_Selected_Transposed[[gsva_gene_set]] <- as.numeric(GSVA_GeneExpression_Not_Selected_Transposed[[gsva_gene_set]])
}

# Test class after
class(GSVA_GeneExpression_Selected_Transposed$A18)
class(GSVA_GeneExpression_Selected_Transposed$FAM163A)
class(GSVA_GeneExpression_Selected_Transposed$FAM163B)
class(GSVA_GeneExpression_Selected_Transposed$NGP_Up)
class(GSVA_GeneExpression_Selected_Transposed$Type_III_Intermediate_Filament)
class(GSVA_GeneExpression_Selected_Transposed$EZHIP)

##_______________________________________________________##
# 9) Filter DF with Selected Terms Using dplyr

library(dplyr)
# For some reason, in this case, calling library(dplyr) is not helping.
# However, calling as below does the task.
# This selects terms of interest

GSVA_GeneExpression_selected_Terms <- GSVA_GeneExpression_Selected_Transposed %>%
  dplyr::select(my_gsva_term)

##_______________________________________________________##
# 10) Join DFs and ensure as.data.frame

GSVA_GeneExpression_selected_Terms["Sample_ID"] <- rownames(GSVA_GeneExpression_selected_Terms)

dt1 <- data.table(cell_line_pheno_RNA_Seq_Realignment, key = "FILE_NAME") 
dt2 <- data.table(GSVA_GeneExpression_selected_Terms, key = "Sample_ID")
joined.dt1.dt.2 <- dt1[dt2]
GSVA_GeneExpression_SampleIDs <- dt1[dt2]

##_______________________________________________________##
# 11) Plot GSVA Scores and Evaluate Effect of Hypoxia on MESenchymalness

# GSVA_GeneExpression_Cell_Line <- GSVA_GeneExpression_SampleIDs[ which(
#   GSVA_GeneExpression_SampleIDs$Cell_line==my_cell_line
#     ), ]

mean_ADRN_vec <- c()
mean_MES_vec <- c()
term_interest_vector <- c()
for (gsva_gene_set in my_gsva_term){

term_interest_vector <- c(term_interest_vector, gsva_gene_set)
  
# This line assigns the value
plot_gsva_gene_set <- print(ggboxplot(GSVA_GeneExpression_SampleIDs, x = "Identity", y = gsva_gene_set,
          color = "Cell_line", palette = "jco",
          add = "jitter") + stat_compare_means(method = "t.test"))

# This assigns the value to the object
name <- paste0("plot_gsva_gene_set_", gsva_gene_set)
assign(name, plot_gsva_gene_set)
print(name)

# Find out Max difference between ADRN and MES GSVA Means

ADRN_gsva_Scores_DF <- GSVA_GeneExpression_SampleIDs[ which(
  GSVA_GeneExpression_SampleIDs$Identity=="ADRN"
    ), ]
mean_ADRN <- mean(ADRN_gsva_Scores_DF[[gsva_gene_set]])
mean_ADRN_vec <- c(mean_ADRN_vec, mean_ADRN)

MES_gsva_Scores_DF <- GSVA_GeneExpression_SampleIDs[ which(
  GSVA_GeneExpression_SampleIDs$Identity=="MES"
    ), ]
mean_MES <- mean(MES_gsva_Scores_DF[[gsva_gene_set]])
mean_MES_vec <- c(mean_MES_vec, mean_MES)

}

##_______________________________________________________##
# 11) Construct DF Difference ADRN and MES scores

Max_Diff_DF <- cbind(
  term_interest_vector,
  cbind(mean_ADRN_vec, mean_MES_vec)
                     )

Max_Diff_DF <- as.data.frame(Max_Diff_DF)

Max_Diff_DF["Diff"] <- as.numeric(Max_Diff_DF$mean_ADRN_vec) - as.numeric(Max_Diff_DF$mean_MES_vec)

# Sort by Max ADRN/MES difference so this can be used to get gene sets in order
# This list/DF is to be used to access the order of ADRNergicness

Max_Diff_DF_sorted <- Max_Diff_DF[order(-Max_Diff_DF$Diff),]

##_______________________________________________________##
# 12) Construct plots By Maximum ADRN/MES Score Difference

# This prints the Genes and Gene sets in order of sorting

for (i in Max_Diff_DF_sorted$term_interest_vector){
  print(paste0("plot_gsva_gene_set_",i))
  # This line assigns the value
  plot_gsva_gene_set <- print(ggboxplot(GSVA_GeneExpression_SampleIDs, x = "Identity", y = gsva_gene_set,
          color = "Cell_line", palette = "jco",
          add = "jitter") + stat_compare_means(method = "t.test"))

  # This assigns the value to the object
  name <- paste0("plot_gsva_gene_set_", gsva_gene_set)
  assign(name, plot_gsva_gene_set)
  print(name)
}

##_______________________________________________________##
# 13) Plot in order found

pdf(file = paste0("~/Desktop/Gepoliano/Hypoxia/Cytoskeleton/Hypoxia_Cytoskeleton_Sets_Ranked_3.pdf"),   
    width = 20, 
    height = 18)
grid.arrange(plot_gsva_gene_set_FAM163B,
             plot_gsva_gene_set_PHOX2A, plot_gsva_gene_set_PHOX2B, 
             plot_gsva_gene_set_ACTL6B, plot_gsva_gene_set_DACH1, plot_gsva_gene_set_MYCN,
             plot_gsva_gene_set_CKB,plot_gsva_gene_set_A12, plot_gsva_gene_set_A13,
             plot_gsva_gene_set_A3, plot_gsva_gene_set_A4,plot_gsva_gene_set_A5,
             plot_gsva_gene_set_A14, plot_gsva_gene_set_A11, plot_gsva_gene_set_A1,
             plot_gsva_gene_set_A9, 
             #plot_gsva_gene_set_B4GALNT1,
             plot_gsva_gene_set_Down_5hmC,plot_gsva_gene_set_A10, 
             plot_gsva_gene_set_A2, plot_gsva_gene_set_A15, plot_gsva_gene_set_A18,
             # These terms do not need to be included in this trial:
             # plot_gsva_gene_set_Down_5hmC, 
             # plot_gsva_gene_set_MYCN, 
             plot_gsva_gene_set_A17, plot_gsva_gene_set_A8, plot_gsva_gene_set_A16,
             plot_gsva_gene_set_A7, plot_gsva_gene_set_A6, plot_gsva_gene_set_VIM,
             #plot_gsva_gene_set_PRRX1,
             plot_gsva_gene_set_SPARC,
             plot_gsva_gene_set_NGP_Up,
             plot_gsva_gene_set_Kelly_Up,
                         top=paste0("GSVA Scores in ", " my_cell_line Here "," for Different Gene Sets"),
             ncol = 6)
dev.off()

##_______________________________________________________##
# 14) Print sigle Figure

pdf(file = paste0("~/Desktop/Gepoliano/Hypoxia/Cytoskeleton/ADRN_MES_Candidates.pdf"),   
    width = 16, 
    height = 10)
grid.arrange(plot_gsva_gene_set_SPARC, plot_gsva_gene_set_VIM,
             plot_gsva_gene_set_ANXA1, plot_gsva_gene_set_COL1A1,
             plot_gsva_gene_set_Actin_Filament, 
             plot_gsva_gene_set_CHGA, plot_gsva_gene_set_CPLX2,
              plot_gsva_gene_set_PHOX2B, #plot_gsva_gene_set_DACH1,
             plot_gsva_gene_set_Intermediate_filament, 
             plot_gsva_gene_set_Tubulin_Complex, plot_gsva_gene_set_Mielin_Sheath,
             plot_gsva_gene_set_Type_III_Intermediate_Filament, 
                         top=paste0("GSVA Scores in ", " my_cell_line Here "," for Different Gene Sets"),
             ncol = 5)
dev.off()
```

# 4.8) ADRN vs. MES (in N and H)

# 4.8.1) All ADRN all MES cells

```{r}
# library
library(ggplot2)

#my_gene <- readline(prompt="Enter Gene of Interest: ")
# Gene hard name: B4GALNT1

# DF_Identity_Pheno_Col <- add_column(GSVA_GeneExpression_SampleIDs, ID_Pheno_Col = eff_vector, .after = "Identity")

DF_Identity_Pheno_Col <- GSVA_GeneExpression_SampleIDs

##_______________________________________________________##
# 2) Prepare Data for Statistics

library(ggpubr)
library(rstatix)

# 1) Data preparation

# Transform `dose` into factor variable
df <- DF_Identity_Pheno_Col

# Select pair of interest
# df <- df[ which(
#   df$Cell_line=="55N" |
#      df$Cell_line=="5S" ), ]

df$Identity <- as.factor(df$Identity)

# Add a random grouping variable
df$group <- factor(rep(c("grp1", "grp2"), 24))

##_______________________________________________________##
# 3) Facet wrap by one variable (Pheno)

##______________________________________##
# 3.1) Constructs Statistics Object

stat.test <- df %>%
  group_by(Pheno) %>%
  t_test(EZHIP ~ Identity) %>%
  adjust_pvalue(method = "bonferroni") %>%
  add_significance()
# View(stat.test) 

##______________________________________##
# 3.2) Construct Boxplot and Facet (by Pheno)

# Create a box plot
bxp <- ggboxplot(
  df, x = "Identity", y = "EZHIP", 
  color = "Cell_line",
  facet.by = "Pheno"
  )

##______________________________________##
# 3.3) Add p-values

# Make facet and add p-values
stat.test <- stat.test %>% add_xy_position(x = "Identity")
bxp + stat_pvalue_manual(stat.test)

##_______________________________________________________##
# 4) Facet wrap by one variable (Identity)

# 4.1) Facet wrap by one variable

stat.test <- df %>%
  group_by(Identity) %>%
  t_test(EZHIP ~ Pheno) %>%
  adjust_pvalue(method = "bonferroni") %>%
  add_significance()
# View(stat.test) 

# 3) Boxplot t-test

# Create a box plot
plot_object <- ggboxplot(
  df, x = "Pheno", y = "EZHIP",
  color = "Cell_line",
  facet.by = "Identity"
  )

# Make facet and add p-values
stat.test <- stat.test %>% add_xy_position(x = "Pheno")
plot_object + stat_pvalue_manual(y.position = 12, stat.test, bracket.nudge.y = c(.1, .2), hide.ns = FALSE,
    label = "{p.adj} ({p.adj.signif})") # Included space between p-value and asterisks
```

# 4.8.2) One Cell Pair at a Time

```{r}
# library
library(ggplot2)

#my_gene <- readline(prompt="Enter Gene of Interest: ")
# Gene hard name: B4GALNT1

# DF_Identity_Pheno_Col <- add_column(GSVA_GeneExpression_SampleIDs, ID_Pheno_Col = eff_vector, .after = "Identity")

DF_Identity_Pheno_Col <- GSVA_GeneExpression_SampleIDs

##_______________________________________________________##
# 2) Prepare Data for Statistics

library(ggpubr)
library(rstatix)

# 1) Data preparation

# Transform `dose` into factor variable
df <- DF_Identity_Pheno_Col

# Select pair of interest
df <- df[ which(
  df$Cell_line=="WN" |
     df$Cell_line=="WS" ), ]

df$Cell_line_type <- as.factor(df$Cell_line_type)

# Add a random grouping variable
df$group <- factor(rep(c("grp1", "grp2"), 6))

##_______________________________________________________##
# 3) Facet wrap by one variable (Pheno)

##______________________________________##
# 3.1) Constructs Statistics Object

stat.test <- df %>%
  group_by(Pheno) %>%
  t_test(TGFB1 ~ Cell_line_type) %>%
  adjust_pvalue(method = "bonferroni") %>%
  add_significance()
# View(stat.test) 

##______________________________________##
# 3.2) Construct Boxplot and Facet (by Pheno)

# Create a box plot
bxp <- ggboxplot(
  df, x = "Cell_line_type", y = "TGFB1", 
  color = "Cell_line",
  facet.by = "Pheno", scales = "free", add = "jitter"
  )

##______________________________________##
# 3.3) Add p-values

# Make facet and add p-values
stat.test <- stat.test %>% add_xy_position(x = "Cell_line_type")
bxp + stat_pvalue_manual(stat.test)

##_______________________________________________________##
# 4) Facet wrap by one variable (Identity)

# 4.1) Facet wrap by one variable

stat.test <- df %>%
  group_by(Cell_line_type) %>%
  t_test(TGFB1 ~ Pheno) %>%
  adjust_pvalue(method = "bonferroni") %>%
  add_significance()
# View(stat.test) 

# 3) Boxplot t-test

# Create a box plot
plot_object <- ggboxplot(
  df, x = "Pheno", y = "TGFB1",
  color = "Cell_line",
  facet.by = "Cell_line_type", 
  scales = "free", 
  add = "jitter"
  )

# Make facet and add p-values
stat.test <- stat.test %>% add_xy_position(x = "Pheno")
plot_object + stat_pvalue_manual(stat.test, bracket.nudge.y = c(.1, .2), hide.ns = FALSE,
    label = "{p.adj} ({p.adj.signif})") # Included space between p-value and asterisks
```

# 4.9) ADRN vs. MES (in N and H) Bar Graph

```{r}
# library
library(ggplot2)

# DF_Identity_Pheno_Col <- add_column(GSVA_GeneExpression_SampleIDs, ID_Pheno_Col = eff_vector, .after = "Identity")

DF_Identity_Pheno_Col <- GSVA_GeneExpression_SampleIDs

##_______________________________________________________##
# 2) Prepare Data for Statistics

library(ggpubr)
library(rstatix)

# 1) Data preparation

# Transform `dose` into factor variable
df <- DF_Identity_Pheno_Col

# Select pair of interest
# df <- df[ which(
#   df$Cell_line=="55N" |
#      df$Cell_line=="5S" ), ]

df$Identity <- as.factor(df$Identity)

# Add a random grouping variable
df$group <- factor(rep(c("grp1", "grp2"), 24))

##_______________________________________________________##
# 3) Facet wrap by one variable (Pheno)

##______________________________________##
# 3.1) Constructs Statistics Object

stat.test <- df %>%
  group_by(Pheno) %>%
  t_test(B4GALNT1 ~ Identity) %>%
  adjust_pvalue(method = "bonferroni") %>%
  add_significance()
# View(stat.test) 

##______________________________________##
# 3.2) Construct Boxplot and Facet (by Pheno)

# Create a box plot
plot_object <- ggbarplot(
  df, x = "Identity", y = "B4GALNT1", fill = "#00AFBB", add = "mean_sd",
  facet.by = "Pheno"
  )

##______________________________________##
# 3.3) Add p-values

# Make facet and add p-values
stat.test <- stat.test %>% add_xy_position(fun = "mean_sd", x = "Identity")
plot_object + stat_pvalue_manual(stat.test)

##_______________________________________________________##
# 4) Facet wrap by one variable (Identity)

# 4.1) Facet wrap by one variable

stat.test <- df %>%
  group_by(Identity) %>%
  t_test(B4GALNT1 ~ Pheno) %>%
  adjust_pvalue(method = "bonferroni") %>%
  add_significance()
# View(stat.test) 

# 3) Boxplot t-test

# Create a box plot
plot_object <- ggbarplot(
  df, x = "Pheno", y = "B4GALNT1", add = "mean_sd",
  color = "Cell_line",
  facet.by = "Identity"
  )

# Make facet and add p-values
stat.test <- stat.test %>% add_xy_position(fun = "mean_sd", x = "Pheno")
plot_object + stat_pvalue_manual(stat.test)
```

# 4.10) Compare MES w/ ADRN Hypoxia

# 4.10.1) Define List of (MES) Genes of Interest

```{r}
A5 <- scan(text="AKT3
ATP1B1
ATP6V1B2
BEX1
C14orf132
C1orf21
CADM1
CAMK4
CCND1
CD24
CDC42EP3
CHGA
CHN1
CKB
CNRIP1
CPLX2
CRIPT
CRMP1
DOK4
DPH2
DPYSL2
DPYSL5
ELAVL3
ELAVL4
ENO2
FAM107B
FAM163A
GATA2
GATA3
GDAP1
GINS2
GNAO1
GNAQ
GRK3
HAND2
IGFBP2
INA
KIF5C
KLF7
KLHL23
MACROH2A2
MAP1B
MAP2
MARCKSL1
MEX3A
MTURN
MTX2
MYEF2
MYO1B
NDRG4
NMNAT2
NSG2
PFDN6
PGAP1
PHOX2A
PHYHIPL
PPM1G
PROX1
PUM2
RAP2A
RBP1
RGS4
RGS5
RIMS3
RNF144A
RRM2
RUFY3
SCAMP5
SCG3
SDC2
SELENOI
SH3BGRL2
SLC25A29
SLC5A6
SNAP25
SORD
SOX11
STMN2
STXBP1
SYP
SYT11
TCF4
TIMM13
TIPRL
TMEM170B
TMOD2
TUBA1A
TUBB2B
UBE2S
UBE2T
ZNF704
", what="")

A7 <- scan(text="AHNAK
ASPH
ATP2B4
CD151
CD59
CNN2
COL1A1
COL4A1
COL4A2
DCBLD2
FSTL1
GPX8
HMGA2
IFITM2
IFITM3
ITGA3
ITGB5
KIRREL1
LPP
MSRB3
MYH9
NOTCH2
PAWR
PDIA4
PLS3
PODXL
PRNP
RAI14
RCN1
S100A13
SDC4
SMAD3
SPARC
SQSTM1
TPM1
YAP1
YBX3
ZYX", what="")
```

# 4.10.2) Define List of (MES) Genes of Interest

# There was an error so I removed some lines of code in 4.10.3

```{r, eval=F}
for (term in A5){
# library
library(ggplot2)

#my_gene <- readline(prompt="Enter Gene of Interest: ")
# Gene hard name: B4GALNT1

# DF_Identity_Pheno_Col <- add_column(GSVA_GeneExpression_SampleIDs, ID_Pheno_Col = eff_vector, .after = "Identity")

# Use DF that was not filtered by terms
DF_Identity_Pheno_Col <- GSVA_GeneExpression_Selected_Transposed

##_______________________________________________________##
# 1) Join DFs and ensure as.data.frame, but w/ all terms

DF_Identity_Pheno_Col["Sample_ID"] <- rownames(DF_Identity_Pheno_Col)

dt1 <- data.table(cell_line_pheno_RNA_Seq_Realignment, key = "FILE_NAME") 
dt2 <- data.table(DF_Identity_Pheno_Col, key = "Sample_ID")
joined.dt1.dt.2 <- dt1[dt2]
DF_Identity_Pheno_Col <- dt1[dt2]

# Make tarms of interest, numeric
for (term in A7){
DF_Identity_Pheno_Col[[term]] <- as.numeric(DF_Identity_Pheno_Col[[term]])
}

##_______________________________________________________##
# 2) Prepare Data for Statistics

library(ggpubr)
library(rstatix)

# 2.1) Data preparation

# Transform `dose` into factor variable
df <- DF_Identity_Pheno_Col

# Select pair of interest
# df <- df[ which(
#   df$Cell_line=="55N" |
#      df$Cell_line=="5S" ), ]

df$Identity <- as.factor(df$Identity)

# Add a random grouping variable
df$group <- factor(rep(c("grp1", "grp2"), 24))

##_______________________________________________________##
# 3) Facet wrap by one variable (Pheno)

##______________________________________##
# 3.1) Constructs Statistics Object

my_gene_input <- term
var_name <- 'my_gene_output'
assign(var_name, as.name(my_gene_input))
my_gene_output

# stat.test <- df %>%
#   group_by(Pheno) %>%
#   t_test( PHOX2B ~ Identity) %>%
#   adjust_pvalue(method = "bonferroni") %>%
#   add_significance()
# # View(stat.test) 

##______________________________________##
# 3.2) Apply T-test in Identity

# Identity (ADRN vs. MES)
t.test(df[[my_gene_output]][df$Identity=="ADRN"], df[[my_gene_output]][df$Identity=="MES"])$p.value
# Hypoxia
t_pval_H <- t.test(df[[my_gene_output]][df$Identity=="ADRN"][df$Pheno=="H"], df[[my_gene_output]][df$Identity=="MES"][df$Pheno=="H"])$p.value
# Normoxia
t_pval_N <- t.test(df[[my_gene_output]][df$Identity=="ADRN"][df$Pheno=="N"], df[[my_gene_output]][df$Identity=="MES"][df$Pheno=="N"])$p.value

ggboxplot(df, x = "Identity", y = my_gene_input,
          color = "Cell_line", palette = "jco",
          add = "jitter") + stat_compare_means(method = "t.test")

##______________________________________##
# 3.3) Construct Boxplot and Facet (by Identity)

# Create a box plot
bxp <- ggboxplot(
  df, x = "Identity", y = "PHOX2B", 
  color = "Cell_line",
  facet.by = "Pheno"
  )

##______________________________________##
# 3.4) Add p-values

# Make facet and add p-values
stat.test <- stat.test %>% add_xy_position(x = "Identity")
bxp + stat_pvalue_manual(stat.test)

##_______________________________________________________##
# 4) Facet wrap by one variable (Identity)

# 4.1) Facet wrap by one variable

stat.test <- df %>%
  group_by(Identity) %>%
  t_test(PHOX2B ~ Pheno) %>%
  adjust_pvalue(method = "bonferroni") %>%
  add_significance()
# View(stat.test) 

##______________________________________##
# 4.2) Apply T-test

##______________________________________##
# 4.2.1) Apply T-test in Pheno (Hypoxia)

#term <- "SPARC"
my_gene_input <- term
var_name <- 'my_gene_output'
assign(var_name, as.name(my_gene_input))
my_gene_output

# ADRN Pheno (Normoxia vs. Hypoxia)
t_pval_ADN_HStat <- t.test(df[[my_gene_output]][df$Identity=="ADRN"][df$Pheno=="N"], df[[my_gene_output]][df$Identity=="ADRN"][df$Pheno=="H"])$p.value

# MES Pheno (Normoxia vs. Hypoxia)
t_pval_MES_HStat <- t.test(df[[my_gene_output]][df$Identity=="MES"][df$Pheno=="N"], df[[my_gene_output]][df$Identity=="MES"][df$Pheno=="H"])$p.value

df_adrn <- df[df$Identity=="ADRN"]

##______________________________________##
# 4.2.2) Plot if significant

ifelse(
  t_pval_ADN_HStat < 0.05, 
  print(ggboxplot(df_adrn, x = "Pheno", y = term,
          color = "Cell_line", palette = "jco",
          add = "jitter") + stat_compare_means(method = "t.test")),
  print(paste0(term, " p-val is ", t_pval_ADN_HStat))
)

##______________________________________##
# 4.3) Construct Boxplot and Facet (by Pheno)

# Create a box plot
plot_object <- ggboxplot(
  df, x = "Pheno", y = "PHOX2B",
  color = "Cell_line",
  facet.by = "Identity"
  )

##______________________________________##
# 4.4) Add p-values (Pheno)

# Make facet and add p-values
stat.test <- stat.test %>% add_xy_position(x = "Pheno")
plot_object + stat_pvalue_manual(stat.test, bracket.nudge.y = c(.1, .2), hide.ns = FALSE,
    label = "{p.adj} ({p.adj.signif})") # Included space between p-value and asterisks
}
```

# 4.10.3) As clean as possible

* This algorithm plots gene expression in all cell lines;
* These genes comes from Alex's definitions of ADRN and MES genes;
* Criteria if interest as agreed on 03/21/2022 is that genes are ADRN with decrease in expression in ADRN cells by hypoxia
* Observe that some of these genes already fall into the desired criteria, although we have not yet openly used an algorithm to force identification of ADRN genes down-regulated by hypoxia;


```{r}
my_list_input <- "A5"
var_name <- 'my_list_output'
assign(var_name, as.name(my_list_input))
my_list_output

for (term in A5){
print(paste0("Before make numeric, starting with term,  ", term))
library(ggplot2)

# Use DF that was not filtered by terms
DF_Identity_Pheno_Col <- GSVA_GeneExpression_Selected_Transposed

##_______________________________________________________##
# 1) Join DFs and ensure as.data.frame, but w/ all terms

DF_Identity_Pheno_Col["Sample_ID"] <- rownames(DF_Identity_Pheno_Col)

dt1 <- data.table(cell_line_pheno_RNA_Seq_Realignment, key = "FILE_NAME") 
dt2 <- data.table(DF_Identity_Pheno_Col, key = "Sample_ID")
joined.dt1.dt.2 <- dt1[dt2]
DF_Identity_Pheno_Col <- dt1[dt2]

# Make tarms of interest, numeric
# If use term here, will make term store the wrong value
for (term_to_numeric in A5){
DF_Identity_Pheno_Col[[term_to_numeric]] <- as.numeric(DF_Identity_Pheno_Col[[term_to_numeric]])
}

##_______________________________________________________##
# 2) Prepare Data for Statistics

library(ggpubr)
library(rstatix)

# 2.1) Data preparation

# Transform `dose` into factor variable
df <- DF_Identity_Pheno_Col

df$Identity <- as.factor(df$Identity)

# Add a random grouping variable
df$group <- factor(rep(c("grp1", "grp2"), 24))





# 4.2) Apply T-test

##______________________________________##
# 4.2.1) Apply T-test in Pheno (Hypoxia)

#term <- "SPARC"
my_gene_input <- term
var_name <- 'my_gene_output'
assign(var_name, as.name(my_gene_input))
my_gene_output

# ADRN Pheno (Normoxia vs. Hypoxia)
t_pval_ADN_HStat <- t.test(df[[my_gene_output]][df$Identity=="ADRN"][df$Pheno=="N"], df[[my_gene_output]][df$Identity=="ADRN"][df$Pheno=="H"])$p.value

df_adrn <- df[df$Identity=="ADRN"]

##______________________________________##
# 4.2.2) Plot if significant

# ifelse(
#   t_pval_ADN_HStat < 0.05, 
#   (print(paste0(ggboxplot(df_adrn, x = "Pheno", y = term,
#           color = "Cell_line", palette = "jco",
#           add = "jitter") + stat_compare_means(method = "t.test"), "term is ", term))),
#   print(paste0(term, " p-val is ", t_pval_ADN_HStat))
# )

print(paste0("term is ", term))


ifelse(
  t_pval_ADN_HStat < 0.05,
print(ggboxplot(df_adrn, x = "Pheno", y = term,
          color = "Cell_line", palette = "jco",
          add = "jitter") + stat_compare_means(method = "t.test")),
print(paste0(term, " p-val is ", t_pval_ADN_HStat))
)
}
```

# 5) Identify ADRN genes down-regulated by hypoxia in every isogenic cell pair

# 5.1) Combined ADRN cells

* Plots means of ADRN compared to MES, not each cell line isolated

* Here I had to play with isolating the Means from the T-test as done to isolating the p-value

```{r}
my_list_input <- "A5"
var_name <- 'my_list_output'
assign(var_name, as.name(my_list_input))
my_list_output

for (term in A5){
print(paste0("Before make numeric, starting with term,  ", term))
library(ggplot2)

# Use DF that was not filtered by terms
DF_Identity_Pheno_Col <- GSVA_GeneExpression_Selected_Transposed

##_______________________________________________________##
# 1) Join DFs and ensure as.data.frame, but w/ all terms

DF_Identity_Pheno_Col["Sample_ID"] <- rownames(DF_Identity_Pheno_Col)

dt1 <- data.table(cell_line_pheno_RNA_Seq_Realignment, key = "FILE_NAME") 
dt2 <- data.table(DF_Identity_Pheno_Col, key = "Sample_ID")
joined.dt1.dt.2 <- dt1[dt2]
DF_Identity_Pheno_Col <- dt1[dt2]

# Make tarms of interest, numeric
# If use term here, will make term store the wrong value
for (term_to_numeric in A5){
DF_Identity_Pheno_Col[[term_to_numeric]] <- as.numeric(DF_Identity_Pheno_Col[[term_to_numeric]])
}

##_______________________________________________________##
# 2) Prepare Data for Statistics

library(ggpubr)
library(rstatix)

# 2.1) Data preparation

# Transform `dose` into factor variable
df <- DF_Identity_Pheno_Col

df$Identity <- as.factor(df$Identity)

# Add a random grouping variable
df$group <- factor(rep(c("grp1", "grp2"), 24))

# 4.2) Apply T-test

##______________________________________##
# 4.2.1) Apply T-test in Pheno (Hypoxia)

#term <- "CHN1" # If this is left uncommented, only this term will be plotted
my_gene_input <- term
var_name <- 'my_gene_output'
assign(var_name, as.name(my_gene_input))
my_gene_output

# ADRN Pheno (Normoxia vs. Hypoxia)
t_pval_ADN_HStat <- t.test(df[[my_gene_output]][df$Identity=="ADRN"][df$Pheno=="N"], df[[my_gene_output]][df$Identity=="ADRN"][df$Pheno=="H"])$p.value

# Isolate Means
t_estimate_ADN_HStat <- t.test(df[[my_gene_output]][df$Identity=="ADRN"][df$Pheno=="N"], df[[my_gene_output]][df$Identity=="ADRN"][df$Pheno=="H"])$estimate

t_estimate_ADN_HStat_DF <- as.data.frame(t_estimate_ADN_HStat)

df_adrn <- df[df$Identity=="ADRN"]

##______________________________________##
# 4.2.2) Plot if significant

# Plot if the T-test means follow the desired trend for ADRN genes under Hypoxia.

print(paste0("term is ", term))

ifelse(
  (t_pval_ADN_HStat < 0.05) & 
    # Impose condition to print ADRN genes that are down-regulated by Hypoxia (ADRN cells combined)
    # The condition is based on Means of Normoxia and Hypoxia treatments
    (t_estimate_ADN_HStat_DF[1,1] > t_estimate_ADN_HStat_DF[2,1]),
print(ggboxplot(df_adrn, x = "Pheno", y = term,
          color = "Cell_line", palette = "jco",
          add = "jitter") + stat_compare_means(method = "t.test")),
print(paste0(term, " p-val is ", t_pval_ADN_HStat))
)
}
```

# 5.2) Impose Identification of genes if condition is valid (ADRN Genes decreased by Hypoxia) in Five ADRN Cells

This chunk generates the vector of genes for each cell line

```{r}
my_list_input <- "A5"
var_name <- 'my_list_output'
assign(var_name, as.name(my_list_input))
my_list_output

# List of ADRN cells to look at
cell_line_list <- c("WN", "55N", "SY5Y", "BE2", "NBLS")
# cell_line_list <- c("WN")
# cell_line_list <- c("55N")
# cell_line_list <- c("SY5Y")
# cell_line_list <- c("BE2")
cell_line_list <- c("NBLS")
# cell_line_list <- c("WN")

significant_adrn_down <-c()
significant_wn_down <-c()
significant_wn_down_character <- c("CADM1","CHN1", "CKB","CNRIP1","DPYSL2",
                                   "DPYSL5","FAM107B","GINS2","IGFBP2","MACROH2A2",
                                   "PHYHIPL","PUM2","RAP2A","RBP1","RRM2","SDC2",
                                   "SELENOI","SH3BGRL2","SLC5A6","UBE2S","UBE2T")
significant_55n_down <-c()
significant_55n_down_character <- c("CADM1","CPLX2","CRMP1","DPH2","DPYSL2","DPYSL5",
                                   "MYO1B","PHYHIPL","PPM1G","RBP1","SCAMP5","SDC2",
                                   "SLC25A29","SLC5A6","TCF4","TUBA1A","UBE2T")
significant_sy5y_down <-c()
significant_sy5y_down_character <-c("CHN1","CKB","CRMP1","DPYSL2","ELAVL4","FAM163A",
                                    "GINS2","IGFBP2","KIF5C","KLHL23","MTX2","NMNAT2",
                                    "PFDN6","RBP1","SDC2","SELENOI","SH3BGRL2","SLC5A6",
                                    "STXBP1","TCF4","TMOD2","TUBB2B","UBE2T")
significant_be2_down <-c()
significant_be2_down_character <- c("AKT3","CADM1","CAMK4","CCND1","CD24","CDC42EP3",
                                     "CHN1","CKB","CNRIP1","CPLX2","CRMP1","DPH2",
                                     "DPYSL5","ELAVL4","FAM107B","FAM163A",
                                     "GINS2","INA","KIF5C","KLHL23","MACROH2A2","MARCKSL1",
                                     "MYO1B","PFDN6","PPM1G","PROX1","RBP1","RIMS3","RNF144A",
                                     "RRM2","SCAMP5","SDC2","SH3BGRL2","SLC5A6","SOX11",
                                     "TCF4","TIMM13","TMOD2","UBE2S","UBE2T","ZNF704")
significant_nbls_down <-c()
significant_nbls_down_character <- c("CCND1","CHN1","CPLX2","CRMP1","DPYSL2","DPYSL5","ELAVL4",
                                  "FAM163A","GDAP1","GINS2","IGFBP2","KIF5C","KLF7","KLHL23",
                                  "MAP2","MTX2","MYO1B","NDRG4","NMNAT2","PFDN6","PHYHIPL",
                                  "RBP1","RIMS3","RNF144A","SDC2","SLC25A29","SLC5A6","SOX11",
                                  "TCF4", "TMOD2","TUBA1A","TUBB2B","UBE2T")

for (cell_line_input in cell_line_list) {

for (term in A5){
print(paste0("Before make numeric, starting with term,  ", term))
library(ggplot2)

# Use DF that was not filtered by terms
DF_Identity_Pheno_Col <- GSVA_GeneExpression_Selected_Transposed

##_______________________________________________________##
# 1) Join DFs and ensure as.data.frame, but w/ all terms

DF_Identity_Pheno_Col["Sample_ID"] <- rownames(DF_Identity_Pheno_Col)

dt1 <- data.table(cell_line_pheno_RNA_Seq_Realignment, key = "FILE_NAME") 
dt2 <- data.table(DF_Identity_Pheno_Col, key = "Sample_ID")
joined.dt1.dt.2 <- dt1[dt2]
DF_Identity_Pheno_Col <- dt1[dt2]

# Make tarms of interest, numeric
# If use term here, will make term store the wrong value
for (term_to_numeric in A5){
DF_Identity_Pheno_Col[[term_to_numeric]] <- as.numeric(DF_Identity_Pheno_Col[[term_to_numeric]])
}

##_______________________________________________________##
# 2) Prepare Data for Statistics

library(ggpubr)
library(rstatix)

# 2.1) Data preparation

# Transform `dose` into factor variable
df <- DF_Identity_Pheno_Col

df$Identity <- as.factor(df$Identity)

# Add a random grouping variable
df$group <- factor(rep(c("grp1", "grp2"), 24))

# 4.2) Apply T-test

##______________________________________##
# 4.2.1) Apply T-test in Pheno (Hypoxia)

#term <- "CHN1" # If this is left uncommented, only this term will be plotted
my_gene_input <- term
var_name <- 'my_gene_output'
assign(var_name, as.name(my_gene_input))
my_gene_output

# ADRN Pheno (Normoxia vs. Hypoxia)
t_pval_ADN_HStat <- t.test(df[[my_gene_output]][df$Cell_line==cell_line_list][df$Pheno=="N"], df[[my_gene_output]][df$Cell_line==cell_line_list][df$Pheno=="H"])$p.value

# Isolate Means by/for cell line list
t_estimate_ADN_HStat <- t.test(df[[my_gene_output]][df$Cell_line==cell_line_list][df$Pheno=="N"], df[[my_gene_output]][df$Cell_line==cell_line_list][df$Pheno=="H"])$estimate

t_estimate_ADN_HStat_DF <- as.data.frame(t_estimate_ADN_HStat)

df_adrn <- df[df$Cell_line==cell_line_input]

##______________________________________##
# 4.2.2) Store in DF if significant

# Plot if the T-test means follow the desired trend for ADRN genes under Hypoxia.

print(paste0("term is ", term))

cell_line_input # each cell line in the vector-declaring ADRN cells
cell_line_sig_Genes <- paste0("cell_line_sig_Genes_", cell_line_input)

# my_list_input <- "A5"
# var_name <- 'my_list_output'
# assign(var_name, as.name(my_list_input))
# my_list_output

my_list_input <- cell_line_sig_Genes
var_name <- 'my_list_output'
assign(var_name, as.name(my_list_input))
my_list_output

ifelse(
  (t_pval_ADN_HStat < 0.05) & 
    # Impose condition to print ADRN genes that are down-regulated by Hypoxia (ADRN cells combined)
    # The condition is based on Means of Normoxia and Hypoxia treatments
    (t_estimate_ADN_HStat_DF[1,1] > t_estimate_ADN_HStat_DF[2,1]),
  # Impose creation of list with ADRN genes that go down by Hypoxia
  significant_nbls_down <- c(significant_nbls_down, my_gene_output)
,
print(paste0(term, " p-val is ", t_pval_ADN_HStat))
)

}
}

##______________________________________##
# 5) Overlap and print genes in the final list

# # Overlap
# intersect_isogenic <- intersect(intersect(significant_wn_down,significant_55n_down),significant_sy5y_down)
# intersect_be2_nbls <- intersect(significant_be2_down,significant_nbls_down)
# intersect_final <- intersect(intersect_isogenic, intersect_be2_nbls)
# 
# # Print final list
# for (i in intersect_final){
#   print(i)
# }
```


# 5.3) List of ADRN genes Down-regulated by Hypoxia in ADRN Cells from Alex's list

```{r}
intersect_final <- c("SDC2", "SLC5A6", "UBE2T")

intersect_final <- scan(text="AHSA1
ASRGL1
ATP6V0E2
CSE1L
FKBP4
IGSF3
ISL1
LSM4
NET1
NRSN1
RBP1
TAGLN3
TFAP2B
UBE2T
", what="")



my_list_input <- "intersect_final"
var_name <- 'my_list_output'
assign(var_name, as.name(my_list_input))
my_list_output

for (term in intersect_final){
print(paste0("Before make numeric, starting with term,  ", term))
library(ggplot2)

# Use DF that was not filtered by terms
DF_Identity_Pheno_Col <- GSVA_GeneExpression_Selected_Transposed

##_______________________________________________________##
# 1) Join DFs and ensure as.data.frame, but w/ all terms

DF_Identity_Pheno_Col["Sample_ID"] <- rownames(DF_Identity_Pheno_Col)

dt1 <- data.table(cell_line_pheno_RNA_Seq_Realignment, key = "FILE_NAME") 
dt2 <- data.table(DF_Identity_Pheno_Col, key = "Sample_ID")
joined.dt1.dt.2 <- dt1[dt2]
DF_Identity_Pheno_Col <- dt1[dt2]

# Make tarms of interest, numeric
# If use term here, will make term store the wrong value
for (term_to_numeric in intersect_final){
DF_Identity_Pheno_Col[[term_to_numeric]] <- as.numeric(DF_Identity_Pheno_Col[[term_to_numeric]])
}

##_______________________________________________________##
# 2) Prepare Data for Statistics

library(ggpubr)
library(rstatix)

# 2.1) Data preparation

# Transform `dose` into factor variable
df <- DF_Identity_Pheno_Col

df$Identity <- as.factor(df$Identity)

# Add a random grouping variable
df$group <- factor(rep(c("grp1", "grp2"), 24))

# 4.2) Apply T-test

##______________________________________##
# 4.2.1) Apply T-test in Pheno (Hypoxia)

#term <- "CHN1" # If this is left uncommented, only this term will be plotted
my_gene_input <- term
var_name <- 'my_gene_output'
assign(var_name, as.name(my_gene_input))
my_gene_output

# ADRN Pheno (Normoxia vs. Hypoxia)
t_pval_ADN_HStat <- t.test(df[[my_gene_output]][df$Identity=="ADRN"][df$Pheno=="N"], df[[my_gene_output]][df$Identity=="ADRN"][df$Pheno=="H"])$p.value

# Isolate Means
t_estimate_ADN_HStat <- t.test(df[[my_gene_output]][df$Identity=="ADRN"][df$Pheno=="N"], df[[my_gene_output]][df$Identity=="ADRN"][df$Pheno=="H"])$estimate

t_estimate_ADN_HStat_DF <- as.data.frame(t_estimate_ADN_HStat)

df_adrn <- df[df$Identity=="ADRN"]

##______________________________________##
# 4.2.2) Plot if significant

# Plot if the T-test means follow the desired trend for ADRN genes under Hypoxia.

print(paste0("term is ", term))

ifelse(
  (t_pval_ADN_HStat < 0.05) & 
    # Impose condition to print ADRN genes that are down-regulated by Hypoxia (ADRN cells combined)
    # The condition is based on Means of Normoxia and Hypoxia treatments
    (t_estimate_ADN_HStat_DF[1,1] > t_estimate_ADN_HStat_DF[2,1]),
print(ggboxplot(df_adrn, x = "Pheno", y = term,
          color = "Cell_line", palette = "jco",
          add = "jitter") + stat_compare_means(method = "t.test")),
print(paste0(term, " p-val is ", t_pval_ADN_HStat))
)
}
```

# 6) Hypoxia decrease in ADRN All Van Groningen ADRN Genes

# 6.1.1) Impose Identification of genes if condition is valid (ADRN Genes decreased by Hypoxia) in Five ADRN Cells

This chunk generates the vector of genes for each cell line

```{r}
# ADRBK2 C7orf55 EPB41L4A-AS1 FAM60A HMP19 HN1 KIAA1211 LINC00888 LOC100507194 LOC101928409 MIAT
intersect_final <- scan(text=
"ABCA3	ABCB1	ABLIM1	ACOT7	ACTL6B	ACVR1B	ADAM22	ADCYAP1R1	ADGRB3	AGTPBP1	AHSA1	AKAP1	AKAP12	ALK	ANK2	ANKRD46	ANP32A	AP1S2	ARHGEF7	ARL6IP1	ASCL1	ASRGL1	ATCAY	ATL1	ATP6V0E2	ATP6V1B2	AUTS2	BEND4	BEX1	BEX2	BIRC5	BMP7	BMPR1B	C11orf95	C14orf132	C3orf14	C4orf48	C7orf55	CACNA1B	CACNA2D2	CADM1	CAMSAP1	CCDC167	CCND1	CCNI	CCP110	CCSAP	CD200	CDC42EP3	CDCA5	CDKN2C	CDKN3	CELF2	CENPU	CENPV	CEP44	CERK	CETN3	CHGA	CHGB	CHML	CHRNA3	CKB	CLASP2	CLGN	CRH	CRMP1	CSE1L	CXADR	CXCR4	CYFIP2	CYGB	DACH1	DAPK1	DBH	DCX	DDC	DDX39A	DIABLO	DKK1	DLK1	DNAJB1	DNAJC6	DNAJC9	DNER	DPYSL2	DPYSL3	DPYSL5	DTD1	DUSP4	EEF1A2	EIF1B	ELAVL2	ELAVL3	ELAVL4	EML4	EML6	ENDOG	ENO2	EPB41L4A-AS1	ESRRG	EVL	EXOC5	EYA1	FABP6	FAM107B	FAM155A	FAM163A	FAM167A	FAM169A	FAM171B	FAM60A	FAXC	FBLL1	FBXO8	FEV	FHOD3	FIGNL1	FKBP1B	FKBP4	FOXM1	FOXO3	FSD1	FZD3	GABRB3	GAL	GAP43	GATA2	GATA3	GCH1	GDAP1	GDAP1L1	GDI1	GDPD1	GGCT	GGH	GLCCI1	GLDC	GLRX	GMNN	GNB1	GNG4	GPR22	GPR27	GRB10	GRIA2	H1FX	HAND1	HAND2-AS1	HES6	HEY1	HK2	HMGA1	HMP19	HN1	HNRNPA0	HS6ST2	ICA1	IGFBPL1	IGSF3	INA	INO80C	INSM1	INSM2	IRS2	ISL1	KDM1A	KIAA1211	KIDINS220	KIF15	KIF1A	KIF21A	KIF2A	KIF5C	KLC1	KLF13	KLF7	KLHL13	KLHL23	KNSTRN	L1CAM	LEPROTL1	LIN28B	LINC00888	LMO3	LOC100507194	LOC101928409	LRRTM2	LSM3	LSM4	LYN	MAGI3	MANEAL	MAP1B	MAP2	MAP6	MAPK8	MAPT	MCM2	MCM6	MCM7	MIAT	MMD	MRPL48	MSH6	MSI2	MTCL1	MXI1	MYBL2	MYEF2	MYO5A	MYRIP	NANOS1	NAP1L5	NAPB	NARS2	NBEA	NCAM1	NCAN	NCOA7	NCS1	NEFL	NEFM	NELFCD	NELL2	NET1	NFIL3	NGRN	NMNAT2	NNAT	NOL4	NPTX2	NPY	NRCAM	NRSN1	NSG1	NUDT11	NUF2	NUSAP1	OLA1	OLFM1	PARP6	PBK	PBX3	PDK1	PEG3	PHF21B	PHOX2A	PHOX2B	PHPT1	PHYHIPL	PIK3R1	PKIA	PLPPR5	PNMA2	POLB	POPDC3	PPM1E	PPP1R9A	PPP2R3C	PRC1	PRCD	PRIM1	PRPH	PRSS12	PRSS3	PTS	QDPR	RAB33A	RAB6B	RALGDS	RANBP1	RBBP8	RBMS3	RBP1	REC8	REEP1	RET	RFC4	RGS17	RGS5	RIMBP2	RIMS3	RNF144A	RNF150	RNF157	RNF165	RNFT2	RPS6KA2	RRM2	RTN1	RTN2	RUFY3	RUNDC3A	RUNDC3B	SATB1	SBK1	SCAMP5	SCG2	SCG3	SCN3A	SEC11C	SERP2	SETD7	SHC3	SHD	SIX3	SLC10A4	SLC35G2	SLIT1	SLIT3	SNAP25	SNAP91	SOX11	ST3GAL6	STMN2	STMN4	STRA6	STXBP1	SV2C	SYNPO2	SYT1	SYT4	TACC2	TAGLN3	TBC1D30	TBPL1	TCEAL7	TDG	TENM4	TFAP2B	TH	THSD7A	TIAM1	TMEM108	TMEM178B	TMEM97	TMOD1	TMOD2	TMTC4	TOX2	TRAP1	TSPAN13	TSPAN7	TTC8	TUB	TUBB2A	TUBB2B	TUBB3	TUBB4B	UBE2C	UBE2T	UCP2	UNC79	VRK1	ZNF195	ZNF22	ZNF24	ZNF512	ZNF536	ZNF704	ZNF711	ZNF738	ZNF91	ZWILCH", what="")

intersect_final <- scan(text=
"ABCA3	ABCB1	ABLIM1	ACOT7	ACTL6B ACVR1B	ADAM22	ADCYAP1R1	ADGRB3	AGTPBP1	AHSA1	AKAP1	AKAP12	ALK	ANK2 ANKRD46	ANP32A	AP1S2	ARHGEF7	ARL6IP1	ASCL1	ASRGL1	ATCAY	ATL1	ATP6V0E2	ATP6V1B2	AUTS2	BEND4	BEX1	BEX2	BIRC5 BMP7 BMPR1B	C11orf95	C14orf132	C3orf14 C4orf48 CACNA1B	CACNA2D2	CADM1 CAMSAP1	CCDC167	CCND1	CCNI	CCP110 CCSAP	CD200	CDC42EP3	CDCA5	CDKN2C	CDKN3	CELF2	CENPU	CENPV	CEP44	CERK	CETN3	CHGA	CHGB	CHML	CHRNA3	CKB	CLASP2 CLGN	CRH	CRMP1	CSE1L	CXADR	CXCR4	CYFIP2	CYGB	DACH1	DAPK1	DBH	DCX	DDC	DDX39A	DIABLO	DKK1	DLK1	DNAJB1	DNAJC6 DNAJC9	DNER	DPYSL2	DPYSL3	DPYSL5	DTD1	DUSP4	EEF1A2	EIF1B	ELAVL2	ELAVL3	ELAVL4	EML4	EML6	ENDOG	ENO2	ESRRG	EVL	EXOC5	EYA1 FABP6	FAM107B	FAM155A FAM163A	FAM167A	FAM169A FAM171B FAXC	FBLL1	FBXO8 FEV	FHOD3	FIGNL1	FKBP1B	FKBP4	FOXM1	FOXO3	FSD1	FZD3	GABRB3	GAL	GAP43	GATA2	GATA3	GCH1	GDAP1	GDAP1L1	GDI1 HMGA1 HNRNPA0 HS6ST2 ICA1 IGFBPL1 IGSF3 INA INO80C	INSM1	INSM2 IRS2	ISL1	KDM1A KIDINS220 KIF15	KIF1A	KIF21A	KIF2A	KIF5C	KLC1	KLF13	KLF7	KLHL13 KLHL23 KNSTRN L1CAM LEPROTL1	LIN28B LMO3 LRRTM2 LSM3	LSM4	LYN	MAGI3	MANEAL	MAP1B	MAP2	MAP6	MAPK8	MAPT	MCM2	MCM6	MCM7 MMD	MRPL48	MSH6	MSI2	MTCL1	MXI1	MYBL2	MYEF2	MYO5A	MYRIP	NANOS1	NAP1L5	NAPB	NARS2	NBEA	NCAM1	NCAN NCOA7	NCS1	NEFL	NEFM	NELFCD	NELL2	NET1	NFIL3	NGRN	NMNAT2	NNAT	NOL4	NPTX2	NPY	NRCAM	NRSN1	NSG1	NUDT11	NUF2 NUSAP1	OLA1	OLFM1	PARP6	PBK	PBX3	PDK1	PEG3	PHF21B	PHOX2A	PHOX2B	PHPT1	PHYHIPL	PIK3R1	PKIA	PLPPR5	PNMA2 POLB	POPDC3	PPM1E	PPP1R9A	PPP2R3C	PRC1	PRCD	PRIM1	PRPH	PRSS12	PRSS3	PTS	QDPR	RAB33A	RAB6B	RALGDS	RANBP1 RBBP8	RBMS3	RBP1	REC8	REEP1	RET	RFC4	RGS17	RGS5	RIMBP2	RIMS3	RNF144A	RNF150	RNF157	RNF165	RNFT2	RPS6KA2	RRM2 RTN1	RTN2	RUFY3	RUNDC3A	RUNDC3B	SATB1	SBK1	SCAMP5	SCG2	SCG3	SCN3A	SEC11C	SERP2	SETD7	SHC3	SHD	SIX3	SLC10A4 SLC35G2	SLIT1	SLIT3	SNAP25	SNAP91	SOX11	ST3GAL6	STMN2	STMN4	STRA6	STXBP1	SV2C	SYNPO2	SYT1	SYT4	TACC2	TAGLN3 TBC1D30	TBPL1	TCEAL7	TDG	TENM4	TFAP2B	TH	THSD7A	TIAM1	TMEM108	TMEM178B	TMEM97	TMOD1	TMOD2	TMTC4	TOX2	TRAP1 TSPAN13	TSPAN7	TTC8	TUB	TUBB2A	TUBB2B	TUBB3	TUBB4B	UBE2C	UBE2T	UCP2	UNC79	VRK1	ZNF195	ZNF22	ZNF24	ZNF512 ZNF536	ZNF704	ZNF711	ZNF738	ZNF91	ZWILCH", what = "")

my_list_input <- "intersect_final"
var_name <- 'my_list_output'
assign(var_name, as.name(my_list_input))
my_list_output

# List of ADRN cells to look at
cell_line_list <- c("WN", "55N", "SY5Y", "BE2", "NBLS")
# cell_line_list <- c("WN")
# cell_line_list <- c("55N")
# cell_line_list <- c("SY5Y")
# cell_line_list <- c("BE2")
cell_line_list <- c("NBLS")

# Define name for the list of interest
significant_adrnCell_down <-c()

for (cell_line_input in cell_line_list) {

for (term in intersect_final){
  tryCatch({
print(paste0("Before make numeric, starting with term,  ", term))
library(ggplot2)

# Use DF that was not filtered by terms
DF_Identity_Pheno_Col <- GSVA_GeneExpression_Selected_Transposed

##_______________________________________________________##
# 1) Join DFs and ensure as.data.frame, but w/ all terms

DF_Identity_Pheno_Col["Sample_ID"] <- rownames(DF_Identity_Pheno_Col)

dt1 <- data.table(cell_line_pheno_RNA_Seq_Realignment, key = "FILE_NAME") 
dt2 <- data.table(DF_Identity_Pheno_Col, key = "Sample_ID")
joined.dt1.dt.2 <- dt1[dt2]
DF_Identity_Pheno_Col <- dt1[dt2]

# Make tarms of interest, numeric
# If use term here, will make term store the wrong value
for (term_to_numeric in intersect_final){
DF_Identity_Pheno_Col[[term_to_numeric]] <- as.numeric(DF_Identity_Pheno_Col[[term_to_numeric]])
}

##_______________________________________________________##
# 2) Prepare Data for Statistics

library(ggpubr)
library(rstatix)

# 2.1) Data preparation

# Transform `dose` into factor variable
df <- DF_Identity_Pheno_Col

df$Identity <- as.factor(df$Identity)

# Add a random grouping variable
df$group <- factor(rep(c("grp1", "grp2"), 24))

# 4.2) Apply T-test

##______________________________________##
# 4.2.1) Apply T-test in Pheno (Hypoxia)

#term <- "CHN1" # If this is left uncommented, only this term will be plotted
my_gene_input <- term
var_name <- 'my_gene_output'
assign(var_name, as.name(my_gene_input))
my_gene_output

# ADRN Pheno (Normoxia vs. Hypoxia)
t_pval_ADN_HStat <- t.test(df[[my_gene_output]][df$Cell_line==cell_line_list][df$Pheno=="N"], df[[my_gene_output]][df$Cell_line==cell_line_list][df$Pheno=="H"])$p.value

# Isolate Means
t_estimate_ADN_HStat <- t.test(df[[my_gene_output]][df$Cell_line==cell_line_list][df$Pheno=="N"], df[[my_gene_output]][df$Cell_line==cell_line_list][df$Pheno=="H"])$estimate

t_estimate_ADN_HStat_DF <- as.data.frame(t_estimate_ADN_HStat)

df_adrn <- df[df$Cell_line==cell_line_input]

##______________________________________##
# 4.2.2) Store in DF if significant

# Plot if the T-test means follow the desired trend for ADRN genes under Hypoxia.

print(paste0("term is ", term))

cell_line_input # each cell line in the vector-declaring ADRN cells
cell_line_sig_Genes <- paste0("cell_line_sig_Genes_", cell_line_input)

# my_list_input <- "A5"
# var_name <- 'my_list_output'
# assign(var_name, as.name(my_list_input))
# my_list_output

my_list_input <- cell_line_sig_Genes
var_name <- 'my_list_output'
assign(var_name, as.name(my_list_input))
my_list_output

ifelse(
  (t_pval_ADN_HStat < 0.05) & 
    # Impose condition to print ADRN genes that are down-regulated by Hypoxia (ADRN cells combined)
    # The condition is based on Means of Normoxia and Hypoxia treatments
    (t_estimate_ADN_HStat_DF[1,1] > t_estimate_ADN_HStat_DF[2,1]),
  # Impose creation of list with ADRN genes that go down by Hypoxia
  significant_adrnCell_down <- c(significant_adrnCell_down, my_gene_output)
,
print(paste0(term, " p-val is ", t_pval_ADN_HStat))
)
}, error=function(e){cat("ERROR :",conditionMessage(e), "\n")})
}
}

# significant_wn_down <- significant_adrnCell_down
# significant_55n_down <- significant_adrnCell_down
# significant_sy5y_down <- significant_adrnCell_down
# significant_be2_down <- significant_adrnCell_down
significant_nbls_down <- significant_adrnCell_down

##______________________________________##
# 5) Overlap and print genes in the final list

# Overlap
intersect_isogenic <- intersect(intersect(significant_wn_down,significant_55n_down),significant_sy5y_down)
intersect_be2_nbls <- intersect(significant_be2_down,significant_nbls_down)
intersect_final <- intersect(intersect_isogenic, intersect_be2_nbls)

# Print final list
for (i in intersect_final){
  print(i)
}
```

# 6.1.2)  Use our LFC (Search the Candidate MES Genes going Up in Hypoxia)

This chunk generates the vector of genes for each cell line to be intersected

```{r}
##_______________________________________________________##
# 1) Define range of Genes based on LFC

DF_RNA_Normoxia_ADRN_vs_MES <- as.data.frame(dds.RNA_Normoxia_ADRN_vs_MES.res)

# Split LFC
DF_select_ID_LFC_Pos <- DF_RNA_Normoxia_ADRN_vs_MES[ which(
  DF_RNA_Normoxia_ADRN_vs_MES$log2FoldChange > 2 & DF_RNA_Normoxia_ADRN_vs_MES$padj < 0.05), ]
# list_LFC10_Range_Pos <- rownames(DF_select_ID_LFC_Pos)
DF_ordered_LFC_Pos <- DF_select_ID_LFC_Pos[order(-DF_select_ID_LFC_Pos$log2FoldChange),] 
DF_ordered_LFC_Range_Pos <- DF_ordered_LFC_Pos[1000:1050,] # Choose gene ranges to evaluate
list_LFC_Range_Pos <- rownames(DF_ordered_LFC_Range_Pos)

DF_select_ID_LFC_Neg <- DF_RNA_Normoxia_ADRN_vs_MES[ which(
  DF_RNA_Normoxia_ADRN_vs_MES$log2FoldChange < -2 & DF_RNA_Normoxia_ADRN_vs_MES$padj < 0.05), ]
list_LFC10_Range_Neg <- rownames(DF_select_ID_LFC_Neg)
DF_ordered_LFC_Neg <- DF_select_ID_LFC_Neg[order(DF_select_ID_LFC_Neg$log2FoldChange),]
DF_ordered_LFC_Range_Neg <- DF_ordered_LFC_Neg[1000:1050,]
list_LFC_Range_Neg <- rownames(DF_ordered_LFC_Range_Neg)

# intersect_final <- rownames(DF_ordered_LFC_Range_Neg)
# intersect_final <- first_list_interest
intersect_final <- c("VIM", "SPARC", "C1S", "COL5A2")
intersect_final <- list_LFC_Range_Neg
# intersect_final <- list_LFC10_Range_Neg

my_list_input <- "intersect_final"
var_name <- 'my_list_output'
assign(var_name, as.name(my_list_input))
my_list_output

# List of ADRN cells to look at
cell_line_list <- c("WN", "55N", "SY5Y", "BE2", "NBLS")
# cell_line_list <- c("WN")#  24
# cell_line_list <- c("55N")# 57
# cell_line_list <- c("SY5Y") # 45
# cell_line_list <- c("BE2") # 67
# cell_line_list <- c("NBLS") # 74


for (cell_line_input in cell_line_list) {
print(paste0("This starts for loop for cells in list. Cell line of list is: ", cell_line_input))
  term_count = 0
# Define name for the list of interest inside cell_line_input for loop.
# Otherwise the vector accumulates genes for following for loops.
significant_adrnCell_up <-c()

for (term in intersect_final){
  tryCatch({
term_count = term_count + 1
print(paste0("################################################################ For loop for term: ", term, " starts here"))
print(paste0("################################################################ In list cell_line_list, ", term," is element number: ", term_count))
print(paste0("#################################################################################################################"))

library(ggplot2)

# Use DF that was not filtered by terms
DF_Identity_Pheno_Col <- GSVA_GeneExpression_Selected_Transposed

##_______________________________________________________##
# 1) Join DFs and ensure as.data.frame, but w/ all terms
print(paste0("Now we will join Pheno information DF and Gene Expression DFs. Term in intersect_final is: ", term))

DF_Identity_Pheno_Col["Sample_ID"] <- rownames(DF_Identity_Pheno_Col)

dt1 <- data.table(cell_line_pheno_RNA_Seq_Realignment, key = "FILE_NAME") 
dt2 <- data.table(DF_Identity_Pheno_Col, key = "Sample_ID")
joined.dt1.dt.2 <- dt1[dt2]
DF_Identity_Pheno_Col <- dt1[dt2]

# Make tarms of interest, numeric
# If use term here, will make term store the wrong value
print(paste0("In this for loop, the print statement will be printed the same number of elements in list intersect_final"))
for (term_to_numeric in intersect_final){
print(paste0("In this for loop, we make sure term scores are numeric. Term is: ", term, " and term_to_numeric is :", term_to_numeric))
DF_Identity_Pheno_Col[[term_to_numeric]] <- as.numeric(DF_Identity_Pheno_Col[[term_to_numeric]])
}

##_______________________________________________________##
# 2) Prepare Data for Statistics

library(ggpubr)
library(rstatix)

# 2.1) Data preparation

# Transform `dose` into factor variable
df <- DF_Identity_Pheno_Col

df$Identity <- as.factor(df$Identity)

# Add a random grouping variable
df$group <- factor(rep(c("grp1", "grp2"), 24))

# 4.2) Apply T-test

##______________________________________##
# 4.2.1) Apply T-test in Pheno (Hypoxia)

print(paste0("Now we start t-test for current term. Here, my_gene_input equals to term and term is: ", term))

#term <- "CHN1" # If this is left uncommented, only this term will be plotted
my_gene_input <- term
var_name <- 'my_gene_output'
assign(var_name, as.name(my_gene_input))
my_gene_output

# ADRN Pheno (Normoxia vs. Hypoxia)
t_pval_ADN_HStat <- t.test(df[[my_gene_output]][df$Cell_line==cell_line_input][df$Pheno=="N"], df[[my_gene_output]][df$Cell_line==cell_line_input][df$Pheno=="H"])$p.value

# Isolate Means
t_estimate_ADN_HStat <- t.test(df[[my_gene_output]][df$Cell_line==cell_line_input][df$Pheno=="N"], df[[my_gene_output]][df$Cell_line==cell_line_input][df$Pheno=="H"])$estimate

t_estimate_ADN_HStat_DF <- as.data.frame(t_estimate_ADN_HStat)

df_adrn <- df[df$Cell_line==cell_line_input]

print(paste0("At the end of this part, t-test DF was built to test the effect of hypoxia in the expression of term ", term))

##______________________________________##
# 4.2.2) Store in DF if significant

# Plot if the T-test means follow the desired trend for ADRN genes under Hypoxia.

print(paste0("## 1) In this part, we will plot graph if the T-test means follow the desired trend for ADRN genes under Hypoxia for term ", term))
print(paste0("#######################################################################################################################"))

print(paste0("The cell line here is: ", cell_line_input))
print(paste0("Define significant genes for cell line ", cell_line_input))
cell_line_sig_Genes <- paste0("cell_line_sig_Genes_", cell_line_input)

# my_list_input <- "A5"
# var_name <- 'my_list_output'
# assign(var_name, as.name(my_list_input))
# my_list_output

my_list_input <- cell_line_sig_Genes
var_name <- 'my_list_output'
assign(var_name, as.name(my_list_input))
print(paste0("In this part, we define the object of cell line, my_list_output: ", my_list_output))
print(paste0("The values received by ", my_list_output, " are supposed to store sig genes for cell_line_input ", cell_line_input))
my_list_output

##______________________________________##
# 4.2.3) ifelse to ensure identity of cell line

print(paste0("Now we start different types of ifelse statements. Term is: ", term))

print(paste0("In the first ifelse statement, my_gene_input still equals to term. Term is: ", term))

print(paste0("In this statement, we are imposing the expression of MES genes increased by hypoxia in ADRN cells (N compared to H). Term is: ", term))

print(paste0("## 2) Ifelse Statements Significant and increases in hypoxia ", term))
print(paste0("#######################################################################################################################"))

print(paste0("p-val for term ", term, " is: ", t_pval_ADN_HStat, "for cell line ", cell_line_input))

ifelse(
  (t_pval_ADN_HStat < 0.05) & 
    # Impose condition to print ADRN genes that are down-regulated by Hypoxia (ADRN cells combined)
    # The condition is based on Means of Normoxia and Hypoxia treatments
    (t_estimate_ADN_HStat_DF[1,1] < t_estimate_ADN_HStat_DF[2,1]),
  #print(paste0("Append my_gene_output to significant_adrnCell_up))#&
  significant_adrnCell_up <- c(significant_adrnCell_up, my_gene_output)
,
print(paste0("Because p-val is not significant for ", term, " we show that p-val is ", t_pval_ADN_HStat))
)
}, error=function(e){cat("ERROR :",conditionMessage(e), "\n")})
print(paste0("for loop for each term in intersect_final finishes here for current term: ",
             term, 
             " But I moved to after ifelse statements."))
} # End of for loop for terms or genes

print(paste0("For loop just finished for term ", term, 
             ". Now we will start another ifelse statement to store genes for each cell line. Current cell line is ", 
             cell_line_input))

print(paste0("For term ", 
             term, 
             " check if cell line is WN. Cell line is: ", 
             cell_line_input))

print(paste0("## 2.1) ifelse statement to check cell_line_input is WN for term ", term))
print(paste0("#######################################################"))

if (cell_line_input=="WN") {
  #if( length(significant_adrnCell_up)!=0 ){
    significant_wn_up_store <- significant_adrnCell_up
  #}
}

if (cell_line_input=="55N") {
  # if( length(significant_adrnCell_up)!=0 ){
    significant_55n_up_store <- significant_adrnCell_up
  # }
}

if (cell_line_input=="SY5Y") {
  # if( length(significant_adrnCell_up)!=0 ){
    significant_sy5y_up_store <- significant_adrnCell_up
  # }
}

if (cell_line_input=="BE2") {
  # if( length(significant_adrnCell_up)!=0 ){
    significant_be2_up_store <- significant_adrnCell_up
  # }
}

if (cell_line_input=="NBLS") {
  # if( length(significant_adrnCell_up)!=0 ){
    significant_nbls_up_store <- significant_adrnCell_up
  # }
}



}

##______________________________________##
# 5) Overlap and print genes in the final list

# Overlap
intersect_isogenic_up <- intersect(
  intersect(
    significant_wn_up_store,significant_55n_up_store),significant_sy5y_up_store)
intersect_be2_nbls_up <- intersect(significant_be2_up_store,significant_nbls_up_store)
intersect_allCellsUp <- intersect(intersect_isogenic_up, intersect_be2_nbls_up)

# Print final list
for (i in intersect_allCellsUp){
  print(i)
}
```

# 6.1.3)  Use our LFC (Search the Candidate ADRN Genes going Down in Hypoxia)

```{r}
##_______________________________________________________##
# 1) Define range of Genes based on LFC

DF_RNA_Normoxia_ADRN_vs_MES <- as.data.frame(dds.RNA_Normoxia_ADRN_vs_MES.res)

# Split LFC
DF_select_ID_LFC_Pos <- DF_RNA_Normoxia_ADRN_vs_MES[ which(
  DF_RNA_Normoxia_ADRN_vs_MES$log2FoldChange > 3 & DF_RNA_Normoxia_ADRN_vs_MES$padj < 0.05), ]
# list_LFC10_Range_Pos <- rownames(DF_select_ID_LFC_Pos)
DF_ordered_LFC_Pos <- DF_select_ID_LFC_Pos[order(-DF_select_ID_LFC_Pos$log2FoldChange),] 
DF_ordered_LFC_Range_Pos <- DF_ordered_LFC_Pos[850:862,] # Choose gene ranges to evaluate
list_LFC_Range_Pos <- rownames(DF_ordered_LFC_Range_Pos)

DF_select_ID_LFC_Neg <- DF_RNA_Normoxia_ADRN_vs_MES[ which(
  DF_RNA_Normoxia_ADRN_vs_MES$log2FoldChange < -3 & DF_RNA_Normoxia_ADRN_vs_MES$padj < 0.05), ]
list_LFC10_Range_Neg <- rownames(DF_select_ID_LFC_Neg)
DF_ordered_LFC_Neg <- DF_select_ID_LFC_Neg[order(DF_select_ID_LFC_Neg$log2FoldChange),]
DF_ordered_LFC_Range_Neg <- DF_ordered_LFC_Neg[850:862,]
list_LFC_Range_Neg <- rownames(DF_ordered_LFC_Range_Neg)

# intersect_final <- rownames(DF_ordered_LFC_Range_Neg)
# intersect_final <- first_list_interest
intersect_final <- c("VIM", "SPARC", "C1S", "COL5A2")
intersect_final <- list_LFC_Range_Pos
# intersect_final <- list_LFC10_Range_Neg

my_list_input <- "intersect_final"
var_name <- 'my_list_output'
assign(var_name, as.name(my_list_input))
my_list_output

# List of ADRN cells to look at
cell_line_list <- c("WN", "55N", "SY5Y", "BE2", "NBLS")
# cell_line_list <- c("WN")#  24
# cell_line_list <- c("55N")# 57
# cell_line_list <- c("SY5Y") # 45
# cell_line_list <- c("BE2") # 67
# cell_line_list <- c("NBLS") # 74


for (cell_line_input in cell_line_list) {
print(paste0("This starts for loop for cells in list. Cell line of list is: ", cell_line_input))
  term_count = 0
# Define name for the list of interest inside cell_line_input for loop.
# Otherwise the vector accumulates genes for following for loops.
significant_adrnCell_down <-c()

for (term in intersect_final){
  tryCatch({
term_count = term_count + 1
print(paste0("################################################################ For loop for term: ", term, " starts here"))
print(paste0("################################################################ In list cell_line_list, ", term," is element number: ", term_count))
print(paste0("#################################################################################################################"))

library(ggplot2)

# Use DF that was not filtered by terms
DF_Identity_Pheno_Col <- GSVA_GeneExpression_Selected_Transposed

##_______________________________________________________##
# 1) Join DFs and ensure as.data.frame, but w/ all terms
print(paste0("Now we will join Pheno information DF and Gene Expression DFs. Term in intersect_final is: ", term))

DF_Identity_Pheno_Col["Sample_ID"] <- rownames(DF_Identity_Pheno_Col)

dt1 <- data.table(cell_line_pheno_RNA_Seq_Realignment, key = "FILE_NAME") 
dt2 <- data.table(DF_Identity_Pheno_Col, key = "Sample_ID")
joined.dt1.dt.2 <- dt1[dt2]
DF_Identity_Pheno_Col <- dt1[dt2]

# Make tarms of interest, numeric
# If use term here, will make term store the wrong value
print(paste0("In this for loop, the print statement will be printed the same number of elements in list intersect_final"))
for (term_to_numeric in intersect_final){
print(paste0("In this for loop, we make sure term scores are numeric. Term is: ", term, " and term_to_numeric is :", term_to_numeric))
DF_Identity_Pheno_Col[[term_to_numeric]] <- as.numeric(DF_Identity_Pheno_Col[[term_to_numeric]])
}

##_______________________________________________________##
# 2) Prepare Data for Statistics

library(ggpubr)
library(rstatix)

# 2.1) Data preparation

# Transform `dose` into factor variable
df <- DF_Identity_Pheno_Col

df$Identity <- as.factor(df$Identity)

# Add a random grouping variable
df$group <- factor(rep(c("grp1", "grp2"), 24))

# 4.2) Apply T-test

##______________________________________##
# 4.2.1) Apply T-test in Pheno (Hypoxia)

print(paste0("Now we start t-test for current term. Here, my_gene_input equals to term and term is: ", term))

#term <- "CHN1" # If this is left uncommented, only this term will be plotted
my_gene_input <- term
var_name <- 'my_gene_output'
assign(var_name, as.name(my_gene_input))
my_gene_output

# ADRN Pheno (Normoxia vs. Hypoxia)
t_pval_ADN_HStat <- t.test(df[[my_gene_output]][df$Cell_line==cell_line_input][df$Pheno=="N"], df[[my_gene_output]][df$Cell_line==cell_line_input][df$Pheno=="H"])$p.value

# Isolate Means
t_estimate_ADN_HStat <- t.test(df[[my_gene_output]][df$Cell_line==cell_line_input][df$Pheno=="N"], df[[my_gene_output]][df$Cell_line==cell_line_input][df$Pheno=="H"])$estimate

t_estimate_ADN_HStat_DF <- as.data.frame(t_estimate_ADN_HStat)

df_adrn <- df[df$Cell_line==cell_line_input]

print(paste0("At the end of this part, t-test DF was built to test the effect of hypoxia in the expression of term ", term))

##______________________________________##
# 4.2.2) Store in DF if significant

# Plot if the T-test means follow the desired trend for ADRN genes under Hypoxia.

print(paste0("## 1) In this part, we will plot graph if the T-test means follow the desired trend for ADRN genes under Hypoxia for term ", term))
print(paste0("#######################################################################################################################"))

print(paste0("The cell line here is: ", cell_line_input))
print(paste0("Define significant genes for cell line ", cell_line_input))
cell_line_sig_Genes <- paste0("cell_line_sig_Genes_", cell_line_input)

# my_list_input <- "A5"
# var_name <- 'my_list_output'
# assign(var_name, as.name(my_list_input))
# my_list_output

my_list_input <- cell_line_sig_Genes
var_name <- 'my_list_output'
assign(var_name, as.name(my_list_input))
print(paste0("In this part, we define the object of cell line, my_list_output: ", my_list_output))
print(paste0("The values received by ", my_list_output, " are supposed to store sig genes for cell_line_input ", cell_line_input))
my_list_output

##______________________________________##
# 4.2.3) ifelse to ensure identity of cell line

print(paste0("Now we start different types of ifelse statements. Term is: ", term))

print(paste0("In the first ifelse statement, my_gene_input still equals to term. Term is: ", term))

print(paste0("In this statement, we are imposing the expression of MES genes increased by hypoxia in ADRN cells (N compared to H). Term is: ", term))

print(paste0("## 2) Ifelse Statements Significant and increases in hypoxia ", term))
print(paste0("#######################################################################################################################"))

print(paste0("p-val for term ", term, " is: ", t_pval_ADN_HStat, "for cell line ", cell_line_input))

ifelse(
  (t_pval_ADN_HStat < 0.05) & 
    # Impose condition to print ADRN genes that are down-regulated by Hypoxia (ADRN cells combined)
    # The condition is based on Means of Normoxia and Hypoxia treatments
    (t_estimate_ADN_HStat_DF[1,1] > t_estimate_ADN_HStat_DF[2,1]),
  #print(paste0("Append my_gene_output to significant_adrnCell_up))#&
  significant_adrnCell_down <- c(significant_adrnCell_down, my_gene_output)
,
print(paste0("Because p-val is not significant for ", term, " we show that p-val is ", t_pval_ADN_HStat))
)
}, error=function(e){cat("ERROR :",conditionMessage(e), "\n")})
print(paste0("for loop for each term in intersect_final finishes here for current term: ",
             term, 
             " But I moved to after ifelse statements."))
} # End of for loop for terms or genes

print(paste0("For loop just finished for term ", term, 
             ". Now we will start another ifelse statement to store genes for each cell line. Current cell line is ", 
             cell_line_input))

print(paste0("For term ", 
             term, 
             " check if cell line is WN. Cell line is: ", 
             cell_line_input))

print(paste0("## 2.1) ifelse statement to check cell_line_input is WN for term ", term))
print(paste0("#######################################################"))

if (cell_line_input=="WN") {
  #if( length(significant_adrnCell_up)!=0 ){
    significant_wn_down_store <- significant_adrnCell_down
  #}
}

if (cell_line_input=="55N") {
  # if( length(significant_adrnCell_up)!=0 ){
    significant_55n_down_store <- significant_adrnCell_down
  # }
}

if (cell_line_input=="SY5Y") {
  # if( length(significant_adrnCell_up)!=0 ){
    significant_sy5y_down_store <- significant_adrnCell_down
  # }
}

if (cell_line_input=="BE2") {
  # if( length(significant_adrnCell_up)!=0 ){
    significant_be2_down_store <- significant_adrnCell_down
  # }
}

if (cell_line_input=="NBLS") {
  # if( length(significant_adrnCell_up)!=0 ){
    significant_nbls_down_store <- significant_adrnCell_down
  # }
}



}

##______________________________________##
# 5) Overlap and print genes in the final list

# Overlap
intersect_isogenic_down <- intersect(
  intersect(
    significant_wn_down_store,significant_55n_down_store),significant_sy5y_down_store)
intersect_be2_nbls_down <- intersect(significant_be2_down_store,significant_nbls_down_store)
intersect_allCellsDown <- intersect(intersect_isogenic_down, intersect_be2_nbls_down)

# Print final list
for (i in intersect_allCellsDown){
  print(i)
}

##______________________________________##
# 6) Plot Hypoxia on ADRN cells

adrn_down_intersectAll <- scan(text="PPEF1			
POU3F2			
ISL1			
HOXD3
FAM155B
RAB39A
OPRD1
RBP1			
OAS1
", what="")

  
ggplot(df_adrn_modified, aes(fill=Pheno, y=VIM, x=reorder(Cell_line, VIM))) + 
  geom_bar(position="dodge", stat="identity")+
  theme(axis.text.x = element_text(angle=45, hjust=1))+
  scale_fill_manual(values = c("#ff0c0c","#00008b"), labels = c("N","H")) +
  stat_compare_means(method = "t.test", label = "p.signif", label.y = c(3000, 6500, 6000, 6500, 7500 ))

ggplot(df_adrn_modified, aes(fill=Pheno, y=as.numeric(PHOX2B), x=reorder(Cell_line, as.numeric(PHOX2B)))) + 
  geom_bar(position="dodge", stat="identity")+
  theme(axis.text.x = element_text(angle=45, hjust=1))+
  scale_fill_manual(values = c("#ff0c0c","#00008b"), labels = c("N","H")) +
  stat_compare_means(method = "t.test", label = "p.signif", label.y = c(3000, 6500, 6000, 6500, 7500 ))

ggplot(df_adrn_modified, aes(fill=Pheno, y=as.numeric(PPEF1), x=reorder(Cell_line, as.numeric(PPEF1)))) + 
  geom_bar(position="dodge", stat="identity")+
  theme(axis.text.x = element_text(angle=45, hjust=1))+
  scale_fill_manual(values = c("#ff0c0c","#00008b"), labels = c("N","H")) +
  stat_compare_means(method = "t.test", label = "p.signif", label.y = c(3000, 6500, 6000, 6500, 7500 ))  

ggplot(df_adrn_modified, aes(fill=Pheno, y=as.numeric(POU3F2), x=reorder(Cell_line, as.numeric(POU3F2)))) + 
  geom_bar(position="dodge", stat="identity")+
  theme(axis.text.x = element_text(angle=45, hjust=1))+
  scale_fill_manual(values = c("#ff0c0c","#00008b"), labels = c("N","H")) +
  stat_compare_means(method = "t.test", label = "p.signif", label.y = c(3000, 6500, 6000, 6500, 7500 )) 

ggplot(df_adrn_modified, aes(fill=Pheno, y=as.numeric(ISL1), x=reorder(Cell_line, as.numeric(ISL1)))) + 
  geom_bar(position="dodge", stat="identity")+
  theme(axis.text.x = element_text(angle=45, hjust=1))+
  scale_fill_manual(values = c("#ff0c0c","#00008b"), labels = c("N","H")) +
  stat_compare_means(method = "t.test", label = "p.signif", label.y = c(2000, 2000, 4000, 4000, 8000 )) 
```


# 6.1.4)  Use our LFC (Search the Candidate MES Genes going Up in Hypoxia)

This chunk generates the vector of genes for each cell line to be intersected

```{r}
library(data.table)
test <- read.table("~/Desktop/GSVA_GeneExpression_Selected_Transposed.txt", 
                   sep = "\t",
                   header = T, 
                   row.names = 1,
                   check.names=FALSE)

cell_line_pheno_RNA_Seq_Realignment <- read.delim(
  "~/Desktop/Gepoliano/DE_RNA-Seq_Cells/DE_RNA-Seq_Cells_ALIGNMENT_HG38/pheno_CellLine_RNASeq_Realignment.txt")

DF_Identity_Pheno_Col <- test

# DF_Identity_Pheno_Col["Sample_ID"] <- rownames(test)

# intersect_final <- rownames(DF_select_ID_LFC_Neg)
# intersect_final <- first_list_interest
intersect_final <- c("SPARC", "VIM", "C1S", "COL5A2")

my_list_input <- "intersect_final"
var_name <- 'my_list_output'
assign(var_name, as.name(my_list_input))
my_list_output

# List of ADRN cells to look at
cell_line_list <- c("WN", "55N", "SY5Y", "BE2", "NBLS")
# cell_line_list <- c("WN")#  24
# cell_line_list <- c("55N")# 57
# cell_line_list <- c("SY5Y") # 45
# cell_line_list <- c("BE2") # 67
# cell_line_list <- c("NBLS") # 74


for (cell_line_input in cell_line_list) {
print(paste0("This starts for loop for cells in list. Cell line of list is: ", cell_line_input))
  term_count = 0
# Define name for the list of interest inside cell_line_input for loop.
# Otherwise the vector accumulates genes for following for loops.
significant_adrnCell_up <-c()
for (term in intersect_final){
  tryCatch({
term_count = term_count + 1
print(paste0("################################################################ For loop for term: ", term, " starts here"))
print(paste0("################################################################ In list cell_line_list, ", term," is element number: ", term_count))
print(paste0("#################################################################################################################"))

library(ggplot2)

##_______________________________________________________##
# 1) Join DFs and ensure as.data.frame, but w/ all terms
print(paste0("Now we will join Pheno information DF and Gene Expression DFs. Term in intersect_final is: ", term))

dt1 <- data.table(cell_line_pheno_RNA_Seq_Realignment, key = "FILE_NAME") 
dt2 <- data.table(DF_Identity_Pheno_Col, key = "Sample_ID")
joined.dt1.dt.2 <- dt1[dt2]
DF_Identity_Pheno_Col <- dt1[dt2]

# Make tarms of interest, numeric
# If use term here, will make term store the wrong value
print(paste0("In this for loop, the print statement will be printed the same number of elements in list intersect_final"))
for (term_to_numeric in intersect_final){
print(paste0("In this for loop, we make sure term scores are numeric. Term is: ", term, " and term_to_numeric is :", term_to_numeric))
DF_Identity_Pheno_Col[[term_to_numeric]] <- as.numeric(DF_Identity_Pheno_Col[[term_to_numeric]])
}

##_______________________________________________________##
# 2) Prepare Data for Statistics

library(ggpubr)
library(rstatix)

# 2.1) Data preparation

# Transform `dose` into factor variable
df <- DF_Identity_Pheno_Col

df$Identity <- as.factor(df$Identity)

# Add a random grouping variable
df$group <- factor(rep(c("grp1", "grp2"), 24))

# 4.2) Apply T-test

##______________________________________##
# 4.2.1) Apply T-test in Pheno (Hypoxia)

print(paste0("Now we start t-test for current term. Here, my_gene_input equals to term and term is: ", term))

#term <- "CHN1" # If this is left uncommented, only this term will be plotted
my_gene_input <- term
var_name <- 'my_gene_output'
assign(var_name, as.name(my_gene_input))
my_gene_output

# ADRN Pheno (Normoxia vs. Hypoxia)
t_pval_ADN_HStat <- t.test(df[[my_gene_output]][df$Cell_line==cell_line_input][df$Pheno=="N"], df[[my_gene_output]][df$Cell_line==cell_line_input][df$Pheno=="H"])$p.value

# Isolate Means
t_estimate_ADN_HStat <- t.test(df[[my_gene_output]][df$Cell_line==cell_line_input][df$Pheno=="N"], df[[my_gene_output]][df$Cell_line==cell_line_input][df$Pheno=="H"])$estimate

t_estimate_ADN_HStat_DF <- as.data.frame(t_estimate_ADN_HStat)

df_adrn <- df[df$Cell_line==cell_line_input]

print(paste0("At the end of this part, t-test DF was built to test the effect of hypoxia in the expression of term ", term))

##______________________________________##
# 4.2.2) Store in DF if significant

# Plot if the T-test means follow the desired trend for ADRN genes under Hypoxia.

print(paste0("## 1) In this part, we will plot graph if the T-test means follow the desired trend for ADRN genes under Hypoxia for term ", term))
print(paste0("#######################################################################################################################"))

print(paste0("The cell line here is: ", cell_line_input))
print(paste0("Define significant genes for cell line ", cell_line_input))
cell_line_sig_Genes <- paste0("cell_line_sig_Genes_", cell_line_input)

# my_list_input <- "A5"
# var_name <- 'my_list_output'
# assign(var_name, as.name(my_list_input))
# my_list_output

my_list_input <- cell_line_sig_Genes
var_name <- 'my_list_output'
assign(var_name, as.name(my_list_input))
print(paste0("In this part, we define the object of cell line, my_list_output: ", my_list_output))
print(paste0("The values received by ", my_list_output, " are supposed to store sig genes for cell_line_input ", cell_line_input))
my_list_output

##______________________________________##
# 4.2.3) ifelse to ensure identity of cell line

print(paste0("Now we start different types of ifelse statements. Term is: ", term))

print(paste0("In the first ifelse statement, my_gene_input still equals to term. Term is: ", term))

print(paste0("In this statement, we are imposing the expression of MES genes increased by hypoxia in ADRN cells (N compared to H). Term is: ", term))

print(paste0("## 2) Ifelse Statements Significant and increases in hypoxia ", term))
print(paste0("#######################################################################################################################"))

print(paste0("p-val for term ", term, " is: ", t_pval_ADN_HStat, "for cell line ", cell_line_input))

ifelse(
  (t_pval_ADN_HStat < 0.05) & 
    # Impose condition to print ADRN genes that are down-regulated by Hypoxia (ADRN cells combined)
    # The condition is based on Means of Normoxia and Hypoxia treatments
    (t_estimate_ADN_HStat_DF[1,1] < t_estimate_ADN_HStat_DF[2,1]),
  #print(paste0("Append my_gene_output to significant_adrnCell_up))#&
  significant_adrnCell_up <- c(significant_adrnCell_up, my_gene_output)
,
print(paste0("Because p-val is not significant for ", term, " we show that p-val is ", t_pval_ADN_HStat))
)
}, error=function(e){cat("ERROR :",conditionMessage(e), "\n")})
print(paste0("for loop for each term in intersect_final finishes here for current term: ",
             term, 
             " But I moved to after ifelse statements."))
} # End of for loop for terms or genes

print(paste0("For loop just finished for term ", term, 
             ". Now we will start another ifelse statement to store genes for each cell line. Current cell line is ", 
             cell_line_input))

print(paste0("For term ", 
             term, 
             " check if cell line is WN. Cell line is: ", 
             cell_line_input))

print(paste0("## 2.1) ifelse statement to check cell_line_input is WN for term ", term))
print(paste0("#######################################################"))

if (cell_line_input=="WN") {
  if( length(significant_adrnCell_up)!=0 ){
    significant_wn_up_store <- significant_adrnCell_up
  }
}

if (cell_line_input=="55N") {
  if( length(significant_adrnCell_up)!=0 ){
    significant_55n_up_store <- significant_adrnCell_up
  }
}

if (cell_line_input=="SY5Y") {
  if( length(significant_adrnCell_up)!=0 ){
    significant_sy5y_up_store <- significant_adrnCell_up
  }
}

if (cell_line_input=="BE2") {
  if( length(significant_adrnCell_up)!=0 ){
    significant_be2_up_store <- significant_adrnCell_up
  }
}

if (cell_line_input=="NBLS") {
  if( length(significant_adrnCell_up)!=0 ){
    significant_nbls_up_store <- significant_adrnCell_up
  }
}



}

##______________________________________##
# 5) Overlap and print genes in the final list

# Overlap
intersect_isogenic_up <- intersect(
  intersect(
    significant_wn_up_store,significant_55n_up_store),significant_sy5y_up_store)
intersect_be2_nbls_up <- intersect(significant_be2_up_store,significant_nbls_up_store)
intersect_allCellsUp <- intersect(intersect_isogenic_up, intersect_be2_nbls_up)

# Print final list
for (i in intersect_allCellsUp){
  print(i)
}
```


# 6.1.5) Use our LFC (Search the Candidate ADRN Genes going down in Hypoxia)

This chunk generates the vector of genes for each cell line to be intersected

```{r}

intersect_final <- rownames(DF_select_ID_LFC_Pos)

my_list_input <- "intersect_final"
var_name <- 'my_list_output'
assign(var_name, as.name(my_list_input))
my_list_output

# List of ADRN cells to look at
cell_line_list <- c("WN", "55N", "SY5Y", "BE2", "NBLS")
# cell_line_list <- c("WN")
# cell_line_list <- c("55N")
# cell_line_list <- c("SY5Y")
# cell_line_list <- c("BE2")
cell_line_list <- c("NBLS")

# Define name for the list of interest
significant_adrnCell_down <-c()

for (cell_line_input in cell_line_list) {

for (term in intersect_final){
  tryCatch({
print(paste0("Before make numeric, starting with term,  ", term))
library(ggplot2)

# Use DF that was not filtered by terms
DF_Identity_Pheno_Col <- GSVA_GeneExpression_Selected_Transposed

##_______________________________________________________##
# 1) Join DFs and ensure as.data.frame, but w/ all terms

DF_Identity_Pheno_Col["Sample_ID"] <- rownames(DF_Identity_Pheno_Col)

dt1 <- data.table(cell_line_pheno_RNA_Seq_Realignment, key = "FILE_NAME") 
dt2 <- data.table(DF_Identity_Pheno_Col, key = "Sample_ID")
joined.dt1.dt.2 <- dt1[dt2]
DF_Identity_Pheno_Col <- dt1[dt2]

# Make tarms of interest, numeric
# If use term here, will make term store the wrong value
for (term_to_numeric in intersect_final){
DF_Identity_Pheno_Col[[term_to_numeric]] <- as.numeric(DF_Identity_Pheno_Col[[term_to_numeric]])
}

##_______________________________________________________##
# 2) Prepare Data for Statistics

library(ggpubr)
library(rstatix)

# 2.1) Data preparation

# Transform `dose` into factor variable
df <- DF_Identity_Pheno_Col

df$Identity <- as.factor(df$Identity)

# Add a random grouping variable
df$group <- factor(rep(c("grp1", "grp2"), 24))

# 4.2) Apply T-test

##______________________________________##
# 4.2.1) Apply T-test in Pheno (Hypoxia)

#term <- "CHN1" # If this is left uncommented, only this term will be plotted
my_gene_input <- term
var_name <- 'my_gene_output'
assign(var_name, as.name(my_gene_input))
my_gene_output

# ADRN Pheno (Normoxia vs. Hypoxia)
t_pval_ADN_HStat <- t.test(df[[my_gene_output]][df$Cell_line==cell_line_list][df$Pheno=="N"], df[[my_gene_output]][df$Cell_line==cell_line_list][df$Pheno=="H"])$p.value

# Isolate Means
t_estimate_ADN_HStat <- t.test(df[[my_gene_output]][df$Cell_line==cell_line_list][df$Pheno=="N"], df[[my_gene_output]][df$Cell_line==cell_line_list][df$Pheno=="H"])$estimate

t_estimate_ADN_HStat_DF <- as.data.frame(t_estimate_ADN_HStat)

df_adrn <- df[df$Cell_line==cell_line_input]

##______________________________________##
# 4.2.2) Store in DF if significant

# Plot if the T-test means follow the desired trend for ADRN genes under Hypoxia.

print(paste0("term is ", term))

cell_line_input # each cell line in the vector-declaring ADRN cells
cell_line_sig_Genes <- paste0("cell_line_sig_Genes_", cell_line_input)

# my_list_input <- "A5"
# var_name <- 'my_list_output'
# assign(var_name, as.name(my_list_input))
# my_list_output

my_list_input <- cell_line_sig_Genes
var_name <- 'my_list_output'
assign(var_name, as.name(my_list_input))
my_list_output

ifelse(
  (t_pval_ADN_HStat < 0.05) & 
    # Impose condition to print ADRN genes that are down-regulated by Hypoxia (ADRN cells combined)
    # The condition is based on Means of Normoxia and Hypoxia treatments
    (t_estimate_ADN_HStat_DF[1,1] > t_estimate_ADN_HStat_DF[2,1]),
  # Impose creation of list with ADRN genes that go down by Hypoxia
  significant_adrnCell_down <- c(significant_adrnCell_down, my_gene_output)
,
print(paste0(term, " p-val is ", t_pval_ADN_HStat))
)
}, error=function(e){cat("ERROR :",conditionMessage(e), "\n")})
}
}

#significant_wn_down <- significant_adrnCell_down
significant_wn_down_store <- c("PDE4B","PLPPR5","NGF","DPT","RD3","PLD5","KCNK3",
                               "GALNT14","SCN1A","ZNF385D","TAGLN3","GAP43","SPTSSB",
                               "BCHE","SHISA3","UGT8","NRSN1","SCGN","LRFN2","TTBK1",
                               "TFAP2B","FUT9","VIP","ACTL6B","PRRT4","PI15","PAX5",
                               "RASEF","ROR2","BRINP1","ADARB2","DRGX","PHYHIPL","ANO5",
                               "C11orf96","FAT3","PTPRO","BTBD11","RPH3A","CCNA1","SERTM1",
                               "SLITRK1","DHRS2","INSM2","CCDC177","KCNK10","PRIMA1","SLC27A2",
                               "HS3ST2","KCNH4","TMEM200C","ZNF536","PPP1R16B","RIMS4","PPEF1")

# significant_55n_down <- significant_adrnCell_down
significant_55n_down_store <- c("GRIK3","RGS16","ESRRG","KCNK3","RUFY4","TAGLN3","GAP43",
                                "CLSTN2","PARM1","POU4F2","RAB3C","GFRA3","CPLX2","NRSN1",
                                "TFAP2B","ACTL6B","PRRT4","ROR2","TMEM273","PHYHIPL","CHST15",
                                "INSC","DRD2","HTR3A","SCN3B","CNTN1","ASCL1","FOXN4","TUBA3C",
                                "CARMIL3","INSM2","PELI2","PRIMA1","TUNAR","SLC27A2","KCNH4",
                                "UNC13A","CHGB","BMP7","HRH3","PPEF1","TSPAN7")

# significant_sy5y_down <- significant_adrnCell_down
significant_sy5y_down_store <- c("IGSF21","NT5C1A","ELAVL4","PLPPR5","KCNA3","NGF","TNR",
                                 "FAM163A","RD3","ESRRG","PLD5","MYT1L","ALK","GALNT14",
                                 "CNTNAP5","CDK5R2","KIF1A","ATP2B2","ZNF385D","CACNA1D",
                                 "TAGLN3","GAP43","TMEM108","CLSTN2","B3GALNT1","SPTSSB",
                                 "SOX2","C4orf50","PPP2R2C","KIT","NEUROG2","UGT8","MARCHF11",
                                 "NRSN1","GLP1R","LRFN2","TFAP2D","TFAP2B","SNAP91","FUT9",
                                 "SLC35D3","IL20RA","ICA1","NXPH1","NPY","ACTL6B","PRRT4",
                                 "STMN4","CA8","EYA1","PI15","MTSS1","FAM135B","TMEM215",
                                 "PAX5","IGFBPL1","RASEF","DIRAS2","SYK","ROR2","ADARB2",
                                 "AKR1E2","VSTM4","NRG3","CDHR1","PRLHR","INSC","NELL1",
                                 "ANO5","RAG2","TSPAN18","SYT13","FAT3","GRIA4","DRD2",
                                 "SCN3B","PTPRO","BTBD11","WSCD2","SVOP","RPH3A","EFS",
                                 "DHRS2","TUNAR","GABRG3","SLC27A2","HS3ST2","KCNH6","TMEM200C",
                                 "L3MBTL4","CELF4","MAPK4","UNC13A","NCAN","AC010615.4","ZNF536",
                                 "LAMP5","OTOR","CCM2L","PPP1R16B","SALL4","BMP7","PHACTR3",
                                 "HRH3","ABCG1","NEFH","CACNG2","PPEF1","ZFP92")

# significant_be2_down <- significant_adrnCell_down
significant_be2_down_store <- c("NT5C1A","RIMS3","ELAVL4","PDE4B","ERICH3","PLPPR5",
                                 "PLPPR4","DPT","FAM163A","NFASC","RD3","PROX1","ESRRG",
                                 "MAP10","PLD5","KCNK3","ALK","GALNT14","CNTNAP5","SCN1A",
                                 "CRYGD","UNC80","CDK5R2","ATP2B2","ZNF385D","TAGLN3",
                                 "TMEM108","CLSTN2","B3GALNT1","SPTSSB","BCHE","SOX2",
                                 "ZNF718","PPP2R2C","SHISA3","PABPC4L","POU4F2","RNF175",
                                 "ENPP6","CTNND2","MARCHF11","RAB3C","LIX1","FSTL4","GFRA3",
                                 "HAND1","CPLX2","HK3","GRM6","NHLRC1","NRSN1","SCGN","BTN1A1","
                                 LRFN2","TFAP2D","TFAP2B","PKHD1","SNAP91","FUT9","CLVS2",
                                 "IL20RA","VIP","OPRM1","NXPH1","NPY","VSTM2A","RELN","PRRT4",
                                 "GIMAP4","EYA1","PI15","ENPP2","COL22A1","TMEM215","PAX5",
                                 "IGFBPL1","RASEF","ROR2","ADARB2","VSTM4","TMEM273","DRGX","GFRA1",
                                 "GPR26","CHST15","HTATIP2","ANO5","MPPED2","RAG2","C11orf96","TSPAN18",
                                 "SYT13","APLNR","FAT3","DRD2","SCN3B","PTPRO","ASCL1","BTBD11","WSCD2",
                                 "FOXN4","RPH3A","NOS1","SERTM1","SIAH3","SLITRK1","OXGR1","INSM2",
                                 "PELI2","KCNK10","PRIMA1","GABRG3","GJD2","HS3ST2","IRX6","SLC6A2",
                                 "ACE","TMEM200C","L3MBTL4","AC010615.4","TEX101","LAMP5","CD93",
                                 "SLC32A1","PPP1R16B","BMP7","EDN3","HRH3","SAMSN1","ABCG1",
                                 "NEFH","RIBC2","ANOS1","PPEF1","CAPN6")

significant_nbls_down <- significant_adrnCell_down
significant_nbls_down_store <- c("EPHA8","RIMS3","ELAVL4","PTGER3","PLPPR5","PLPPR4",
                                 "OLFM3","NTNG1","SLC6A17","KCNA3","TNR","FAM163A","RGS16",
                                 "NFASC","CNTN2","RD3","ESRRG","PLD5","MYT1L","KCNK3","GALNT14",
                                 "ADD2","CNTNAP5","SCN3A","CRYGD","CDK5R2","ATP2B2","VILL",
                                 "CACNA2D2","CACNA1D","EPHA6","TAGLN3","GAP43","TMEM108","CLSTN2",
                                 "B3GALNT1","SPTSSB","SOX2","ZNF718","C4orf50","PPP2R2C","PROM1",
                                 "SHISA3","KIT","NEUROG2","C7","RAB3C","LIX1","SPINK5","CPLX2",
                                 "AC139491.7","HK3","NRSN1","BTN1A1","GLP1R","LRFN2","TFAP2D",
                                 "TFAP2B","PKHD1","SNAP91","FUT9","CLVS2","SLC35D3","VIP","OPRM1",
                                 "NXPH1","NPY","ACTL6B","RELN","KCND2","PRRT4","STMN4","CA8","EYA1",
                                 "KCNB2","PI15","FAM135B","PAX5","IGFBPL1","RASEF","DIRAS2","SYK",
                                 "ROR2","ADARB2","AKR1E2","CXCL12","GDF10","CHAT","PHYHIPL","GFRA1",
                                 "CHST15","HTATIP2","MPPED2","TSPAN18","APLNR","P2RX3","MAP6","FAT3",
                                 "GRIA4","DRD2","SCN3B","PTPRO","ASCL1","WSCD2","SVOP","FOXN4","RPH3A",
                                 "NOS1","SRRM4","TUBA3C","SERTM1","PCDH17","DHRS2","CARMIL3","INSM2",
                                 "PELI2","CCDC177","SLC8A3","KCNK10","UNC79","TUNAR","BEGAIN","GABRG3",
                                 "SLC27A2","HS3ST2","BCL6B","KCNH4","CA10","ACE","AKAIN1","L3MBTL4",
                                 "ASXL3","CELF4","SYT4","GRP","UNC13A","NCAN","ZNF208","ZNF536","TEX101",
                                 "LAMP5","PAK5","OTOR","SLC32A1","PPP1R16B","SALL4","BMP7","HRH3",
                                 "SAMSN1","ABCG1","SEZ6L","NEFH","CACNG2","RIBC2","PPEF1","TSPAN7",
                                 "MAOB","DGKK","CHRDL1","CAPN6","DCX","SERTM2","IL13RA2","ARHGAP36","TSPY4")

##______________________________________##
# 5) Overlap and print genes in the final list

# Overlap
intersect_isogenic_down <- intersect(
  intersect(
    significant_wn_down_store,significant_55n_down_store),significant_sy5y_down_store)
intersect_be2_nbls_down <- intersect(significant_be2_down_store,significant_nbls_down_store)
intersect_allCellsDown <- intersect(intersect_isogenic_down, intersect_be2_nbls_down)

# Print final list
for (i in intersect_allCellsDown){
  print(i)
}
```


# 6.2) Plot Genes Down-regulated in All ADRN

This chunk plot the genes of interest

```{r}
# intersect_final <- c("SDC2", "SLC5A6", "UBE2T")
# 
# intersect_final <- scan(text="AHSA1
# ASRGL1
# ATP6V0E2
# CSE1L
# FKBP4
# IGSF3
# ISL1
# LSM4
# NET1
# NRSN1
# RBP1
# TAGLN3
# TFAP2B
# UBE2T
# ", what="")



my_list_input <- "intersect_final"
var_name <- 'my_list_output'
assign(var_name, as.name(my_list_input))
my_list_output

for (term in intersect_final){
print(paste0("Before make numeric, starting with term,  ", term))
library(ggplot2)

# Use DF that was not filtered by terms
DF_Identity_Pheno_Col <- GSVA_GeneExpression_Selected_Transposed

##_______________________________________________________##
# 1) Join DFs and ensure as.data.frame, but w/ all terms

DF_Identity_Pheno_Col["Sample_ID"] <- rownames(DF_Identity_Pheno_Col)

dt1 <- data.table(cell_line_pheno_RNA_Seq_Realignment, key = "FILE_NAME") 
dt2 <- data.table(DF_Identity_Pheno_Col, key = "Sample_ID")
joined.dt1.dt.2 <- dt1[dt2]
DF_Identity_Pheno_Col <- dt1[dt2]

# Make tarms of interest, numeric
# If use term here, will make term store the wrong value
for (term_to_numeric in intersect_final){
DF_Identity_Pheno_Col[[term_to_numeric]] <- as.numeric(DF_Identity_Pheno_Col[[term_to_numeric]])
}

##_______________________________________________________##
# 2) Prepare Data for Statistics

library(ggpubr)
library(rstatix)

# 2.1) Data preparation

# Transform `dose` into factor variable
df <- DF_Identity_Pheno_Col

df$Identity <- as.factor(df$Identity)

# Add a random grouping variable
df$group <- factor(rep(c("grp1", "grp2"), 24))

# 4.2) Apply T-test

##______________________________________##
# 4.2.1) Apply T-test in Pheno (Hypoxia)

#term <- "CHN1" # If this is left uncommented, only this term will be plotted
my_gene_input <- term
var_name <- 'my_gene_output'
assign(var_name, as.name(my_gene_input))
my_gene_output

# ADRN Pheno (Normoxia vs. Hypoxia)
t_pval_ADN_HStat <- t.test(df[[my_gene_output]][df$Identity=="ADRN"][df$Pheno=="N"], df[[my_gene_output]][df$Identity=="ADRN"][df$Pheno=="H"])$p.value

# Isolate Means
t_estimate_ADN_HStat <- t.test(df[[my_gene_output]][df$Identity=="ADRN"][df$Pheno=="N"], df[[my_gene_output]][df$Identity=="ADRN"][df$Pheno=="H"])$estimate

t_estimate_ADN_HStat_DF <- as.data.frame(t_estimate_ADN_HStat)

df_adrn <- df[df$Identity=="ADRN"]

##______________________________________##
# 4.2.2) Plot if significant

# Plot if the T-test means follow the desired trend for ADRN genes under Hypoxia.

print(paste0("term is ", term))

ifelse(
  (t_pval_ADN_HStat < 0.05) & 
    # Impose condition to print ADRN genes that are down-regulated by Hypoxia (ADRN cells combined)
    # The condition is based on Means of Normoxia and Hypoxia treatments
    (t_estimate_ADN_HStat_DF[1,1] > t_estimate_ADN_HStat_DF[2,1]),
print(ggboxplot(df_adrn, x = "Pheno", y = term,
          color = "Cell_line", palette = "jco",
          add = "jitter") + stat_compare_means(method = "t.test")),
print(paste0(term, " p-val is ", t_pval_ADN_HStat))
)
}
```

# 6.3) Plot Genes Down-regulated in All ADRN

```{r}
intersect_final <- c("SDC2", "SLC5A6", "UBE2T")

intersect_final <- scan(text="AHSA1
ASRGL1
ATP6V0E2
CSE1L
FKBP4
IGSF3
ISL1
LSM4
NET1
NRSN1
RBP1
TAGLN3
TFAP2B
UBE2T
", what="")



my_list_input <- "intersect_final"
var_name <- 'my_list_output'
assign(var_name, as.name(my_list_input))
my_list_output

for (term in intersect_final){
print(paste0("Before make numeric, starting with term,  ", term))
library(ggplot2)

# Use DF that was not filtered by terms
DF_Identity_Pheno_Col <- GSVA_GeneExpression_Selected_Transposed

##_______________________________________________________##
# 1) Join DFs and ensure as.data.frame, but w/ all terms

DF_Identity_Pheno_Col["Sample_ID"] <- rownames(DF_Identity_Pheno_Col)

dt1 <- data.table(cell_line_pheno_RNA_Seq_Realignment, key = "FILE_NAME") 
dt2 <- data.table(DF_Identity_Pheno_Col, key = "Sample_ID")
joined.dt1.dt.2 <- dt1[dt2]
DF_Identity_Pheno_Col <- dt1[dt2]

# Make tarms of interest, numeric
# If use term here, will make term store the wrong value
for (term_to_numeric in intersect_final){
DF_Identity_Pheno_Col[[term_to_numeric]] <- as.numeric(DF_Identity_Pheno_Col[[term_to_numeric]])
}

##_______________________________________________________##
# 2) Prepare Data for Statistics

library(ggpubr)
library(rstatix)

# 2.1) Data preparation

# Transform `dose` into factor variable
df <- DF_Identity_Pheno_Col

df$Identity <- as.factor(df$Identity)

# Add a random grouping variable
df$group <- factor(rep(c("grp1", "grp2"), 24))

# 4.2) Apply T-test

##______________________________________##
# 4.2.1) Apply T-test in Pheno (Hypoxia)

#term <- "CHN1" # If this is left uncommented, only this term will be plotted
my_gene_input <- term
var_name <- 'my_gene_output'
assign(var_name, as.name(my_gene_input))
my_gene_output

# ADRN Pheno (Normoxia vs. Hypoxia)
t_pval_ADN_HStat <- t.test(df[[my_gene_output]][df$Identity=="ADRN"][df$Pheno=="N"], df[[my_gene_output]][df$Identity=="ADRN"][df$Pheno=="H"])$p.value

# Isolate Means
t_estimate_ADN_HStat <- t.test(df[[my_gene_output]][df$Identity=="ADRN"][df$Pheno=="N"], df[[my_gene_output]][df$Identity=="ADRN"][df$Pheno=="H"])$estimate

t_estimate_ADN_HStat_DF <- as.data.frame(t_estimate_ADN_HStat)

df_adrn <- df[df$Identity=="ADRN"]

##______________________________________##
# 4.2.2) Plot if significant

# Plot if the T-test means follow the desired trend for ADRN genes under Hypoxia.

print(paste0("term is ", term))

ifelse(
  (t_pval_ADN_HStat < 0.05) & 
    # Impose condition to print ADRN genes that are down-regulated by Hypoxia (ADRN cells combined)
    # The condition is based on Means of Normoxia and Hypoxia treatments
    (t_estimate_ADN_HStat_DF[1,1] > t_estimate_ADN_HStat_DF[2,1]),
print(ggboxplot(df_adrn, x = "Pheno", y = term,
          color = "Cell_line", palette = "jco",
          add = "jitter") + stat_compare_means(method = "t.test")),
print(paste0(term, " p-val is ", t_pval_ADN_HStat))
)
}

ggplot(df_adrn, aes(x = reorder(Cell_line, PHOX2B), y = PHOX2B)) + geom_bar(stat = "identity")

df_adrn_modified <- df_adrn
df_adrn_modified$Pheno <- factor(df_adrn_modified$Pheno, levels = rev(unique(df_adrn_modified$Pheno)), ordered=TRUE)
df_adrn_modified$Cell_line <- factor(df_adrn_modified$Cell_line, levels = rev(unique(df_adrn_modified$Cell_line)), ordered=TRUE)

ggbarplot(df_adrn_modified, x = "Pheno", y = "PHOX2B", fill = "Pheno",
          sort.by.groups = FALSE
          )

library(ggpubr)
df <- df_adrn_modified
df$Pheno <- as.factor(df$Pheno)
ggbarplot(df, x = "Pheno", y = "PHOX2B",
                fill = "Cell_line",
                add = "mean_sd", add.params = list(group = "Cell_line"),
                position = position_dodge(0.8)) 

ggplot(df_adrn_modified, aes(fill=Pheno, y=PHOX2B, x=Cell_line)) + 
    geom_bar(position="dodge", stat="identity")+
  theme(axis.text.x = element_text(angle=45, hjust=1))

ggplot(df_adrn_modified, aes(fill=Pheno, y=SPARC, x=reorder(Cell_line, SPARC))) + 
    geom_bar(position="dodge", stat="identity")+
  theme(axis.text.x = element_text(angle=45, hjust=1))+
  scale_fill_manual(values = c("#00008b","#ff0c0c"), labels = c("H","N"))

# Change order manually Pheno
df_adrn_modified$Pheno <- factor(df_adrn_modified$Pheno,          
                  levels = c("N",  
                             "H"))
ggplot(df_adrn_modified, aes(fill=Pheno, y=SPARC, x=reorder(Cell_line, SPARC))) + 
    geom_bar(position="dodge", stat="identity")+
  theme(axis.text.x = element_text(angle=45, hjust=1))+
  scale_fill_manual(values = c("#ff0c0c","#00008b"), labels = c("N","H"))

dat <- data.frame(Group = c("S1", "S1", "S2", "S2"),
              Sub   = c("A", "B", "A", "B"),
              Value = c(3,5,7,8))  

ggplot(dat, aes(Group, Value)) +
  geom_bar(aes(fill = Sub), stat="identity", position="dodge", width=.5) +
  geom_signif(stat="identity",
              data=data.frame(x=c(0.875, 1.875), xend=c(1.125, 2.125),
                              y=c(5.8, 8.5), annotation=c("**", "NS")),
              aes(x=x,xend=xend, y=y, yend=y, annotation=annotation)) +
  geom_signif(comparisons=list(c("S1", "S2")), annotations="***",
              y_position = 9.3, tip_length = 0, vjust=0.4) +
  scale_fill_manual(values = c("grey80", "grey20"))
```

# 6.4) Plot Gene of Interest

This chunk also plots the DFs with Candidate Genes of Interest

```{r}
intersect_final <- c("VIM")

my_list_input <- "intersect_final"
var_name <- 'my_list_output'
assign(var_name, as.name(my_list_input))
my_list_output

for (term in intersect_final){
print(paste0("Before make numeric, starting with term,  ", term))
library(ggplot2)

# Use DF that was not filtered by terms
DF_dds.RNA_General_ADRN_vs_MES <- as.data.frame(assay(dds.RNA_General_ADRN_vs_MES))
DF_dds.RNA_General_ADRN_vs_MES <- t(DF_dds.RNA_General_ADRN_vs_MES)
Sample_ID = rownames(DF_dds.RNA_General_ADRN_vs_MES)
df <- cbind(Sample_ID, DF_dds.RNA_General_ADRN_vs_MES)

##_______________________________________________________##
# 1) Join DFs and ensure as.data.frame, but w/ all terms

dt1 <- data.table(cell_line_pheno_RNA_Seq_Realignment, key = "FILE_NAME") 
dt2 <- data.table(df, key = "Sample_ID")
joined.dt1.dt.2 <- dt1[dt2]
DF_Identity_Pheno_Col <- dt1[dt2]

# Make tarms of interest, numeric
# If use term here, will make term store the wrong value
for (term_to_numeric in intersect_final){
DF_Identity_Pheno_Col[[term_to_numeric]] <- as.numeric(DF_Identity_Pheno_Col[[term_to_numeric]])
}

##_______________________________________________________##
# 2) Prepare Data for Statistics

library(ggpubr)
library(rstatix)

# 2.1) Data preparation

# Transform `dose` into factor variable
df <- DF_Identity_Pheno_Col

df$Identity <- as.factor(df$Identity)

# Add a random grouping variable
df$group <- factor(rep(c("grp1", "grp2"), 24))

# 4.2) Apply T-test

##______________________________________##
# 4.2.1) Apply T-test in Pheno (Hypoxia)

#term <- "CHN1" # If this is left uncommented, only this term will be plotted
my_gene_input <- term
var_name <- 'my_gene_output'
assign(var_name, as.name(my_gene_input))
my_gene_output

# ADRN Pheno (Normoxia vs. Hypoxia)
t_pval_ADN_HStat <- t.test(df[[my_gene_output]][df$Identity=="ADRN"][df$Pheno=="N"], df[[my_gene_output]][df$Identity=="ADRN"][df$Pheno=="H"])$p.value

# Isolate Means
t_estimate_ADN_HStat <- t.test(df[[my_gene_output]][df$Identity=="ADRN"][df$Pheno=="N"], df[[my_gene_output]][df$Identity=="ADRN"][df$Pheno=="H"])$estimate

t_estimate_ADN_HStat_DF <- as.data.frame(t_estimate_ADN_HStat)

df_adrn <- df[df$Identity=="ADRN"]

##______________________________________##
# 4.2.2) Plot if significant

# Plot if the T-test means follow the desired trend for ADRN genes under Hypoxia.

print(paste0("term is ", term))

ifelse(
  (t_pval_ADN_HStat < 0.05) & 
    # Impose condition to print ADRN genes that are down-regulated by Hypoxia (ADRN cells combined)
    # The condition is based on Means of Normoxia and Hypoxia treatments
    (t_estimate_ADN_HStat_DF[1,1] < t_estimate_ADN_HStat_DF[2,1]),# Condition ADRN_H > ADRN_N
print(ggboxplot(df_adrn, x = "Pheno", y = term,
          color = "Cell_line", palette = "jco",
          add = "jitter") + stat_compare_means(method = "t.test")),
print(paste0(term, " p-val is ", t_pval_ADN_HStat))
)
}

ggplot(df_adrn, aes(x = reorder(Cell_line, PHOX2B), y = PHOX2B)) + geom_bar(stat = "identity")

df_adrn_modified <- df_adrn
df_adrn_modified$Pheno <- factor(df_adrn_modified$Pheno, levels = rev(unique(df_adrn_modified$Pheno)), ordered=TRUE)
df_adrn_modified$Cell_line <- factor(df_adrn_modified$Cell_line, levels = rev(unique(df_adrn_modified$Cell_line)), ordered=TRUE)

ggbarplot(df_adrn_modified, x = "Pheno", y = intersect_final, fill = "Pheno",
          sort.by.groups = FALSE
          )

##______________________________________##
# 4.2.3) ggbarplot Graph

library(ggpubr)
df <- df_adrn_modified

# Change order manually Pheno
df_adrn_modified$Pheno <- factor(df_adrn_modified$Pheno,          
                  levels = c("N",  
                             "H"))
ggbarplot(df_adrn_modified, x = "Pheno", y = intersect_final,
                fill = "Cell_line",
                add = "mean_sd", add.params = list(group = "Cell_line"),
                position = position_dodge(0.8)) 

ggplot(df_adrn_modified, aes(fill=Pheno, y=PHOX2B, x=Cell_line)) + 
    geom_bar(position="dodge", stat="identity")+
  theme(axis.text.x = element_text(angle=45, hjust=1))

ggplot(df_adrn_modified, aes(fill=Pheno, y=VIM, x=reorder(Cell_line, VIM))) + 
    geom_bar(position="dodge", stat="identity")+
  theme(axis.text.x = element_text(angle=45, hjust=1))+
  scale_fill_manual(values = c("#00008b","#ff0c0c"), labels = c("H","N"))

##______________________________________##
# 5) Select Genes to Print Hypoxia/ADRN

# Change order manually Pheno
df_adrn_modified$Pheno <- factor(df_adrn_modified$Pheno,          
                  levels = c("N",  
                             "H"))
ggplot(df_adrn_modified, aes(fill=Pheno, y=VIM, x=reorder(Cell_line, c(VIM)))) + 
    geom_bar(position="dodge", stat="identity")+
  theme(axis.text.x = element_text(angle=45, hjust=1))+
  scale_fill_manual(values = c("#ff0c0c","#00008b"), labels = c("N","H")) +
  stat_compare_means(method = "t.test", label = "p.signif", label.y = c(3000, 6500, 6000, 6500, 7500 ))

DF2 <- df_adrn_modified %>% dplyr::select("FILE_NAME", "Pheno", "Cell_line", "Identity",
                                        "SPARC", "VIM", "PRRX1",
                                        # Impose MES Hypoxia Up
                                        "COL5A2","C1S",
                                        # Impose MES Hypoxia Up
                                        "COL27A1", "SPARC","SPARC","ADM","TAGLN","SLC16A3","ALDH3B1","SEMA3B",
                                        "STING1","MYLK3","VWF","STC1","BHLHE40","GLIPR1","PPP1R3G","C1RL","ORAI3",	
                                        "PHOX2B", "DACH1", "ASCL1",
                                        "TUBA1A", "TUBB2B",
                                        # Impose ADRN Hypoxia Down
                                        "TAGLN3","NRSN1","TFAP2B","PRRT4","ROR2","PPEF1",
                                        # Impose ADRN Hypoxia Down
                                        "PPEF1", "POU3F2", "ISL1", "HOXD3", "FAM155B", "RAB39A", "OPRD1", "RBP1", "OAS1"
                                        )

write.table(DF2, "~/Desktop/Gepoliano/Hypoxia/Time Course/qPCR/DF2_RNA_ADRN_N_vs_H_4.txt",
            quote = F, sep = "\t", row.names = F)

dat <- data.frame(Group = c("S1", "S1", "S2", "S2"),
              Sub   = c("A", "B", "A", "B"),
              Value = c(3,5,7,8))  

ggplot(dat, aes(Group, Value)) +
  geom_bar(aes(fill = Sub), stat="identity", position="dodge", width=.5) +
  geom_signif(stat="identity",
              data=data.frame(x=c(0.875, 1.875), xend=c(1.125, 2.125),
                              y=c(5.8, 8.5), annotation=c("**", "NS")),
              aes(x=x,xend=xend, y=y, yend=y, annotation=annotation)) +
  geom_signif(comparisons=list(c("S1", "S2")), annotations="***",
              y_position = 9.3, tip_length = 0, vjust=0.4) +
  scale_fill_manual(values = c("grey80", "grey20"))

##______________________________________##
# 6) Select Genes to Print Normoxia/Hypoxia
DF1 <- DF_Identity_Pheno_Col %>% dplyr::select("FILE_NAME", "Pheno", "Cell_line", "Identity",
                                        "SPARC", "VIM", "PRRX1",
                                        # Impose MES Hypoxia Up
                                        "COL5A2","C1S",
                                        # Impose MES Hypoxia Up
                                        "COL27A1", "SPARC","SPARC","ADM","TAGLN","SLC16A3","ALDH3B1","SEMA3B",
                                        "STING1","MYLK3","VWF","STC1","BHLHE40","GLIPR1","PPP1R3G","C1RL","ORAI3",	
                                        "PHOX2B", "DACH1", "ASCL1",
                                        "TUBA1A", "TUBB2B",
                                        # Impose ADRN Hypoxia Down
                                        "TAGLN3","NRSN1","TFAP2B","PRRT4","ROR2","PPEF1",
                                        # Impose ADRN Hypoxia Down
                                        "PPEF1", "POU3F2", "ISL1", "HOXD3", "FAM155B", "RAB39A", "OPRD1", "RBP1", "OAS1"
                                        )





DF1_RNA_Normoxia <- DF1[ which(
    DF1$Pheno == 'N'  ), ]

write.table(DF1_RNA_Normoxia, "~/Desktop/Gepoliano/Hypoxia/Time Course/qPCR/DF1_RNA_Normoxia_4.txt",
            quote = F, sep = "\t", row.names = F)

DF1_RNA_Hypoxia <- DF1[ which(
    DF1$Pheno == 'H'  ), ]

write.table(DF1_RNA_Hypoxia, "~/Desktop/Gepoliano/Hypoxia/Time Course/qPCR/DF1_RNA_Hypoxia_4.txt",
            quote = F, sep = "\t", row.names = F)
```

# 13) Plot Proliferation Results

Here, factors need to be sorted manually

```{r}
MTT_Proliferation_Hypoxia <- read_excel("~/Desktop/Gepoliano/Hypoxia_Paper/MTT/MTT_Proliferation.xlsx")

# Change order manually Cell_Line_Treatment
MTT_Proliferation_Hypoxia$Cell_Line_Treatment <- factor(MTT_Proliferation_Hypoxia$Cell_Line_Treatment,          
                  levels = c("BE2_N", "BE2_H", 
                             "SY5Y_N", "SY5Y_H", 
                             "55N_N", "55N_H", 
                             "WN_N","WN_H",
                             "NBLS_N","NBLS_H",
                             "5S_N","5S_H",
                             "WS_N","WS_H",
                             "SHEP_N","SHEP_H"))

# Change order manually Cell_Line
MTT_Proliferation_Hypoxia$Cell_Line <- factor(MTT_Proliferation_Hypoxia$Cell_Line,          
                  levels = c("BE2",  
                             "SY5Y", 
                             "55N", 
                             "WN",
                             "NBLS",
                             "5S",
                             "WS",
                             "SHEP"))

# Change order manually Oxygen_Status
MTT_Proliferation_Hypoxia$Oxygen_Status <- factor(MTT_Proliferation_Hypoxia$Oxygen_Status,          
                  levels = c("Normoxia",  
                             "Hypoxia"))

ggplot(MTT_Proliferation_Hypoxia, aes(fill=Oxygen_Status, y=Absorbance_490, x=Cell_Line)) + 
    geom_bar(position="dodge", stat="identity")+
  theme(axis.text.x = element_text(angle=45, hjust=1))

```

# 7) Plot the Average Gene Expression in Normoxia

```{r}
df <- GSVA_GeneExpression_vsd_RNA_cells_DF # Genes x Samples Matrix

##______________________________________##
# 1) BE2
# Normoxia
df_select <- df %>% select("MA-001_FL_counts.txt", "MA-003_FL_counts.txt", "MA-005_FL_counts.txt")
df["BE2_Norm_Mean"] <- rowMeans(df_select)
# Hypoxia
df_select <- df %>% select("MA-002_FL_counts.txt", "MA-004_FL_counts.txt", "MA-006_FL_counts.txt")
df["BE2_Hypo_Mean"] <- rowMeans(df_select)

##______________________________________##
# 2) NBLS
# Normoxia
df_select <- df %>% select("MA-007_FL_counts.txt", "MA-009_FL_counts.txt", "MA-011_FL_counts.txt")
df["NBLS_Norm_Mean"] <- rowMeans(df_select)
# Hypoxia
df_select <- df %>% select("MA-008_FL_counts.txt", "MA-010_FL_counts.txt", "MA-012_FL_counts.txt")
df["NBLS_Hypo_Mean"] <- rowMeans(df_select)

##______________________________________##
# 3) SY5Y
# Normoxia
df_select <- df %>% select("MA-013_FL_counts.txt", "MA-015_FL_counts.txt", "MA-017_FL_counts.txt")
df["SY5Y_Norm_Mean"] <- rowMeans(df_select)
# Hypoxia
df_select <- df %>% select("MA-014_FL_counts.txt", "MA-016_FL_counts.txt", "MA-018_FL_counts.txt")
df["SY5Y_Hypo_Mean"] <- rowMeans(df_select)

##______________________________________##
# 4) SHEP
# Normoxia
df_select <- df %>% select("MA-019_FL_counts.txt", "MA-021_FL_counts.txt", "MA-023_FL_counts.txt")
df["SHEP_Norm_Mean"] <- rowMeans(df_select)
# Hypoxia
df_select <- df %>% select("MA-020_FL_counts.txt", "MA-022_FL_counts.txt", "MA-024_FL_counts.txt")
df["SHEP_Hypo_Mean"] <- rowMeans(df_select)

##______________________________________##
# 5) 55N
# Normoxia
df_select <- df %>% select("MA-025_FL_counts.txt", "MA-027_FL_counts.txt", "MA-029_FL_counts.txt")
df["55N_Norm_Mean"] <- rowMeans(df_select)
# Hypoxia
df_select <- df %>% select("MA-026_FL_counts.txt", "MA-028_FL_counts.txt", "MA-030_FL_counts.txt")
df["55N_Hypo_Mean"] <- rowMeans(df_select)

##______________________________________##
# 6) 5S
# Normoxia
df_select <- df %>% select("MA-031_FL_counts.txt", "MA-033_FL_counts.txt", "MA-035_FL_counts.txt")
df["5S_Norm_Mean"] <- rowMeans(df_select)
# Hypoxia
df_select <- df %>% select("MA-032_FL_counts.txt", "MA-034_FL_counts.txt", "MA-036_FL_counts.txt")
df["5S_Hypo_Mean"] <- rowMeans(df_select)

##______________________________________##
# 7) WN
# Normoxia
df_select <- df %>% select("MA-037_FL_counts.txt", "MA-039_FL_counts.txt", "MA-041_FL_counts.txt")
df["WN_Norm_Mean"] <- rowMeans(df_select)
# Hypoxia
df_select <- df %>% select("MA-038_FL_counts.txt", "MA-040_FL_counts.txt", "MA-042_FL_counts.txt")
df["WN_Hypo_Mean"] <- rowMeans(df_select)

##______________________________________##
# 8) WS
# Normoxia
df_select <- df %>% select("MA-043_FL_counts.txt", "MA-045_FL_counts.txt", "MA-047_FL_counts.txt")
df["WS_Norm_Mean"] <- rowMeans(df_select)
# Hypoxia
df_select <- df %>% select("MA-044_FL_counts.txt", "MA-046_FL_counts.txt", "MA-048_FL_counts.txt")
df["WS_Hypo_Mean"] <- rowMeans(df_select)

df_Nor_Mean <- df

# Construct DF with Hypoxia and Normoxia Average Expression
MeanGeneExpressionDF <- rbind(
cbind("WN"   , "Normoxia" , round(mean(df_Nor_Mean$WN_Norm_Mean)   , digits = 2)),
cbind("WS"   , "Normoxia" , round(mean(df_Nor_Mean$WS_Norm_Mean)   , digits = 2)),
cbind("55N"  , "Normoxia" , round(mean(df_Nor_Mean$`55N_Norm_Mean`), digits = 2)),
cbind("5S"   , "Normoxia" , round(mean(df_Nor_Mean$`5S_Norm_Mean`) , digits = 2)),
cbind("SY5Y" , "Normoxia" , round(mean(df_Nor_Mean$SY5Y_Norm_Mean) , digits = 2)),
cbind("SHEP" , "Normoxia" , round(mean(df_Nor_Mean$SHEP_Norm_Mean) , digits = 2)),
cbind("BE2"  , "Normoxia" , round(mean(df_Nor_Mean$BE2_Norm_Mean)  , digits = 2)),
cbind("NBLS" , "Normoxia" , round(mean(df_Nor_Mean$NBLS_Norm_Mean) , digits = 2)),
cbind("WN"   , "Hypoxia"  , round(mean(df_Nor_Mean$WN_Hypo_Mean)   , digits = 2)),
cbind("WS"   , "Hypoxia"  , round(mean(df_Nor_Mean$WS_Hypo_Mean)   , digits = 2)),
cbind("55N"  , "Hypoxia"  , round(mean(df_Nor_Mean$`55N_Hypo_Mean`), digits = 2)),
cbind("5S"   , "Hypoxia"  , round(mean(df_Nor_Mean$`5S_Hypo_Mean`) , digits = 2)),
cbind("SY5Y" , "Hypoxia"  , round(mean(df_Nor_Mean$SY5Y_Hypo_Mean) , digits = 2)),
cbind("SHEP" , "Hypoxia"  , round(mean(df_Nor_Mean$SHEP_Hypo_Mean) , digits = 2)),
cbind("BE2"  , "Hypoxia"  , round(mean(df_Nor_Mean$BE2_Hypo_Mean)  , digits = 2)),
cbind("NBLS" , "Hypoxia"  , round(mean(df_Nor_Mean$NBLS_Hypo_Mean) , digits = 2))
)

colnames(MeanGeneExpressionDF) <- c("Cell", "Treatment", "Mean Expression" )

```

# 8) Plot Bargraphs of the Means

# 8.1) WN_Norm_Mean

```{r}
sort(df_Nor_Mean$WN_Norm_Mean)

WN_Normoxia <- df_Nor_Mean$WN_Norm_Mean
hist(WN_Normoxia)

ggplot(df_Nor_Mean, aes(x = reorder(GeneSetTermID, WN_Norm_Mean), y = WN_Norm_Mean)) + geom_bar(stat = "identity")

test <- df_Nor_Mean
test_order_ascending <- test[order(test$WN_Norm_Mean),] 
test_order_ascending$index <- 1:nrow(test_order_ascending)

library("hutils")
quantile_object <- mutate_ntile(
  test_order_ascending,
  WN_Norm_Mean,
  3,
  weights = NULL,
  by = NULL,
  keyby = NULL,
  new.col = NULL,
  character.only = FALSE,
  overwrite = TRUE,
  check.na = FALSE)

rownames(quantile_object) <- rownames(test_order_ascending)

df_GeneIndexes_select <- quantile_object %>% select("WN_Norm_Mean", "index", "WN_Norm_MeanTerciles")

GeneIndexes <- subset(df_GeneIndexes_select, rownames(test_order_ascending) %in% c("PRRX1", "VIM", "SPARC","PHOX2B", "DACH1"))


```


# 10) Transform FPKM

# 10.1) Installation

```{r}
## Install dependencies
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install("ComplexHeatmap")

## Stable version, install countToFPKM from CRAN
install.packages("countToFPKM")

## Lastest version, install countToFPKM from GitHub
if(!require(devtools)) install.packages("devtools")
devtools::install_github("AAlhendi1707/countToFPKM", build_vignettes = TRUE)
```

# 10.2) Usage Example

```{r}
library(countToFPKM)

file.readcounts <- system.file("extdata", "RNA-seq.read.counts.csv", package="countToFPKM")
file.annotations <- system.file("extdata", "Biomart.annotations.hg38.txt", package="countToFPKM")
file.sample.metrics <- system.file("extdata", "RNA-seq.samples.metrics.txt", package="countToFPKM")

# Import the read count matrix data into R.
counts <- as.matrix(read.csv(file.readcounts))

# Import feature annotations. 
# Assign feature lenght into a numeric vector.
gene.annotations <- read.table(file.annotations, sep="\t", header=TRUE)

library("biomaRt")
ensemblsIDS <- gene.annotations$Gene_ID
ensembl = useMart("ensembl",dataset="hsapiens_gene_ensembl")
geneSymbols <- getBM(attributes=c('ensembl_gene_id','hgnc_symbol'), 
      filters = 'ensembl_gene_id', 
      values = ensemblsIDS, 
      mart = ensembl)

dt1 <- data.table(gene.annotations, key = "Gene_ID") 
dt2 <- data.table(geneSymbols, key = "ensembl_gene_id")
joined.dt1.dt.2 <- dt1[dt2]
gene.annotations.IDs <- dt1[dt2]

featureLength <- gene.annotations$length
featureLength_test <- gene.annotations.IDs$length

# Import sample metrics. 
# Assign mean fragment length into a numeric vector.
samples.metrics <- read.table(file.sample.metrics, sep="\t", header=TRUE)
meanFragmentLength <- samples.metrics$meanFragmentLength

# Return FPKM into a numeric matrix.
fpkm_matrix <- fpkm (counts, featureLength, meanFragmentLength)

# Plot log10(FPKM+1) heatmap of top 30 highly variable features
fpkmheatmap(fpkm_matrix, topvar=30, showfeaturenames=TRUE, return_log = TRUE)
```

# 10.3) Usage Our Lab

```{r}
library(countToFPKM)

file.readcounts <- system.file("extdata", "RNA-seq.read.counts.csv", package="countToFPKM")
file.annotations <- system.file("extdata", "Biomart.annotations.hg38.txt", package="countToFPKM")
file.sample.metrics <- system.file("extdata", "RNA-seq.samples.metrics.txt", package="countToFPKM")

# Import the read count matrix data into R.
counts <- as.matrix(read.csv(file.readcounts))
# Use DF_dds.RNA_General_ADRN_vs_MES as counts matrix
counts_transposed <- t(DF_dds.RNA_General_ADRN_vs_MES)

# Import feature annotations. 
# Assign feature lenght into a numeric vector.
gene.annotations <- read.table(file.annotations, sep="\t", header=TRUE)

library("biomaRt")
ensemblsIDS <- gene.annotations$Gene_ID
ensembl = useMart("ensembl",dataset="hsapiens_gene_ensembl")
geneSymbols <- getBM(attributes=c('ensembl_gene_id','hgnc_symbol'), 
      filters = 'ensembl_gene_id', 
      values = ensemblsIDS, 
      mart = ensembl)

dt1 <- data.table(gene.annotations, key = "Gene_ID") 
dt2 <- data.table(geneSymbols, key = "ensembl_gene_id")
joined.dt1.dt.2 <- dt1[dt2]
gene.annotations.IDs <- dt1[dt2]

#############################################
# 1) Match Counts and featureLength object 
#############################################

# featureLength is based on gene.annotations.IDs. 
# Therefore, need to intersect gene.annotations.IDs and counts file
# Below, we will find intersection gene.annotations.IDs and counts_transposed_IDs
# For that, we match  gene.annotations.IDs and counts_transposed_IDs by gene names
# get the columns of the tables:
leftCols <- gene.annotations.IDs$"hgnc_symbol"
rightCols <- rownames(counts_transposed_IDs)
# Find intersect
intersect_IDs <- intersect(leftCols, rightCols)
counts_transposed_IDs <- subset(counts_transposed, rownames(counts_transposed) %in% intersect_IDs)
counts_transposed_IDs <- subset(counts_transposed, rownames(counts_transposed) %in% gene.annotations.IDs$hgnc_symbol)

#############################################
# 2) Impose counts and featureLength same length
#############################################
counts_transposed_IDs <- as.data.frame(counts_transposed_IDs)

# Check lenghts intersect_IDs and counts files are the same
length(intersect_IDs) == length(counts_transposed_IDs$"MA-001_FL_counts.txt")

#############################################
# 3) Join gene.annotations.IDs and counts_transposed_IDs 
#############################################

# Revisit gene.annotations.IDs and include only intersect
# The goal is for counts_transposed_IDs and featureLength.Lab to have the same size
gene.annotations.IDs.2 <- subset(gene.annotations.IDs, gene.annotations.IDs$hgnc_symbol %in% rownames(counts_transposed_IDs))

length(intersect(rownames(counts_transposed_IDs), gene.annotations.IDs.2$hgnc_symbol))
#[1] 19059

intersect_3 <- intersect(rownames(counts_transposed_IDs), gene.annotations.IDs.2$hgnc_symbol)
setdiff_3 <- setdiff(rownames(counts_transposed_IDs), gene.annotations.IDs.2$hgnc_symbol) 

gene.annotations.IDs.3 <- subset(gene.annotations.IDs.2, gene.annotations.IDs.2$hgnc_symbol %in% intersect_3)

gene.annotations.IDs.2 <- subset(gene.annotations.IDs, gene.annotations.IDs$hgnc_symbol %in% gene.annotations.IDs$hgnc_symbol)
rowAnnotation <- rownames(counts_transposed_IDs)
counts_transposed_IDs.2 <- cbind(rowAnnotation, counts_transposed_IDs)

dt1 <- data.table(gene.annotations.IDs.2, key = "hgnc_symbol") 
dt2 <- data.table(counts_transposed_IDs.2, key = "rowAnnotation")
joined.dt1.dt.2 <- dt1[dt2]
gene.annotations.IDs <- dt1[dt2]

#############################################
# 4) Get counts and frag length from joined.dt1.dt.2$length
#############################################

# Remove duplicates from joined.dt1.dt.2
library(dplyr)
joined.dt1.dt.2_dedup <- joined.dt1.dt.2 %>% distinct(hgnc_symbol, .keep_all= TRUE)

counts_Lab <- joined.dt1.dt.2_dedup %>% select("hgnc_symbol", "MA-001_FL_counts.txt":"MA-048_FL_counts.txt")
as.data.frame(counts_Lab)
row.names(counts_Lab) <- counts_Lab$hgnc_symbol
counts_Lab.2 <- subset(counts_Lab, select=-c(hgnc_symbol))

featureLength <- gene.annotations$length
featureLength.Lab <- joined.dt1.dt.2_dedup$length

# Import sample metrics. 
# Assign mean fragment length into a numeric vector.
samples.metrics <- read.table(file.sample.metrics, sep="\t", header=TRUE)
samples.metrics.lab <- cell_line_pheno_RNA_Seq %>% select("FILE_NAME", "meanFragmentLength", "Cell_line")
meanFragmentLength <- samples.metrics$meanFragmentLength
meanFragmentLengthLab <- samples.metrics.lab$meanFragmentLength

# Return FPKM into a numeric matrix.
fpkm_matrix <- fpkm(counts, featureLength, meanFragmentLength)
fpkm_matrix <- fpkm(counts_Lab, featureLength.Lab, meanFragmentLengthLab)
fpkm_matrix <- fpkm(counts_Lab.2, featureLength.Lab, meanFragmentLengthLab)

rownames(fpkm_matrix) <- rownames(counts_Lab)

# Plot log10(FPKM+1) heatmap of top 30 highly variable features
fpkmheatmap(fpkm_matrix, topvar=30, showfeaturenames=TRUE, return_log = TRUE)
```

# 10.4) FPKM

```{r}
library("RNAAgeCalc") # Get FPKM from counts
counts_transposed_fpkm <- count2FPKM(counts_transposed, genelength = NULL, idtype = "SYMBOL")
# Plot log10(FPKM+1) heatmap of top 30 highly variable features
fpkmheatmap(counts_transposed_fpkm, topvar=30, showfeaturenames=TRUE, return_log = TRUE)

rownames(counts_transposed) <- counts_transposed[1,]
```

# 10.5) Plot FPKM Genes of Interest

```{r}
intersect_final <- c("VIM")

my_list_input <- "intersect_final"
var_name <- 'my_list_output'
assign(var_name, as.name(my_list_input))
my_list_output

for (term in intersect_final){
print(paste0("Before make numeric, starting with term,  ", term))
library(ggplot2)

# Use DF that was not filtered by terms
DF_dds.RNA_General_ADRN_vs_MES <- as.data.frame(assay(dds.RNA_General_ADRN_vs_MES))
DF_dds.RNA_General_ADRN_vs_MES <- t(DF_dds.RNA_General_ADRN_vs_MES)
counts_transposed_fpkm <- t(counts_transposed_fpkm)

# write.table(t(counts_transposed_fpkm), 
#             file = "~/Desktop/Gepoliano/Hypoxia/Time Course/qPCR/counts_fpkm.txt", 
#             sep = "\t", 
#             col.names = T,
#             row.names = T)

Sample_ID = cell_line_pheno_RNA_Seq_Realignment$FILE_NAME
df <- cbind(Sample_ID, counts_transposed_fpkm)

##_______________________________________________________##
# 1) Join DFs and ensure as.data.frame, but w/ all terms

dt1 <- data.table(cell_line_pheno_RNA_Seq_Realignment, key = "FILE_NAME") 
dt2 <- data.table(df, key = "Sample_ID")
joined.dt1.dt.2 <- dt1[dt2]
DF_Identity_Pheno_Col <- dt1[dt2]

# Make tarms of interest, numeric
# If use term here, will make term store the wrong value
for (term_to_numeric in intersect_final){
DF_Identity_Pheno_Col[[term_to_numeric]] <- as.numeric(DF_Identity_Pheno_Col[[term_to_numeric]])
}

##_______________________________________________________##
# 2) Prepare Data for Statistics

library(ggpubr)
library(rstatix)

# 2.1) Data preparation

# Transform `dose` into factor variable
df <- DF_Identity_Pheno_Col

df$Identity <- as.factor(df$Identity)

# Add a random grouping variable
df$group <- factor(rep(c("grp1", "grp2"), 24))

# 4.2) Apply T-test

##______________________________________##
# 4.2.1) Apply T-test in Pheno (Hypoxia)

#term <- "CHN1" # If this is left uncommented, only this term will be plotted
my_gene_input <- term
var_name <- 'my_gene_output'
assign(var_name, as.name(my_gene_input))
my_gene_output

# ADRN Pheno (Normoxia vs. Hypoxia)
t_pval_ADN_HStat <- t.test(df[[my_gene_output]][df$Identity=="ADRN"][df$Pheno=="N"], df[[my_gene_output]][df$Identity=="ADRN"][df$Pheno=="H"])$p.value

# Isolate Means
t_estimate_ADN_HStat <- t.test(df[[my_gene_output]][df$Identity=="ADRN"][df$Pheno=="N"], df[[my_gene_output]][df$Identity=="ADRN"][df$Pheno=="H"])$estimate

t_estimate_ADN_HStat_DF <- as.data.frame(t_estimate_ADN_HStat)

df_adrn <- df[df$Identity=="ADRN"]

##______________________________________##
# 4.2.2) Plot if significant

# Plot if the T-test means follow the desired trend for ADRN genes under Hypoxia.

print(paste0("term is ", term))

ifelse(
  (t_pval_ADN_HStat < 0.05) & 
    # Impose condition to print ADRN genes that are down-regulated by Hypoxia (ADRN cells combined)
    # The condition is based on Means of Normoxia and Hypoxia treatments
    (t_estimate_ADN_HStat_DF[1,1] < t_estimate_ADN_HStat_DF[2,1]),# Condition ADRN_H > ADRN_N
print(ggboxplot(df_adrn, x = "Pheno", y = term,
          color = "Cell_line", palette = "jco",
          add = "jitter") + stat_compare_means(method = "t.test")),
print(paste0(term, " p-val is ", t_pval_ADN_HStat))
)
}

ggplot(df_adrn, aes(x = reorder(Cell_line, PHOX2B), y = PHOX2B)) + geom_bar(stat = "identity")

df_adrn_modified <- df_adrn
df_adrn_modified$Pheno <- factor(df_adrn_modified$Pheno, levels = rev(unique(df_adrn_modified$Pheno)), ordered=TRUE)
df_adrn_modified$Cell_line <- factor(df_adrn_modified$Cell_line, levels = rev(unique(df_adrn_modified$Cell_line)), ordered=TRUE)

ggbarplot(df_adrn_modified, x = "Pheno", y = intersect_final, fill = "Pheno",
          sort.by.groups = FALSE
          )

##______________________________________##
# 4.2.3) ggbarplot Graph

library(ggpubr)
df <- df_adrn_modified

# Change order manually Pheno
df_adrn_modified$Pheno <- factor(df_adrn_modified$Pheno,          
                  levels = c("N",  
                             "H"))
ggbarplot(df_adrn_modified, x = "Pheno", y = intersect_final,
                fill = "Cell_line",
                add = "mean_sd", add.params = list(group = "Cell_line"),
                position = position_dodge(0.8)) 

ggplot(df_adrn_modified, aes(fill=Pheno, y=PHOX2B, x=Cell_line)) + 
    geom_bar(position="dodge", stat="identity")+
  theme(axis.text.x = element_text(angle=45, hjust=1))

ggplot(df_adrn_modified, aes(fill=Pheno, y=VIM, x=reorder(Cell_line, VIM))) + 
    geom_bar(position="dodge", stat="identity")+
  theme(axis.text.x = element_text(angle=45, hjust=1))+
  scale_fill_manual(values = c("#00008b","#ff0c0c"), labels = c("H","N"))

##______________________________________##
# 5) Select Genes to Print Hypoxia/ADRN

# Change order manually Pheno
df_adrn_modified$Pheno <- factor(df_adrn_modified$Pheno,          
                  levels = c("N",  
                             "H"))
ggplot(df_adrn_modified, aes(fill=Pheno, y=VIM, x=reorder(Cell_line, VIM))) + 
    geom_bar(position="dodge", stat="identity")+
  theme(axis.text.x = element_text(angle=45, hjust=1))+
  scale_fill_manual(values = c("#ff0c0c","#00008b"), labels = c("N","H")) +
  stat_compare_means(method = "t.test", label = "p.signif", label.y = c(50, 80, 80, 80, 80 ))

library(dplyr)

DF2 <- df_adrn_modified %>% select("FILE_NAME", "Pheno", "Cell_line", "Identity",
                                                      "SPARC", "VIM", "PRRX1", "PHOX2B", "DACH1", "ASCL1",
                                                      "TUBA1A", "TUBB2B")

# write.table(DF1, "~/Desktop/Gepoliano/Hypoxia/Time Course/qPCR/DF2_RNA_ADRN_N_vs_H.txt", 
#             quote = F, sep = "\t", row.names = F)

dat <- data.frame(Group = c("S1", "S1", "S2", "S2"),
              Sub   = c("A", "B", "A", "B"),
              Value = c(3,5,7,8))  

ggplot(dat, aes(Group, Value)) +
  geom_bar(aes(fill = Sub), stat="identity", position="dodge", width=.5) +
  geom_signif(stat="identity",
              data=data.frame(x=c(0.875, 1.875), xend=c(1.125, 2.125),
                              y=c(5.8, 8.5), annotation=c("**", "NS")),
              aes(x=x,xend=xend, y=y, yend=y, annotation=annotation)) +
  geom_signif(comparisons=list(c("S1", "S2")), annotations="***",
              y_position = 9.3, tip_length = 0, vjust=0.4) +
  scale_fill_manual(values = c("grey80", "grey20"))

##______________________________________##
# 6) Select Genes to Print Normoxia/Hypoxia
DF1 <- DF_Identity_Pheno_Col %>% select("FILE_NAME", "Pheno", "Cell_line", "Identity",
                                                      "SPARC", "VIM", "PRRX1", "PHOX2B", "DACH1", "ASCL1",
                                                      "TUBA1A", "TUBB2B")

DF1_RNA_Normoxia <- DF1[ which(
    DF1$Pheno == 'N'  ), ]

# write.table(DF1_RNA_Normoxia, "~/Desktop/Gepoliano/Hypoxia/Time Course/qPCR/DF1_RNA_Normoxia.txt", 
#             quote = F, sep = "\t", row.names = F)

DF1_RNA_Hypoxia <- DF1[ which(
    DF1$Pheno == 'H'  ), ]

# write.table(DF1_RNA_Hypoxia, "~/Desktop/Gepoliano/Hypoxia/Time Course/qPCR/DF1_RNA_Hypoxia.txt", 
#             quote = F, sep = "\t", row.names = F)
```

# 10.6) Plot FPKM Genes of Interest (Expors ADRN/MES and H->ADRN DFs)

Need to figure out why in the first run this code is not successful

```{r}
intersect_final <- c("VIM")

my_list_input <- "intersect_final"
var_name <- 'my_list_output'
assign(var_name, as.name(my_list_input))
my_list_output

for (term in intersect_final){
print(paste0("Before make numeric, starting with term,  ", term))
library(ggplot2)

# Use DF that was not filtered by terms
DF_dds.RNA_General_ADRN_vs_MES <- as.data.frame(assay(dds.RNA_General_ADRN_vs_MES))
DF_dds.RNA_General_ADRN_vs_MES <- t(DF_dds.RNA_General_ADRN_vs_MES)
counts_transposed_fpkm <- t(counts_transposed_fpkm)

##_______________________________________________________##
# Average Gene Expression
counts_transposed_fpkm_df <- as.data.frame(t(counts_transposed_fpkm))
mean(
mean(counts_transposed_fpkm_df$MA.001_FL_counts.txt, na.rm=TRUE),
mean(counts_transposed_fpkm_df$MA.002_FL_counts.txt, na.rm=TRUE),
mean(counts_transposed_fpkm_df$MA.003_FL_counts.txt, na.rm=TRUE),
mean(counts_transposed_fpkm_df$MA.004_FL_counts.txt, na.rm=TRUE),
mean(counts_transposed_fpkm_df$MA.005_FL_counts.txt, na.rm=TRUE),
mean(counts_transposed_fpkm_df$MA.006_FL_counts.txt, na.rm=TRUE),
mean(counts_transposed_fpkm_df$MA.007_FL_counts.txt, na.rm=TRUE),
mean(counts_transposed_fpkm_df$MA.008_FL_counts.txt, na.rm=TRUE),
mean(counts_transposed_fpkm_df$MA.009_FL_counts.txt, na.rm=TRUE),
mean(counts_transposed_fpkm_df$MA.010_FL_counts.txt, na.rm=TRUE),
mean(counts_transposed_fpkm_df$MA.011_FL_counts.txt, na.rm=TRUE),
mean(counts_transposed_fpkm_df$MA.012_FL_counts.txt, na.rm=TRUE),

mean(counts_transposed_fpkm_df$MA.013_FL_counts.txt, na.rm=TRUE),
mean(counts_transposed_fpkm_df$MA.014_FL_counts.txt, na.rm=TRUE),
mean(counts_transposed_fpkm_df$MA.015_FL_counts.txt, na.rm=TRUE),
mean(counts_transposed_fpkm_df$MA.016_FL_counts.txt, na.rm=TRUE),
mean(counts_transposed_fpkm_df$MA.017_FL_counts.txt, na.rm=TRUE),
mean(counts_transposed_fpkm_df$MA.018_FL_counts.txt, na.rm=TRUE),
mean(counts_transposed_fpkm_df$MA.019_FL_counts.txt, na.rm=TRUE),
mean(counts_transposed_fpkm_df$MA.020_FL_counts.txt, na.rm=TRUE),
mean(counts_transposed_fpkm_df$MA.021_FL_counts.txt, na.rm=TRUE),
mean(counts_transposed_fpkm_df$MA.022_FL_counts.txt, na.rm=TRUE),
mean(counts_transposed_fpkm_df$MA.023_FL_counts.txt, na.rm=TRUE),
mean(counts_transposed_fpkm_df$MA.024_FL_counts.txt, na.rm=TRUE),

mean(counts_transposed_fpkm_df$MA.025_FL_counts.txt, na.rm=TRUE),
mean(counts_transposed_fpkm_df$MA.026_FL_counts.txt, na.rm=TRUE),
mean(counts_transposed_fpkm_df$MA.027_FL_counts.txt, na.rm=TRUE),
mean(counts_transposed_fpkm_df$MA.028_FL_counts.txt, na.rm=TRUE),
mean(counts_transposed_fpkm_df$MA.029_FL_counts.txt, na.rm=TRUE),
mean(counts_transposed_fpkm_df$MA.030_FL_counts.txt, na.rm=TRUE),
mean(counts_transposed_fpkm_df$MA.031_FL_counts.txt, na.rm=TRUE),
mean(counts_transposed_fpkm_df$MA.032_FL_counts.txt, na.rm=TRUE),
mean(counts_transposed_fpkm_df$MA.033_FL_counts.txt, na.rm=TRUE),
mean(counts_transposed_fpkm_df$MA.034_FL_counts.txt, na.rm=TRUE),
mean(counts_transposed_fpkm_df$MA.035_FL_counts.txt, na.rm=TRUE),
mean(counts_transposed_fpkm_df$MA.036_FL_counts.txt, na.rm=TRUE),

mean(counts_transposed_fpkm_df$MA.037_FL_counts.txt, na.rm=TRUE),
mean(counts_transposed_fpkm_df$MA.038_FL_counts.txt, na.rm=TRUE),
mean(counts_transposed_fpkm_df$MA.039_FL_counts.txt, na.rm=TRUE),
mean(counts_transposed_fpkm_df$MA.040_FL_counts.txt, na.rm=TRUE),
mean(counts_transposed_fpkm_df$MA.041_FL_counts.txt, na.rm=TRUE),
mean(counts_transposed_fpkm_df$MA.042_FL_counts.txt, na.rm=TRUE),
mean(counts_transposed_fpkm_df$MA.043_FL_counts.txt, na.rm=TRUE),
mean(counts_transposed_fpkm_df$MA.044_FL_counts.txt, na.rm=TRUE),
mean(counts_transposed_fpkm_df$MA.045_FL_counts.txt, na.rm=TRUE),
mean(counts_transposed_fpkm_df$MA.046_FL_counts.txt, na.rm=TRUE),
mean(counts_transposed_fpkm_df$MA.047_FL_counts.txt, na.rm=TRUE),
mean(counts_transposed_fpkm_df$MA.048_FL_counts.txt, na.rm=TRUE)
)


# write.table(t(counts_transposed_fpkm),
#             file = "~/Desktop/Gepoliano/Hypoxia/Time Course/qPCR/counts_fpkm_2_RNA_Hypoxia_AC.txt",
#             sep = "\t",
#             col.names = T,
#             row.names = T)

Sample_ID = cell_line_pheno_RNA_Seq_Realignment$FILE_NAME
df <- cbind(Sample_ID, counts_transposed_fpkm)

##_______________________________________________________##
# 1) Join DFs and ensure as.data.frame, but w/ all terms

dt1 <- data.table(cell_line_pheno_RNA_Seq_Realignment, key = "FILE_NAME") 
dt2 <- data.table(df, key = "Sample_ID")
joined.dt1.dt.2 <- dt1[dt2]
DF_Identity_Pheno_Col <- dt1[dt2]

# Make tarms of interest, numeric
# If use term here, will make term store the wrong value
for (term_to_numeric in intersect_final){
DF_Identity_Pheno_Col[[term_to_numeric]] <- as.numeric(DF_Identity_Pheno_Col[[term_to_numeric]])
}

##_______________________________________________________##
# 2) Prepare Data for Statistics

library(ggpubr)
library(rstatix)

# 2.1) Data preparation

# Transform `dose` into factor variable
df <- DF_Identity_Pheno_Col

df$Identity <- as.factor(df$Identity)

# Add a random grouping variable
df$group <- factor(rep(c("grp1", "grp2"), 24))

# 4.2) Apply T-test

##______________________________________##
# 4.2.1) Apply T-test in Pheno (Hypoxia)

#term <- "CHN1" # If this is left uncommented, only this term will be plotted
my_gene_input <- term
var_name <- 'my_gene_output'
assign(var_name, as.name(my_gene_input))
my_gene_output

# ADRN Pheno (Normoxia vs. Hypoxia)
t_pval_ADN_HStat <- t.test(df[[my_gene_output]][df$Identity=="ADRN"][df$Pheno=="N"], df[[my_gene_output]][df$Identity=="ADRN"][df$Pheno=="H"])$p.value

# Isolate Means
t_estimate_ADN_HStat <- t.test(df[[my_gene_output]][df$Identity=="ADRN"][df$Pheno=="N"], df[[my_gene_output]][df$Identity=="ADRN"][df$Pheno=="H"])$estimate

t_estimate_ADN_HStat_DF <- as.data.frame(t_estimate_ADN_HStat)

df_adrn <- df[df$Identity=="ADRN"]

##______________________________________##
# 4.2.2) Plot if significant

# Plot if the T-test means follow the desired trend for ADRN genes under Hypoxia.

print(paste0("term is ", term))

ifelse(
  (t_pval_ADN_HStat < 0.05) & 
    # Impose condition to print ADRN genes that are down-regulated by Hypoxia (ADRN cells combined)
    # The condition is based on Means of Normoxia and Hypoxia treatments
    (t_estimate_ADN_HStat_DF[1,1] < t_estimate_ADN_HStat_DF[2,1]),# Condition ADRN_H > ADRN_N
print(ggboxplot(df_adrn, x = "Pheno", y = term,
          color = "Cell_line", palette = "jco",
          add = "jitter") + stat_compare_means(method = "t.test")),
print(paste0(term, " p-val is ", t_pval_ADN_HStat))
)
}

ggplot(df_adrn, aes(x = reorder(Cell_line, PHOX2B), y = PHOX2B)) + geom_bar(stat = "identity")

df_adrn_modified <- df_adrn
df_adrn_modified$Pheno <- factor(df_adrn_modified$Pheno, levels = rev(unique(df_adrn_modified$Pheno)), ordered=TRUE)
df_adrn_modified$Cell_line <- factor(df_adrn_modified$Cell_line, levels = rev(unique(df_adrn_modified$Cell_line)), ordered=TRUE)

ggbarplot(df_adrn_modified, x = "Pheno", y = intersect_final, fill = "Pheno",
          sort.by.groups = FALSE
          )

##______________________________________##
# 4.2.3) ggbarplot Graph

library(ggpubr)
df <- df_adrn_modified

# Change order manually Pheno
df_adrn_modified$Pheno <- factor(df_adrn_modified$Pheno,          
                  levels = c("N",  
                             "H"))
ggbarplot(df_adrn_modified, x = "Pheno", y = intersect_final,
                fill = "Cell_line",
                add = "mean_sd", add.params = list(group = "Cell_line"),
                position = position_dodge(0.8)) 

ggplot(df_adrn_modified, aes(fill=Pheno, y=PHOX2B, x=Cell_line)) + 
    geom_bar(position="dodge", stat="identity")+
  theme(axis.text.x = element_text(angle=45, hjust=1))

ggplot(df_adrn_modified, aes(fill=Pheno, y=VIM, x=reorder(Cell_line, VIM))) + 
    geom_bar(position="dodge", stat="identity")+
  theme(axis.text.x = element_text(angle=45, hjust=1))+
  scale_fill_manual(values = c("#00008b","#ff0c0c"), labels = c("H","N"))

##______________________________________##
# 5) Select Genes to Print Hypoxia/ADRN

# Change order manually Pheno
df_adrn_modified$Pheno <- factor(df_adrn_modified$Pheno,          
                  levels = c("N",  
                             "H"))
ggplot(df_adrn_modified, aes(fill=Pheno, y=VIM, x=reorder(Cell_line, VIM))) + 
    geom_bar(position="dodge", stat="identity")+
  theme(axis.text.x = element_text(angle=45, hjust=1))+
  scale_fill_manual(values = c("#ff0c0c","#00008b"), labels = c("N","H")) +
  stat_compare_means(method = "t.test", label = "p.signif", label.y = c(50, 80, 80, 80, 80 ))
 
DF2 <- df_adrn_modified %>% dplyr::select("FILE_NAME", "Pheno", "Cell_line", "Identity",
                                                      "SPARC", "VIM", "PRRX1", "PHOX2B", "DACH1", "ASCL1",
                                                      "TUBA1A", "TUBB2B")

# write.table(DF1, "~/Desktop/Gepoliano/Hypoxia/Time Course/qPCR/DF2_FPKM_RNA_ADRN_N_vs_H.txt",
#             quote = F, sep = "\t", row.names = F)

dat <- data.frame(Group = c("S1", "S1", "S2", "S2"),
              Sub   = c("A", "B", "A", "B"),
              Value = c(3,5,7,8))  

ggplot(dat, aes(Group, Value)) +
  geom_bar(aes(fill = Sub), stat="identity", position="dodge", width=.5) +
  geom_signif(stat="identity",
              data=data.frame(x=c(0.875, 1.875), xend=c(1.125, 2.125),
                              y=c(5.8, 8.5), annotation=c("**", "NS")),
              aes(x=x,xend=xend, y=y, yend=y, annotation=annotation)) +
  geom_signif(comparisons=list(c("S1", "S2")), annotations="***",
              y_position = 9.3, tip_length = 0, vjust=0.4) +
  scale_fill_manual(values = c("grey80", "grey20"))

##______________________________________##
# 6) Select Genes to Print Normoxia/Hypoxia
DF1 <- DF_Identity_Pheno_Col %>% dplyr::select("FILE_NAME", "Pheno", "Cell_line", "Identity",
                                                      "SPARC", "VIM", "PRRX1", "PHOX2B", "DACH1", "ASCL1",
                                                      "TUBA1A", "TUBB2B")

DF1_RNA_Normoxia <- DF1[ which(
    DF1$Pheno == 'N'  ), ]

# write.table(DF1_RNA_Normoxia, "~/Desktop/Gepoliano/Hypoxia/Time Course/qPCR/DF1_FPKM_RNA_Normoxia.txt",
#             quote = F, sep = "\t", row.names = F)

DF1_RNA_Hypoxia <- DF1[ which(
    DF1$Pheno == 'H'  ), ]

# write.table(DF1_RNA_Hypoxia, "~/Desktop/Gepoliano/Hypoxia/Time Course/qPCR/DF1_FPKM_RNA_Hypoxia.txt",
#             quote = F, sep = "\t", row.names = F)
```

# 10.7) Test Canonical Genes (FPKM)

Check that the expression of these canonical markers follow qualitative, known observations for the comparison of ADRN/MES genes. This chunk is based on chunk #4.8.1) All ADRN all MES cells.
In the chunk, I should use just the first part, containing ADRN/MES or second part, containing N/H.

```{r}
##_______________________________________________________##
# 1) Prepare Data for Statistics

library(ggplot2)

# Pass the value of counts_transposed_fpkm to DF with proper name
# The good thing about counts_transposed_fpkm is that it contains 
# only gene names and not a column with rownames
counts_DF_fpkm_AC <- counts_transposed_fpkm

# Figured it is still required to include gene name column 
# Therefore I had to revisit 10.6) Plot FPKM Genes of Interest (Expors ADRN/MES and H->ADRN DFs)
# To make sure I got the Metadata information from cell_line_pheno_RNA_Seq_Realignment

Sample_ID = cell_line_pheno_RNA_Seq_Realignment$FILE_NAME
df <- cbind(Sample_ID, counts_DF_fpkm_AC)

# Merge DFs to include Metadata information
dt1 <- data.table(cell_line_pheno_RNA_Seq_Realignment, key = "FILE_NAME") 
dt2 <- data.table(df, key = "Sample_ID")
joined.dt1.dt.2 <- dt1[dt2]
DF_fpkm_Canonical_Genes <- dt1[dt2]

##_______________________________________________________##
# 2) Prepare Data for Statistics

library(ggpubr)
library(rstatix)

# 1) Data preparation

# Select pair of interest
# df <- df[ which(
#   df$Cell_line=="55N" |
#      df$Cell_line=="5S" ), ]

# Transform `Identity` into factor variable
DF_fpkm_Canonical_Genes$Identity <- as.factor(DF_fpkm_Canonical_Genes$Identity)

# Add a random grouping variable
DF_fpkm_Canonical_Genes$group <- factor(rep(c("grp1", "grp2"), 24))

##_______________________________________________________##
# 3) Facet wrap by one variable (Pheno)

# Make Gene Expression values, numeric. 
# This is peculiar for this DF, because it contains "all" genes.
DF_fpkm_Canonical_Genes$SPARC <- as.numeric(DF_fpkm_Canonical_Genes$SPARC)

##______________________________________##
# 3.1) Constructs Statistics Object
# Note: At one point, I made what should be the fpkm DF, the t-test DF
stat.test_fpkm_Canonical_Genes <- DF_fpkm_Canonical_Genes %>%
  group_by(Pheno) %>%
  t_test(SPARC ~ Identity) %>%
  adjust_pvalue(method = "bonferroni") %>%
  add_significance()

##______________________________________##
# 3.2) Construct Boxplot and Facet (by Pheno)

# Create a box plot
bxp <- ggboxplot(
  DF_fpkm_Canonical_Genes, x = "Identity", y = "SPARC", 
  color = "Cell_line",
  facet.by = "Pheno"
  )

##______________________________________##
# 3.3) Add p-values

# Make facet and add p-values
stat.test_fpkm_Canonical_Genes <- stat.test_fpkm_Canonical_Genes %>% add_xy_position(x = "Identity")
bxp + stat_pvalue_manual(stat.test_fpkm_Canonical_Genes)

##_______________________________________________________##
# 4) Facet wrap by one variable (Identity)

# 4.1) Facet wrap by one variable

stat.test_fpkm_Canonical_Genes <- DF_fpkm_Canonical_Genes %>%
  group_by(Identity) %>%
  t_test(SPARC ~ Pheno) %>%
  adjust_pvalue(method = "bonferroni") %>%
  add_significance()
# View(stat.test) 

# 3) Boxplot t-test

# Create a box plot
plot_object <- ggboxplot(
  DF_fpkm_Canonical_Genes, x = "Pheno", y = "SPARC",
  color = "Cell_line",
  facet.by = "Identity"
  )

# Make facet and add p-values
stat.test_fpkm_Canonical_Genes <- stat.test_fpkm_Canonical_Genes %>% add_xy_position(x = "Pheno")
plot_object + stat_pvalue_manual(stat.test, bracket.nudge.y = c(.1, .2), hide.ns = FALSE,
    label = "{p.adj} ({p.adj.signif})") # Included space between p-value and asterisks
```

# 4.8.2) One Cell Pair at a Time

```{r}
##_______________________________________________________##
# 1) Prepare Data for Statistics

library(ggplot2)

# Pass the value of counts_transposed_fpkm to DF with proper name
# The good thing about counts_transposed_fpkm is that it contains 
# only gene names and not a column with rownames
counts_DF_fpkm_AC <- counts_transposed_fpkm

# Figured it is still required to include gene name column 
# Therefore I had to revisit 10.6) Plot FPKM Genes of Interest (Expors ADRN/MES and H->ADRN DFs)
# To make sure I got the Metadata information from cell_line_pheno_RNA_Seq_Realignment

Sample_ID = cell_line_pheno_RNA_Seq_Realignment$FILE_NAME
df <- cbind(Sample_ID, counts_DF_fpkm_AC)

# Merge DFs to include Metadata information
dt1 <- data.table(cell_line_pheno_RNA_Seq_Realignment, key = "FILE_NAME") 
dt2 <- data.table(df, key = "Sample_ID")
joined.dt1.dt.2 <- dt1[dt2]
DF_fpkm_Canonical_Genes <- dt1[dt2]








# library
library(ggplot2)

#my_gene <- readline(prompt="Enter Gene of Interest: ")
# Gene hard name: B4GALNT1

# DF_Identity_Pheno_Col <- add_column(GSVA_GeneExpression_SampleIDs, ID_Pheno_Col = eff_vector, .after = "Identity")

DF_Identity_Pheno_Col <- GSVA_GeneExpression_SampleIDs

##_______________________________________________________##
# 2) Prepare Data for Statistics

library(ggpubr)
library(rstatix)

# 1) Data preparation

# Transform `dose` into factor variable
df <- DF_Identity_Pheno_Col

# Select pair of interest
df <- DF_fpkm_Canonical_Genes[ which(
  DF_fpkm_Canonical_Genes$Cell_line=="55N" |
     df$Cell_line=="5S" ), ]

df$Cell_line_type <- as.factor(df$Cell_line_type)

# Add a random grouping variable
df$group <- factor(rep(c("grp1", "grp2"), 6))

##_______________________________________________________##
# 3) Facet wrap by one variable (Pheno)

##______________________________________##
# 3.1) Constructs Statistics Object

stat.test <- df %>%
  group_by(Pheno) %>%
  t_test(SPARC ~ Cell_line_type) %>%
  adjust_pvalue(method = "bonferroni") %>%
  add_significance()
# View(stat.test) 

##______________________________________##
# 3.2) Construct Boxplot and Facet (by Pheno)

# Create a box plot
bxp <- ggboxplot(
  df, x = "Cell_line_type", y = "SPARC", 
  color = "Cell_line",
  facet.by = "Pheno", scales = "free", add = "jitter"
  )

##______________________________________##
# 3.3) Add p-values

# Make facet and add p-values
stat.test <- stat.test %>% add_xy_position(x = "Cell_line_type")
bxp + stat_pvalue_manual(stat.test)

##_______________________________________________________##
# 4) Facet wrap by one variable (Identity)

# 4.1) Facet wrap by one variable

stat.test <- df %>%
  group_by(Cell_line_type) %>%
  t_test(SPARC ~ Pheno) %>%
  adjust_pvalue(method = "bonferroni") %>%
  add_significance()
# View(stat.test) 

# 3) Boxplot t-test

# Create a box plot
plot_object <- ggboxplot(
  df, x = "Pheno", y = "SPARC",
  color = "Cell_line",
  facet.by = "Cell_line_type", 
  scales = "free", 
  add = "jitter"
  )

# Make facet and add p-values
stat.test <- stat.test %>% add_xy_position(x = "Pheno")
plot_object + stat_pvalue_manual(stat.test, bracket.nudge.y = c(.1, .2), hide.ns = FALSE,
    label = "{p.adj} ({p.adj.signif})") # Included space between p-value and asterisks
```

# 11) Barplot RT-PCR

# 11.1) SPARC/ACTN

```{r}
library(rstatix)
library(readxl)
library(ggpubr)

directory <- "~/Desktop/Gepoliano/Applebaum Lab Meeting/2022/06-07-2022/"
excel_file <- "SPARC_Expression.xlsx"
sparc_expression <- read_excel(paste0(directory, excel_file))

# Reorder Sample_Treated, so Normoxia comes before Hypoxia in the graph
sparc_expression$Sample_Treated <- factor(
  sparc_expression$Sample_Treated , levels=c("WN_N", "WN_H", "WS_N", "WS_H"))

# Statistical Test
# stat.test is a DF with the padj and the Stars
stat.test <- sparc_expression %>%
  group_by(`Cell Line`) %>%
  t_test(RQ ~ Condition) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")
stat.test

## Create Barplot
# Change the width of bars
barplot <- ggbarplot(sparc_expression, "Sample_Treated", "RQ",
          fill = "Sample_Treated", color = "Sample_Treated",
          palette = c("#0412d4", "#f71105", "#0412d4", "#f71105"))

## This part prevents red and blue plot
# barplot <- ggplot(data=sparc_expression, aes(
#   x=Sample_Treated, 
#   y=RQ,
#   fill="Sample_Treated")) +
#   geom_bar(stat="identity", width=0.5)+
#   scale_fill_manual(breaks = c("WN_N", "WN_H", "WS_N", "WS_H"), 
#                        values=c("red", "blue", "green", "yellow"))
# 
# # ## Add p-values onto the plot
# # # add_xy_position adds x and y positions
# # stat.test <- stat.test %>%
# #   add_xy_position(x = "Cell Line", dodge = 0.8)
# # barplot + stat_pvalue_manual(
# #   stat.test,  
# #   #label = "p.adj.signif",
# #   #tip.length = 0
# #   )

# Add 10% spaces between the p-value labels and the plot border
barplot <- barplot + 
  #stat_pvalue_manual(
  #stat.test,  
  #label = "p.adj.signif",
  #tip.length = 0,
  #y.position = c(12, 52)
  #)+
  #scale_y_continuous(stat.test, expand = expansion(mult = c(0, 0.1)))+
  geom_bracket(
    xmin = c("WN_N", "WS_N","WN_N" ), 
    xmax = c("WN_H", "WS_H", "WS_N"), 
    y.position = c(20, 54, 60),
    label = c("***", "ns", "****")
  ) 

barplot
```

# 11.2) SPARC/GAPDH

```{r}
library(rstatix)
directory <- "~/Desktop/Gepoliano/Applebaum Lab Meeting/2022/06-07-2022/"
excel_file <- "GeneExpressionDF.xlsx"
gene_expression_df <- read_excel(paste0(directory, excel_file))

## Subset target and endogenous control for analysis
# 1) Subset target
gene_expression_df_subset <- gene_expression_df[ which(
    gene_expression_df$`Target Gene Name` == 'SPARC'  ), ]
# 2) Subset Endogenous Control
gene_expression_df_subset <- gene_expression_df[ which(
    gene_expression_df$`Endogenous Control Name` == 'GAPDH'  ), ]

# Reorder Sample, so Normoxia comes before Hypoxia in the graph
gene_expression_df_subset$Sample <- factor(
  gene_expression_df_subset$Sample, levels=c("WN_N", "WN_H", "WS_N", "WS_H"))

# Statistical Test
# stat.test is a DF with the padj and the Stars
stat.test <- gene_expression_df_subset %>%
  group_by(`Cell Line`) %>%
  t_test(RQ ~ Condition) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")
stat.test

## Create Barplot
# Change the width of bars
barplot <- ggbarplot(gene_expression_df_subset, "Sample", "RQ",
          fill = "Sample", color = "Sample",
          ylab = "SPARC/GAPDH RQ",
          palette = c("#0412d4", "#f71105", "#0412d4", "#f71105"))

# Add 10% spaces between the p-value labels and the plot border
barplot <- barplot + 
  #stat_pvalue_manual(
  #stat.test,  
  #label = "p.adj.signif",
  #tip.length = 0,
  #y.position = c(12, 52)
  #)+
  #scale_y_continuous(stat.test, expand = expansion(mult = c(0, 0.1)))+
  geom_bracket(
    xmin = c("WN_N", "WN_N", "WS_N" ), # xmin matches xmax: WN_N goes w/ WN_H
    xmax = c("WN_H", "WS_N", "WS_H"), 
    y.position = c(50, 420, 450),      # Same order as above
    label = c("***", "****", "******")     # Same order as above
  ) 

barplot
```

# 11.3) Plot all known genes

# 11.3.1) SPARC/ACTN and SPARC/GAPDH

```{r}
library(rstatix)
library(readxl)
library(ggpubr)

directory <- "~/Desktop/Gepoliano/Applebaum Lab Meeting/2022/06-07-2022/"
excel_file <- "GeneExpressionDF.xlsx"
gene_expression_df <- read_excel(paste0(directory, excel_file))

###########################################################
###########################################################
################################################### A) ACTN
###########################################################
###########################################################

## Subset target and endogenous control for analysis
# 1) Subset target
gene_expression_df_subset <- gene_expression_df[ which(
    gene_expression_df$`Target Gene Name` == 'SPARC'  ), ]
# 2) Subset Endogenous Control
gene_expression_df_subset <- gene_expression_df_subset[ which(
    gene_expression_df_subset$`Endogenous Control Name` == 'ACTN'  ), ]

# Reorder Sample, so Normoxia comes before Hypoxia in the graph
gene_expression_df_subset$Sample <- factor(
  gene_expression_df_subset$Sample, levels=c("WN_N", "WN_H", "WS_N", "WS_H"))

#### Normoxia-Hypoxia Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.normoxia_hypoxia.actn <- gene_expression_df_subset %>%
  group_by(`Cell Line`) %>%
  t_test(RQ ~ Condition) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")

#### WN/WS Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.wn_ws.actn <- gene_expression_df_subset %>%
  group_by(`Condition`) %>%
  t_test(RQ ~ `Sample`) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")
stat.test.wn_ws.actn

## Create Barplot
# Change the width of bars
barplot <- ggbarplot(gene_expression_df_subset, "Sample", "RQ",
          fill = "Sample", color = "Sample",
          add = "mean_sd",
          ylab = "SPARC/ACTN RQ",
          palette = c("#0412d4", "#f71105", "#0412d4", "#f71105"))

# Add 10% spaces between the p-value labels and the plot border
barplot <- barplot + 
  geom_bracket(
    xmin = c("WN_N", "WN_N", "WN_H", "WS_N"), # xmin matches xmax: WN_N goes w/ WN_H
    xmax = c("WN_H", "WS_N", "WS_H", "WS_H"), 
    y.position = c(10, 20, 25, 30),          # Same order as above
    label = c("***", "***", "**", "NS")         # Same order as above
  ) 
barplot

###########################################################
###########################################################
################################################## B) GAPDH
###########################################################
###########################################################

## Subset target and endogenous control for analysis
# 1) Subset target
gene_expression_df_subset <- gene_expression_df[ which(
    gene_expression_df$`Target Gene Name` == 'SPARC'  ), ]
# 2) Subset Endogenous Control
gene_expression_df_subset <- gene_expression_df_subset[ which(
    gene_expression_df_subset$`Endogenous Control Name` == 'GAPDH'  ), ]

# Reorder Sample, so Normoxia comes before Hypoxia in the graph
gene_expression_df_subset$Sample <- factor(
  gene_expression_df_subset$Sample, levels=c("WN_N", "WN_H", "WS_N", "WS_H"))

#### Normoxia-Hypoxia Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.normoxia_hypoxia.gapdh <- gene_expression_df_subset %>%
  group_by(`Cell Line`) %>%
  t_test(RQ ~ Condition) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")
stat.test.normoxia_hypoxia.gapdh

#### WN/WS Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.wn_ws.gapdh <- gene_expression_df_subset %>%
  group_by(`Condition`) %>%
  t_test(RQ ~ `Sample`) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")
stat.test.wn_ws.gapdh

## Create Barplot
# Change the width of bars
barplot <- ggbarplot(gene_expression_df_subset, "Sample", "RQ",
          fill = "Sample", color = "Sample",
          add = "mean_sd",
          ylab = "SPARC/GAPDH RQ",
          palette = c("#0412d4", "#f71105", "#0412d4", "#f71105"))

# Add 10% spaces between the p-value labels and the plot border
barplot <- barplot + 
  geom_bracket(
    xmin = c("WN_N", "WN_N", "WN_H", "WS_N"), # xmin matches xmax: WN_N goes w/ WN_H
    xmax = c("WN_H", "WS_N", "WS_H", "WS_H"), 
     y.position = c(10, 77, 87, 97),        # Same order as above
    label = c("**", "***", "**", "**")         # Same order as above
  ) 
barplot
```

# 11.3.2) PHOX2B/ACTN and PHOX2B/GAPDH

```{r}
library(rstatix)
library(readxl)
library(ggpubr)

directory <- "~/Desktop/Gepoliano/Applebaum Lab Meeting/2022/06-07-2022/"
excel_file <- "GeneExpressionDF.xlsx"
gene_expression_df <- read_excel(paste0(directory, excel_file))

###########################################################
###########################################################
################################################### A) ACTN
###########################################################
###########################################################

## Subset target and endogenous control for analysis
# 1) Subset target
gene_expression_df_subset <- gene_expression_df[ which(
    gene_expression_df$`Target Gene Name` == 'PHOX2B'  ), ]
# 2) Subset Endogenous Control
gene_expression_df_subset <- gene_expression_df_subset[ which(
    gene_expression_df_subset$`Endogenous Control Name` == 'ACTN'  ), ]

# Reorder Sample, so Normoxia comes before Hypoxia in the graph
gene_expression_df_subset$Sample <- factor(
  gene_expression_df_subset$Sample, levels=c("WN_N", "WN_H", "WS_N", "WS_H"))

#### Normoxia-Hypoxia Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.normoxia_hypoxia.actn <- gene_expression_df_subset %>%
  group_by(`Cell Line`) %>%
  t_test(RQ ~ Condition) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")

#### WN/WS Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.wn_ws.actn <- gene_expression_df_subset %>%
  group_by(`Condition`) %>%
  t_test(RQ ~ `Sample`) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")
stat.test.wn_ws.actn

## Create Barplot
# Change the width of bars
barplot <- ggbarplot(gene_expression_df_subset, "Sample", "RQ",
          fill = "Sample", color = "Sample",
          add = "mean_sd",
          ylab = "PHOX2B/ACTN RQ",
          palette = c("#0412d4", "#f71105", "#0412d4", "#f71105"))

# Add 10% spaces between the p-value labels and the plot border
barplot <- barplot + 
  geom_bracket(
    xmin = c("WN_N", "WN_N", "WN_H", "WS_N"), # xmin matches xmax: WN_N goes w/ WN_H
    xmax = c("WN_H", "WS_N", "WS_H", "WS_H"), 
    y.position = c(2, 3, 4, 5),          # Same order as above
    label = c("***", "***", "**", "NS")         # Same order as above
  ) 
barplot

###########################################################
###########################################################
################################################## B) GAPDH
###########################################################
###########################################################

## Subset target and endogenous control for analysis
# 1) Subset target
gene_expression_df_subset <- gene_expression_df[ which(
    gene_expression_df$`Target Gene Name` == 'PHOX2B'  ), ]
# 2) Subset Endogenous Control
gene_expression_df_subset <- gene_expression_df_subset[ which(
    gene_expression_df_subset$`Endogenous Control Name` == 'GAPDH'  ), ]

# Reorder Sample, so Normoxia comes before Hypoxia in the graph
gene_expression_df_subset$Sample <- factor(
  gene_expression_df_subset$Sample, levels=c("WN_N", "WN_H", "WS_N", "WS_H"))

#### Normoxia-Hypoxia Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.normoxia_hypoxia.gapdh <- gene_expression_df_subset %>%
  group_by(`Cell Line`) %>%
  t_test(RQ ~ Condition) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")
stat.test.normoxia_hypoxia.gapdh

#### WN/WS Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.wn_ws.gapdh <- gene_expression_df_subset %>%
  group_by(`Condition`) %>%
  t_test(RQ ~ `Sample`) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")
stat.test.wn_ws.gapdh

## Create Barplot
# Change the width of bars
barplot <- ggbarplot(gene_expression_df_subset, "Sample", "RQ",
          fill = "Sample", color = "Sample",
          add = "mean_sd",
          ylab = "PHOX2B/GAPDH RQ",
          palette = c("#0412d4", "#f71105", "#0412d4", "#f71105"))

# Add 10% spaces between the p-value labels and the plot border
barplot <- barplot + 
  geom_bracket(
    xmin = c("WN_N", "WN_N", "WN_H", "WS_N"), # xmin matches xmax: WN_N goes w/ WN_H
    xmax = c("WN_H", "WS_N", "WS_H", "WS_H"), 
     y.position = c(3, 5, 7, 9),        # Same order as above
    label = c("**", "***", "**", "**")         # Same order as above
  ) 
barplot
```

# 11.3.3) ASCL1/ACTN and ASCL1/GAPDH

```{r}
library(rstatix)
library(readxl)
library(ggpubr)

directory <- "~/Desktop/Gepoliano/Applebaum Lab Meeting/2022/06-07-2022/"
excel_file <- "GeneExpressionDF.xlsx"
gene_expression_df <- read_excel(paste0(directory, excel_file))

###########################################################
###########################################################
################################################### A) ACTN
###########################################################
###########################################################

## Subset target and endogenous control for analysis
# 1) Subset target
gene_expression_df_subset <- gene_expression_df[ which(
    gene_expression_df$`Target Gene Name` == 'ASCL1'  ), ]
# 2) Subset Endogenous Control
gene_expression_df_subset <- gene_expression_df_subset[ which(
    gene_expression_df_subset$`Endogenous Control Name` == 'ACTN'  ), ]

# Reorder Sample, so Normoxia comes before Hypoxia in the graph
gene_expression_df_subset$Sample <- factor(
  gene_expression_df_subset$Sample, levels=c("WN_N", "WN_H", "WS_N", "WS_H"))

#### Normoxia-Hypoxia Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.normoxia_hypoxia.actn <- gene_expression_df_subset %>%
  group_by(`Cell Line`) %>%
  t_test(RQ ~ Condition) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")

#### WN/WS Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.wn_ws.actn <- gene_expression_df_subset %>%
  group_by(`Condition`) %>%
  t_test(RQ ~ `Sample`) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")
stat.test.wn_ws.actn

## Create Barplot
# Change the width of bars
barplot <- ggbarplot(gene_expression_df_subset, "Sample", "RQ",
          fill = "Sample", color = "Sample",
          add = "mean_sd",
          ylab = "ASCL1/ACTN RQ",
          palette = c("#0412d4", "#f71105", "#0412d4", "#f71105"))

# Add 10% spaces between the p-value labels and the plot border
barplot <- barplot + 
  geom_bracket(
    xmin = c("WN_N", "WN_N", "WN_H", "WS_N"), # xmin matches xmax: WN_N goes w/ WN_H
    xmax = c("WN_H", "WS_N", "WS_H", "WS_H"), 
    y.position = c(3.5, 3, 2.5, 0.5),          # Same order as above
    label = c("***", "****", "****", "NS")         # Same order as above
  ) 
barplot

###########################################################
###########################################################
################################################## B) GAPDH
###########################################################
###########################################################

## Subset target and endogenous control for analysis
# 1) Subset target
gene_expression_df_subset <- gene_expression_df[ which(
    gene_expression_df$`Target Gene Name` == 'ASCL1'  ), ]
# 2) Subset Endogenous Control
gene_expression_df_subset <- gene_expression_df_subset[ which(
    gene_expression_df_subset$`Endogenous Control Name` == 'GAPDH'  ), ]

# Reorder Sample, so Normoxia comes before Hypoxia in the graph
gene_expression_df_subset$Sample <- factor(
  gene_expression_df_subset$Sample, levels=c("WN_N", "WN_H", "WS_N", "WS_H"))

#### Normoxia-Hypoxia Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.normoxia_hypoxia.gapdh <- gene_expression_df_subset %>%
  group_by(`Cell Line`) %>%
  t_test(RQ ~ Condition) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")
stat.test.normoxia_hypoxia.gapdh

#### WN/WS Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.wn_ws.gapdh <- gene_expression_df_subset %>%
  group_by(`Condition`) %>%
  t_test(RQ ~ `Sample`) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")
stat.test.wn_ws.gapdh

## Create Barplot
# Change the width of bars
barplot <- ggbarplot(gene_expression_df_subset, "Sample", "RQ",
          fill = "Sample", color = "Sample",
          add = "mean_sd",
          ylab = "ASCL1/GAPDH RQ",
          palette = c("#0412d4", "#f71105", "#0412d4", "#f71105"))

# Add 10% spaces between the p-value labels and the plot border
barplot <- barplot + 
  geom_bracket(
    xmin = c("WN_N", "WN_N", "WN_H", "WS_N"), # xmin matches xmax: WN_N goes w/ WN_H
    xmax = c("WN_H", "WS_N", "WS_H", "WS_H"), 
     y.position = c(3.3, 2.7, 2.2, 0.5),        # Same order as above
    label = c("**", "****", "***", "*")         # Same order as above
  ) 
barplot
```


# 11.3.4) STC1/ACTN and STC1/GAPDH

```{r}
library(rstatix)
library(readxl)
library(ggpubr)

directory <- "~/Desktop/Gepoliano/Applebaum Lab Meeting/2022/06-07-2022/"
excel_file <- "GeneExpressionDF.xlsx"
gene_expression_df <- read_excel(paste0(directory, excel_file))

###########################################################
###########################################################
################################################### A) ACTN
###########################################################
###########################################################

## Subset target and endogenous control for analysis
# 1) Subset target
gene_expression_df_subset <- gene_expression_df[ which(
    gene_expression_df$`Target Gene Name` == 'STC1'  ), ]
# 2) Subset Endogenous Control
gene_expression_df_subset <- gene_expression_df_subset[ which(
    gene_expression_df_subset$`Endogenous Control Name` == 'ACTN'  ), ]

# Reorder Sample, so Normoxia comes before Hypoxia in the graph
gene_expression_df_subset$Sample <- factor(
  gene_expression_df_subset$Sample, levels=c("WN_N", "WN_H", "WS_N", "WS_H"))

#### Normoxia-Hypoxia Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.normoxia_hypoxia.actn <- gene_expression_df_subset %>%
  group_by(`Cell Line`) %>%
  t_test(RQ ~ Condition) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")

#### WN/WS Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.wn_ws.actn <- gene_expression_df_subset %>%
  group_by(`Condition`) %>%
  t_test(RQ ~ `Sample`) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")
stat.test.wn_ws.actn

## Create Barplot
# Change the width of bars
barplot <- ggbarplot(gene_expression_df_subset, "Sample", "RQ",
          fill = "Sample", color = "Sample",
          add = "mean_sd",
          ylab = "STC1/ACTN RQ",
          palette = c("#0412d4", "#f71105", "#0412d4", "#f71105"))

# Add 10% spaces between the p-value labels and the plot border
barplot <- barplot + 
  geom_bracket(
    xmin = c("WN_N", "WN_N", "WN_H", "WS_N"), # xmin matches xmax: WN_N goes w/ WN_H
    xmax = c("WN_H", "WS_N", "WS_H", "WS_H"), 
    y.position = c(220, 310, 400, 490),          # Same order as above
    label = c("**", "NS", "NS", "NS")         # Same order as above
  ) 
barplot

###########################################################
###########################################################
################################################## B) GAPDH
###########################################################
###########################################################

## Subset target and endogenous control for analysis
# 1) Subset target
gene_expression_df_subset <- gene_expression_df[ which(
    gene_expression_df$`Target Gene Name` == 'STC1'  ), ]
# 2) Subset Endogenous Control
gene_expression_df_subset <- gene_expression_df_subset[ which(
    gene_expression_df_subset$`Endogenous Control Name` == 'GAPDH'  ), ]

# Reorder Sample, so Normoxia comes before Hypoxia in the graph
gene_expression_df_subset$Sample <- factor(
  gene_expression_df_subset$Sample, levels=c("WN_N", "WN_H", "WS_N", "WS_H"))

#### Normoxia-Hypoxia Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.normoxia_hypoxia.gapdh <- gene_expression_df_subset %>%
  group_by(`Cell Line`) %>%
  t_test(RQ ~ Condition) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")
stat.test.normoxia_hypoxia.gapdh

#### WN/WS Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.wn_ws.gapdh <- gene_expression_df_subset %>%
  group_by(`Condition`) %>%
  t_test(RQ ~ `Sample`) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")
stat.test.wn_ws.gapdh

## Create Barplot
# Change the width of bars
barplot <- ggbarplot(gene_expression_df_subset, "Sample", "RQ",
          fill = "Sample", color = "Sample",
          add = "mean_sd",
          ylab = "STC1/GAPDH RQ",
          palette = c("#0412d4", "#f71105", "#0412d4", "#f71105"))

# Add 10% spaces between the p-value labels and the plot border
barplot <- barplot + 
  geom_bracket(
    xmin = c("WN_N", "WN_N", "WN_H", "WS_N"), # xmin matches xmax: WN_N goes w/ WN_H
    xmax = c("WN_H", "WS_N", "WS_H", "WS_H"), 
     y.position = c(300, 550, 800, 1000),        # Same order as above
    label = c("**", "NS", "NS", "**")         # Same order as above
  ) 
barplot
```

# 11.3.5) CD274/ACTN and CD274/GAPDH

```{r}
library(rstatix)
library(readxl)
library(ggpubr)

directory <- "~/Desktop/Gepoliano/Applebaum Lab Meeting/2022/06-07-2022/"
excel_file <- "GeneExpressionDF.xlsx"
gene_expression_df <- read_excel(paste0(directory, excel_file))

###########################################################
###########################################################
################################################### A) ACTN
###########################################################
###########################################################

## Subset target and endogenous control for analysis
# 1) Subset target
gene_expression_df_subset <- gene_expression_df[ which(
    gene_expression_df$`Target Gene Name` == 'CD274'  ), ]
# 2) Subset Endogenous Control
gene_expression_df_subset <- gene_expression_df_subset[ which(
    gene_expression_df_subset$`Endogenous Control Name` == 'ACTN'  ), ]

# Reorder Sample, so Normoxia comes before Hypoxia in the graph
gene_expression_df_subset$Sample <- factor(
  gene_expression_df_subset$Sample, levels=c("WN_N", "WN_H", "WS_N", "WS_H"))

#### Normoxia-Hypoxia Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.normoxia_hypoxia.actn <- gene_expression_df_subset %>%
  group_by(`Cell Line`) %>%
  t_test(RQ ~ Condition) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")

#### WN/WS Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.wn_ws.actn <- gene_expression_df_subset %>%
  group_by(`Condition`) %>%
  t_test(RQ ~ `Sample`) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")
stat.test.wn_ws.actn

## Create Barplot
# Change the width of bars
barplot <- ggbarplot(gene_expression_df_subset, "Sample", "RQ",
          fill = "Sample", color = "Sample",
          add = "mean_sd",
          ylab = "CD274/ACTN RQ",
          palette = c("#0412d4", "#f71105", "#0412d4", "#f71105"))

# Add 10% spaces between the p-value labels and the plot border
barplot <- barplot + 
  geom_bracket(
    xmin = c("WN_N", "WN_N", "WN_H", "WS_N"), # xmin matches xmax: WN_N goes w/ WN_H
    xmax = c("WN_H", "WS_N", "WS_H", "WS_H"), 
    y.position = c(4, 13, 15, 17),          # Same order as above
    label = c("NS", "***", "**", "**")         # Same order as above
  ) 
barplot

###########################################################
###########################################################
################################################## B) GAPDH
###########################################################
###########################################################

## Subset target and endogenous control for analysis
# 1) Subset target
gene_expression_df_subset <- gene_expression_df[ which(
    gene_expression_df$`Target Gene Name` == 'CD274'  ), ]
# 2) Subset Endogenous Control
gene_expression_df_subset <- gene_expression_df_subset[ which(
    gene_expression_df_subset$`Endogenous Control Name` == 'GAPDH'  ), ]

# Reorder Sample, so Normoxia comes before Hypoxia in the graph
gene_expression_df_subset$Sample <- factor(
  gene_expression_df_subset$Sample, levels=c("WN_N", "WN_H", "WS_N", "WS_H"))

#### Normoxia-Hypoxia Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.normoxia_hypoxia.gapdh <- gene_expression_df_subset %>%
  group_by(`Cell Line`) %>%
  t_test(RQ ~ Condition) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")
stat.test.normoxia_hypoxia.gapdh

#### WN/WS Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.wn_ws.gapdh <- gene_expression_df_subset %>%
  group_by(`Condition`) %>%
  t_test(RQ ~ `Sample`) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")
stat.test.wn_ws.gapdh

## Create Barplot
# Change the width of bars
barplot <- ggbarplot(gene_expression_df_subset, "Sample", "RQ",
          fill = "Sample", color = "Sample",
          add = "mean_sd",
          ylab = "CD274/GAPDH RQ",
          palette = c("#0412d4", "#f71105", "#0412d4", "#f71105"))

# Add 10% spaces between the p-value labels and the plot border
barplot <- barplot + 
  geom_bracket(
    xmin = c("WN_N", "WN_N", "WN_H", "WS_N"), # xmin matches xmax: WN_N goes w/ WN_H
    xmax = c("WN_H", "WS_N", "WS_H", "WS_H"), 
     y.position = c(15, 107, 121, 135),        # Same order as above
    label = c("NS", "***", "**", "**")         # Same order as above
  ) 
barplot
```

# 11.3.6) CSF1/ACTN and CSF1/GAPDH

```{r}
library(rstatix)
library(readxl)
library(ggpubr)

directory <- "~/Desktop/Gepoliano/Applebaum Lab Meeting/2022/06-07-2022/"
excel_file <- "GeneExpressionDF.xlsx"
gene_expression_df <- read_excel(paste0(directory, excel_file))

###########################################################
###########################################################
################################################### A) ACTN
###########################################################
###########################################################

## Subset target and endogenous control for analysis
# 1) Subset target
gene_expression_df_subset <- gene_expression_df[ which(
    gene_expression_df$`Target Gene Name` == 'CSF1'  ), ]
# 2) Subset Endogenous Control
gene_expression_df_subset <- gene_expression_df_subset[ which(
    gene_expression_df_subset$`Endogenous Control Name` == 'ACTN'  ), ]

# Reorder Sample, so Normoxia comes before Hypoxia in the graph
gene_expression_df_subset$Sample <- factor(
  gene_expression_df_subset$Sample, levels=c("WN_N", "WN_H", "WS_N", "WS_H"))

#### Normoxia-Hypoxia Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.normoxia_hypoxia.actn <- gene_expression_df_subset %>%
  group_by(`Cell Line`) %>%
  t_test(RQ ~ Condition) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")

#### WN/WS Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.wn_ws.actn <- gene_expression_df_subset %>%
  group_by(`Condition`) %>%
  t_test(RQ ~ `Sample`) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")
stat.test.wn_ws.actn

## Create Barplot
# Change the width of bars
barplot <- ggbarplot(gene_expression_df_subset, "Sample", "RQ",
          fill = "Sample", color = "Sample",
          add = "mean_sd",
          ylab = "CSF1/ACTN RQ",
          palette = c("#0412d4", "#f71105", "#0412d4", "#f71105"))

# Add 10% spaces between the p-value labels and the plot border
barplot <- barplot + 
  geom_bracket(
    xmin = c("WN_N", "WN_N", "WN_H", "WS_N"), # xmin matches xmax: WN_N goes w/ WN_H
    xmax = c("WN_H", "WS_N", "WS_H", "WS_H"), 
    y.position = c(2, 5, 6, 7),          # Same order as above
    label = c("NS", "***", "NS", "**")         # Same order as above
  ) 
barplot

###########################################################
###########################################################
################################################## B) GAPDH
###########################################################
###########################################################

## Subset target and endogenous control for analysis
# 1) Subset target
gene_expression_df_subset <- gene_expression_df[ which(
    gene_expression_df$`Target Gene Name` == 'CSF1'  ), ]
# 2) Subset Endogenous Control
gene_expression_df_subset <- gene_expression_df_subset[ which(
    gene_expression_df_subset$`Endogenous Control Name` == 'GAPDH'  ), ]

# Reorder Sample, so Normoxia comes before Hypoxia in the graph
gene_expression_df_subset$Sample <- factor(
  gene_expression_df_subset$Sample, levels=c("WN_N", "WN_H", "WS_N", "WS_H"))

#### Normoxia-Hypoxia Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.normoxia_hypoxia.gapdh <- gene_expression_df_subset %>%
  group_by(`Cell Line`) %>%
  t_test(RQ ~ Condition) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")
stat.test.normoxia_hypoxia.gapdh

#### WN/WS Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.wn_ws.gapdh <- gene_expression_df_subset %>%
  group_by(`Condition`) %>%
  t_test(RQ ~ `Sample`) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")
stat.test.wn_ws.gapdh

## Create Barplot
# Change the width of bars
barplot <- ggbarplot(gene_expression_df_subset, "Sample", "RQ",
          fill = "Sample", color = "Sample",
          add = "mean_sd",
          ylab = "CSF1/GAPDH RQ",
          palette = c("#0412d4", "#f71105", "#0412d4", "#f71105"))

# Add 10% spaces between the p-value labels and the plot border
barplot <- barplot + 
  geom_bracket(
    xmin = c("WN_N", "WN_N", "WN_H", "WS_N"), # xmin matches xmax: WN_N goes w/ WN_H
    xmax = c("WN_H", "WS_N", "WS_H", "WS_H"), 
     y.position = c(5, 25, 30, 35),        # Same order as above
    label = c("NS", "**", "NS", "*")         # Same order as above
  ) 
barplot
```

# 11.3.7) TET1/ACTN and TET1/GAPDH

```{r}
library(rstatix)
library(readxl)
library(ggpubr)

directory <- "~/Desktop/Gepoliano/Applebaum Lab Meeting/2022/06-07-2022/"
excel_file <- "GeneExpressionDF.xlsx"
gene_expression_df <- read_excel(paste0(directory, excel_file))

###########################################################
###########################################################
################################################### A) ACTN
###########################################################
###########################################################

## Subset target and endogenous control for analysis
# 1) Subset target
gene_expression_df_subset_actn <- gene_expression_df[ which(
    gene_expression_df$`Target Gene Name` == 'TET1'  ), ]
# 2) Subset Endogenous Control
gene_expression_df_subset_actn <- gene_expression_df_subset_actn[ which(
    gene_expression_df_subset_actn$`Endogenous Control Name` == 'ACTN'  ), ]

# Reorder Sample, so Normoxia comes before Hypoxia in the graph
gene_expression_df_subset$Sample <- factor(
  gene_expression_df_subset_actn$Sample, levels=c("WN_N", "WN_H", "WS_N", "WS_H"))

#### Normoxia-Hypoxia Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.normoxia_hypoxia.actn <- gene_expression_df_subset_actn %>%
  group_by(`Cell Line`) %>%
  t_test(RQ ~ Condition) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")

#### WN/WS Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.wn_ws.actn <- gene_expression_df_subset_actn %>%
  group_by(`Condition`) %>%
  t_test(RQ ~ `Sample`) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")
stat.test.wn_ws.actn

## Create Barplot
# Change the width of bars
barplot <- ggbarplot(gene_expression_df_subset_actn, "Sample", "RQ",
          fill = "Sample", color = "Sample",
          add = "mean_sd",
          ylab = "TET1/ACTN RQ",
          palette = c("#0412d4", "#f71105", "#0412d4", "#f71105"))

# Add 10% spaces between the p-value labels and the plot border
barplot <- barplot + 
  geom_bracket(
    xmin = c("WN_N", "WN_N", "WN_H", "WS_N"), # xmin matches xmax: WN_N goes w/ WN_H
    xmax = c("WN_H", "WS_N", "WS_H", "WS_H"), 
    y.position = c(4, 5, 6, 7),          # Same order as above
    label = c("**", "**", "NS", "**")         # Same order as above
  ) 
barplot

###########################################################
###########################################################
################################################## B) GAPDH
###########################################################
###########################################################

## Subset target and endogenous control for analysis
# 1) Subset target
gene_expression_df_subset_gapdh <- gene_expression_df[ which(
    gene_expression_df$`Target Gene Name` == 'TET1'  ), ]
# 2) Subset Endogenous Control
gene_expression_df_subset_gapdh <- gene_expression_df_subset_gapdh[ which(
    gene_expression_df_subset_gapdh$`Endogenous Control Name` == 'GAPDH'  ), ]

# Reorder Sample, so Normoxia comes before Hypoxia in the graph
gene_expression_df_subset_gapdh$Sample <- factor(
  gene_expression_df_subset_gapdh$Sample, levels=c("WN_N", "WN_H", "WS_N", "WS_H"))

#### Normoxia-Hypoxia Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.normoxia_hypoxia.gapdh <- gene_expression_df_subset_gapdh %>%
  group_by(`Cell Line`) %>%
  t_test(RQ ~ Condition) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")
stat.test.normoxia_hypoxia.gapdh

#### WN/WS Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.wn_ws.gapdh <- gene_expression_df_subset_gapdh %>%
  group_by(`Condition`) %>%
  t_test(RQ ~ `Sample`) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")
stat.test.wn_ws.gapdh

## Create Barplot
# Change the width of bars
barplot <- ggbarplot(gene_expression_df_subset_gapdh, "Sample", "RQ",
          fill = "Sample", color = "Sample",
          add = "mean_sd",
          ylab = "TET1/GAPDH RQ",
          palette = c("#0412d4", "#f71105", "#0412d4", "#f71105"))

# Add 10% spaces between the p-value labels and the plot border
barplot <- barplot + 
  geom_bracket(
    xmin = c("WN_N", "WN_N", "WN_H", "WS_N"), # xmin matches xmax: WN_N goes w/ WN_H
    xmax = c("WN_H", "WS_N", "WS_H", "WS_H"), 
     y.position = c(5, 25, 30, 35),        # Same order as above
    label = c("NS", "**", "NS", "*")         # Same order as above
  ) 
barplot
```

# 11.3.8) HIF1/ACTN and HIF1/GAPDH

```{r}
library(rstatix)
library(readxl)
library(ggpubr)

directory <- "~/Desktop/Gepoliano/Applebaum Lab Meeting/2022/06-07-2022/"
excel_file <- "GeneExpressionDF.xlsx"
gene_expression_df <- read_excel(paste0(directory, excel_file))

###########################################################
###########################################################
################################################### A) ACTN
###########################################################
###########################################################

## Subset target and endogenous control for analysis
# 1) Subset target
gene_expression_df_subset_actn <- gene_expression_df[ which(
    gene_expression_df$`Target Gene Name` == 'HIF1A'  ), ]
# 2) Subset Endogenous Control
gene_expression_df_subset_actn <- gene_expression_df_subset_actn[ which(
    gene_expression_df_subset_actn$`Endogenous Control Name` == 'ACTN'  ), ]

# Reorder Sample, so Normoxia comes before Hypoxia in the graph
gene_expression_df_subset$Sample <- factor(
  gene_expression_df_subset_actn$Sample, levels=c("WN_N", "WN_H", "WS_N", "WS_H"))

#### Normoxia-Hypoxia Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.normoxia_hypoxia.actn <- gene_expression_df_subset_actn %>%
  group_by(`Cell Line`) %>%
  t_test(RQ ~ Condition) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")

#### WN/WS Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.wn_ws.actn <- gene_expression_df_subset_actn %>%
  group_by(`Condition`) %>%
  t_test(RQ ~ `Sample`) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")
stat.test.wn_ws.actn

## Create Barplot
# Change the width of bars
barplot <- ggbarplot(gene_expression_df_subset_actn, "Sample", "RQ",
          fill = "Sample", color = "Sample",
          add = "mean_sd",
          ylab = "HIF1A/ACTN RQ",
          palette = c("#0412d4", "#f71105", "#0412d4", "#f71105"))

# Add 10% spaces between the p-value labels and the plot border
barplot <- barplot + 
  geom_bracket(
    xmin = c("WN_N", "WN_N", "WN_H", "WS_N"), # xmin matches xmax: WN_N goes w/ WN_H
    xmax = c("WN_H", "WS_N", "WS_H", "WS_H"), 
    y.position = c(4, 5, 7, 8),          # Same order as above
    label = c("NS", "***", "*", "*")         # Same order as above
  ) 
barplot

###########################################################
###########################################################
################################################## B) GAPDH
###########################################################
###########################################################

## Subset target and endogenous control for analysis
# 1) Subset target
gene_expression_df_subset_gapdh <- gene_expression_df[ which(
    gene_expression_df$`Target Gene Name` == 'HIF1A'  ), ]
# 2) Subset Endogenous Control
gene_expression_df_subset_gapdh <- gene_expression_df_subset_gapdh[ which(
    gene_expression_df_subset_gapdh$`Endogenous Control Name` == 'GAPDH'  ), ]

# Reorder Sample, so Normoxia comes before Hypoxia in the graph
gene_expression_df_subset_gapdh$Sample <- factor(
  gene_expression_df_subset_gapdh$Sample, levels=c("WN_N", "WN_H", "WS_N", "WS_H"))

#### Normoxia-Hypoxia Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.normoxia_hypoxia.gapdh <- gene_expression_df_subset_gapdh %>%
  group_by(`Cell Line`) %>%
  t_test(RQ ~ Condition) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")
stat.test.normoxia_hypoxia.gapdh

#### WN/WS Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.wn_ws.gapdh <- gene_expression_df_subset_gapdh %>%
  group_by(`Condition`) %>%
  t_test(RQ ~ `Sample`) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")
stat.test.wn_ws.gapdh

## Create Barplot
# Change the width of bars
barplot <- ggbarplot(gene_expression_df_subset_gapdh, "Sample", "RQ",
          fill = "Sample", color = "Sample",
          add = "mean_sd",
          ylab = "HIF1A/GAPDH RQ",
          palette = c("#0412d4", "#f71105", "#0412d4", "#f71105"))

# Add 10% spaces between the p-value labels and the plot border
barplot <- barplot + 
  geom_bracket(
    xmin = c("WN_N", "WN_N", "WN_H", "WS_N"), # xmin matches xmax: WN_N goes w/ WN_H
    xmax = c("WN_H", "WS_N", "WS_H", "WS_H"), 
     y.position = c(5, 25, 33, 37),        # Same order as above
    label = c("NS", "**", "***", "NS")         # Same order as above
  ) 
barplot
```

# 12) Include 96h Hypoxia Plots

# 12.1) SPARC

```{r}
library(rstatix)
library(readxl)
library(ggpubr)

directory <- "~/Desktop/Gepoliano/Hypoxia/Time Course/Results/"
excel_file <- "GeneExpressionDF.xlsx"
gene_expression_df <- read_excel(paste0(directory, excel_file))

# Subset Time Point for both ACTN and GAPDH
gene_expression_df <- gene_expression_df[ which(
    gene_expression_df$`Hours` == '96h'  ), ]

###########################################################
###########################################################
################################################### A) ACTN
###########################################################
###########################################################

## Subset target and endogenous control for analysis
# 1) Subset target
gene_expression_df_subset_actn <- gene_expression_df[ which(
    gene_expression_df$`Target Gene Name` == 'SPARC'  ), ]
# 2) Subset Endogenous Control
gene_expression_df_subset_actn <- gene_expression_df_subset_actn[ which(
    gene_expression_df_subset_actn$`Endogenous Control Name` == 'ACTN'  ), ]

# Reorder Sample, so Normoxia comes before Hypoxia in the graph
gene_expression_df_subset$Sample <- factor(
  gene_expression_df_subset_actn$Sample, levels=c("WN_N", "WN_H", "WS_N", "WS_H"))

#### Normoxia-Hypoxia Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.normoxia_hypoxia.actn <- gene_expression_df_subset_actn %>%
  group_by(`Cell Line`) %>%
  t_test(RQ ~ Condition) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")

#### WN/WS Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.wn_ws.actn <- gene_expression_df_subset_actn %>%
  group_by(`Condition`) %>%
  t_test(RQ ~ `Sample`) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")

## Create Barplot
# Change the width of bars
barplot <- ggbarplot(gene_expression_df_subset_actn, "Sample", "RQ",
          fill = "Sample", color = "Sample",
          add = "mean_sd",
          ylab = "SPARC/ACTN RQ",
          palette = c("#0412d4", "#f71105", "#0412d4", "#f71105"))

# Add 10% spaces between the p-value labels and the plot border
barplot <- barplot + 
  geom_bracket(
    xmin = c("WN_N", "WN_N", "WN_H", "WS_N"), # xmin matches xmax: WN_N goes w/ WN_H
    xmax = c("WN_H", "WS_N", "WS_H", "WS_H"), 
    y.position = c(3, 13, 15.5, 18),          # Same order as above
    label = c("NS", "**", "*", "NS")         # Same order as above
  ) 
barplot

###########################################################
###########################################################
################################################## B) GAPDH
###########################################################
###########################################################

## Subset target and endogenous control for analysis
# 1) Subset target
gene_expression_df_subset_gapdh <- gene_expression_df[ which(
    gene_expression_df$`Target Gene Name` == 'SPARC'  ), ]
# 2) Subset Endogenous Control
gene_expression_df_subset_gapdh <- gene_expression_df_subset_gapdh[ which(
    gene_expression_df_subset_gapdh$`Endogenous Control Name` == 'GAPDH'  ), ]

# Reorder Sample, so Normoxia comes before Hypoxia in the graph
gene_expression_df_subset_gapdh$Sample <- factor(
  gene_expression_df_subset_gapdh$Sample, levels=c("WN_N", "WN_H", "WS_N", "WS_H"))

#### Normoxia-Hypoxia Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.normoxia_hypoxia.gapdh <- gene_expression_df_subset_gapdh %>%
  group_by(`Cell Line`) %>%
  t_test(RQ ~ Condition) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")

#### WN/WS Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.wn_ws.gapdh <- gene_expression_df_subset_gapdh %>%
  group_by(`Condition`) %>%
  t_test(RQ ~ `Sample`) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")

## Create Barplot
# Change the width of bars
barplot <- ggbarplot(gene_expression_df_subset_gapdh, "Sample", "RQ",
          fill = "Sample", color = "Sample",
          add = "mean_sd",
          ylab = "SPARC/GAPDH RQ",
          palette = c("#0412d4", "#f71105", "#0412d4", "#f71105"))

# Add 10% spaces between the p-value labels and the plot border
barplot <- barplot + 
  geom_bracket(
    xmin = c("WN_N", "WN_N", "WN_H", "WS_N"), # xmin matches xmax: WN_N goes w/ WN_H
    xmax = c("WN_H", "WS_N", "WS_H", "WS_H"), 
     y.position = c(5, 20, 23, 26),        # Same order as above
    label = c("**", "**", "***", "NS")         # Same order as above
  ) 
barplot
```

# 12.1) ASCL1

```{r}
library(rstatix)
library(readxl)
library(ggpubr)

directory <- "~/Desktop/Gepoliano/Hypoxia/Time Course/Results/"
excel_file <- "GeneExpressionDF.xlsx"
gene_expression_df <- read_excel(paste0(directory, excel_file))

# Subset Time Point for both ACTN and GAPDH
gene_expression_df <- gene_expression_df[ which(
    gene_expression_df$`Hours` == '96h'  ), ]

###########################################################
###########################################################
################################################### A) ACTN
###########################################################
###########################################################

## Subset target and endogenous control for analysis
# 1) Subset target
gene_expression_df_subset_actn <- gene_expression_df[ which(
    gene_expression_df$`Target Gene Name` == 'ASCL1'  ), ]
# 2) Subset Endogenous Control
gene_expression_df_subset_actn <- gene_expression_df_subset_actn[ which(
    gene_expression_df_subset_actn$`Endogenous Control Name` == 'ACTN'  ), ]

# Reorder Sample, so Normoxia comes before Hypoxia in the graph
gene_expression_df_subset$Sample <- factor(
  gene_expression_df_subset_actn$Sample, levels=c("WN_N", "WN_H", "WS_N", "WS_H"))

#### Normoxia-Hypoxia Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.normoxia_hypoxia.actn <- gene_expression_df_subset_actn %>%
  group_by(`Cell Line`) %>%
  t_test(RQ ~ Condition) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")

#### WN/WS Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.wn_ws.actn <- gene_expression_df_subset_actn %>%
  group_by(`Condition`) %>%
  t_test(RQ ~ `Sample`) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")

## Create Barplot
# Change the width of bars
barplot <- ggbarplot(gene_expression_df_subset_actn, "Sample", "RQ",
          fill = "Sample", color = "Sample",
          add = "mean_sd",
          ylab = "ASCL1/ACTN RQ",
          palette = c("#0412d4", "#f71105", "#0412d4", "#f71105"))

# Add 10% spaces between the p-value labels and the plot border
barplot <- barplot + 
  geom_bracket(
    xmin = c("WN_N", "WN_N", "WN_H", "WS_N"), # xmin matches xmax: WN_N goes w/ WN_H
    xmax = c("WN_H", "WS_N", "WS_H", "WS_H"), 
    y.position = c(2, 3, 4, 5),          # Same order as above
    label = c("***", "**", "**", "***")         # Same order as above
  ) 
barplot

###########################################################
###########################################################
################################################## B) GAPDH
###########################################################
###########################################################

## Subset target and endogenous control for analysis
# 1) Subset target
gene_expression_df_subset_gapdh <- gene_expression_df[ which(
    gene_expression_df$`Target Gene Name` == 'ASCL1'  ), ]
# 2) Subset Endogenous Control
gene_expression_df_subset_gapdh <- gene_expression_df_subset_gapdh[ which(
    gene_expression_df_subset_gapdh$`Endogenous Control Name` == 'GAPDH'  ), ]

# Reorder Sample, so Normoxia comes before Hypoxia in the graph
gene_expression_df_subset_gapdh$Sample <- factor(
  gene_expression_df_subset_gapdh$Sample, levels=c("WN_N", "WN_H", "WS_N", "WS_H"))

#### Normoxia-Hypoxia Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.normoxia_hypoxia.gapdh <- gene_expression_df_subset_gapdh %>%
  group_by(`Cell Line`) %>%
  t_test(RQ ~ `Condition`) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")

#### WN/WS Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.wn_ws.gapdh <- gene_expression_df_subset_gapdh %>%
  group_by(`Condition`) %>%
  t_test(RQ ~ `Sample`) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")

## Create Barplot
# Change the width of bars
barplot <- ggbarplot(gene_expression_df_subset_gapdh, "Sample", "RQ",
          fill = "Sample", color = "Sample",
          add = "mean_sd",
          ylab = "ASCL1/GAPDH RQ",
          palette = c("#0412d4", "#f71105", "#0412d4", "#f71105"))

# Add 10% spaces between the p-value labels and the plot border
barplot <- barplot + 
  geom_bracket(
    xmin = c("WN_N", "WN_N", "WN_H", "WS_N"), # xmin matches xmax: WN_N goes w/ WN_H
    xmax = c("WN_H", "WS_N", "WS_H", "WS_H"), 
     y.position = c(3.5, 4.5, 5.5, 6.5),        # Same order as above
    label = c("**", "**", "**", "**")         # Same order as above
  ) 
barplot
```

# 13) Lineplot

```{r}
library(rstatix)
library(readxl)
library(ggpubr)

directory <- "~/Desktop/Gepoliano/Hypoxia/Time Course/Results/"
excel_file <- "GeneExpressionDF.xlsx"
gene_expression_df <- read_excel(paste0(directory, excel_file))

# Subset Date Seeded "20220607"
gene_expression_df <- gene_expression_df[ which(
    gene_expression_df$Date_Seeded == "20220607"), ]

# Subset Number Experiment
gene_expression_df <- gene_expression_df[ which(
  gene_expression_df$Number_Given_to_Experiment == "0" |
    gene_expression_df$Number_Given_to_Experiment == "1" |
      gene_expression_df$Number_Given_to_Experiment == "2" |
      gene_expression_df$Number_Given_to_Experiment == "3"), ]

# Subset Time Point for both ACTN and GAPDH
gene_expression_df <- gene_expression_df[ which(
    gene_expression_df$`Hours` == '48h' |
      gene_expression_df$`Hours` == '96h' |
      gene_expression_df$`Hours` == '144h'), ]

###########################################################
###########################################################
################################################### A) ACTN
###########################################################
###########################################################

gene="ASCL1"

## Subset target and endogenous control for analysis
# 1) Subset target
gene_expression_df_subset_actn <- gene_expression_df[ which(
    gene_expression_df$`Target Gene Name` == gene  ), ]
# 2) Subset Endogenous Control
gene_expression_df_subset_actn <- gene_expression_df_subset_actn[ which(
    gene_expression_df_subset_actn$`Endogenous Control Name` == 'ACTN'  ), ]

# Reorder Sample, so Normoxia comes before Hypoxia in the graph
gene_expression_df_subset_actn$Sample <- factor(
  gene_expression_df_subset_actn$Sample, levels=c("WN_N", "WN_H", "WS_N", "WS_H"))

# Reorder Hours: 48h, 96h, 144h
gene_expression_df_subset_actn$Hours <- factor(
  gene_expression_df_subset_actn$Hours, levels=c("48h", "96h", "144h"))

#### Normoxia-Hypoxia Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.normoxia_hypoxia.actn <- gene_expression_df_subset_actn %>%
  group_by(`Cell_Line`) %>%
  t_test(RQ ~ Condition) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")

#### WN/WS Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.wn_ws.actn <- gene_expression_df_subset_actn %>%
  group_by(`Condition`) %>%
  t_test(RQ ~ `Sample`) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")

## Create Barplot
# Change the width of bars
lineplot <- ggline(gene_expression_df_subset_actn, "Sample", "RQ",
          fill = "Sample", color = "Sample",
          add = "mean_sd",
          ylab = paste0(gene,"/ACTN RQ"),
          palette = c("#0412d4", "#f71105", "#0412d4", "#f71105"))+
  geom_line()+
  geom_point()

# Add 10% spaces between the p-value labels and the plot border
lineplot <- lineplot + 
  geom_bracket(
    xmin = c("WN_N", "WN_N", "WN_H", "WS_N"), # xmin matches xmax: WN_N goes w/ WN_H
    xmax = c("WN_H", "WS_N", "WS_H", "WS_H"), 
    y.position = c(2, 3, 4, 5),          # Same order as above
    label = c("***", "**", "**", "***")         # Same order as above
  ) 
lineplot

ggline(
  gene_expression_df_subset_actn, x="Hours", y="RQ", add = "mean_se",
  color = "Condition",  shape = "Condition",
  ylab = paste0(gene,"/ACTN RQ"),
  palette = c("#0412d4", "#f71105", "#0412d4", "#f71105"), 
  facet.by = "Cell_Line"
  ) 

###########################################################
###########################################################
################################################## B) GAPDH
###########################################################
###########################################################

## Subset target and endogenous control for analysis
# 1) Subset target
gene_expression_df_subset_gapdh <- gene_expression_df[ which(
    gene_expression_df$`Target Gene Name` == 'ASCL1'  ), ]
# 2) Subset Endogenous Control
gene_expression_df_subset_gapdh <- gene_expression_df_subset_gapdh[ which(
    gene_expression_df_subset_gapdh$`Endogenous Control Name` == 'GAPDH'  ), ]

# Reorder Sample, so Normoxia comes before Hypoxia in the graph
gene_expression_df_subset_gapdh$Sample <- factor(
  gene_expression_df_subset_gapdh$Sample, levels=c("WN_N", "WN_H", "WS_N", "WS_H"))

#### Normoxia-Hypoxia Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.normoxia_hypoxia.gapdh <- gene_expression_df_subset_gapdh %>%
  group_by(`Cell_Line`) %>%
  t_test(RQ ~ `Condition`) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")

#### WN/WS Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.wn_ws.gapdh <- gene_expression_df_subset_gapdh %>%
  group_by(`Condition`) %>%
  t_test(RQ ~ `Sample`) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")

## Create Barplot
# Change the width of bars
barplot <- ggbarplot(gene_expression_df_subset_gapdh, "Sample", "RQ",
          fill = "Sample", color = "Sample",
          add = "mean_sd",
          ylab = "ASCL1/GAPDH RQ",
          palette = c("#0412d4", "#f71105", "#0412d4", "#f71105"))

# Add 10% spaces between the p-value labels and the plot border
barplot <- barplot + 
  geom_bracket(
    xmin = c("WN_N", "WN_N", "WN_H", "WS_N"), # xmin matches xmax: WN_N goes w/ WN_H
    xmax = c("WN_H", "WS_N", "WS_H", "WS_H"), 
     y.position = c(3.5, 4.5, 5.5, 6.5),        # Same order as above
    label = c("**", "**", "**", "**")         # Same order as above
  ) 
barplot

ggline(
  gene_expression_df_subset_gapdh, x="Hours", y="RQ", add = "mean_se",
  color = "Condition",  shape = "Condition",
  ylab = paste0(gene,"/GAPDH RQ"),
  palette = c("#0412d4", "#f71105", "#0412d4", "#f71105"), 
  facet.by = "Cell_Line"
  ) 

ggline(
  gene_expression_df_subset_gapdh, 
  x="Hours", y="RQ", 
  add = "mean_se",
  color = "Condition",  
  shape = "Condition",
  ylab = paste0(gene,"/GAPDH RQ"),
  palette = c("#0412d4", "#f71105", "#0412d4", "#f71105")
  ) 


ggline(gene_expression_df_subset_gapdh, "Hours", "RQ",
   linetype = "Sample", 
   shape = "Sample",
   color = "Sample", 
   palette = c("#0412d4", "#f71105", "#0412d4", "#f71105"))
```

# 14) Plot Genes Individually

# 14.1) SPARC

```{r}
library(rstatix)
library(readxl)
library(ggpubr)

gene="SPARC"

directory <- "~/Desktop/Gepoliano/Hypoxia/Time Course/Results/"
excel_file <- "GeneExpressionDF.xlsx"
gene_expression_df <- read_excel(paste0(directory, excel_file))

# Subset Date Seeded "20220607"
gene_expression_df <- gene_expression_df[ which(
    gene_expression_df$Date_Seeded == "20220607"), ]

# Subset Number Experiment
gene_expression_df <- gene_expression_df[ which(
  gene_expression_df$Number_Given_to_Experiment == "0" |
    gene_expression_df$Number_Given_to_Experiment == "1" |
      gene_expression_df$Number_Given_to_Experiment == "2" |
      gene_expression_df$Number_Given_to_Experiment == "3"), ]

# Subset Time Point for both ACTN and GAPDH
gene_expression_df <- gene_expression_df[ which(
    gene_expression_df$`Hours` == '48h' |
      gene_expression_df$`Hours` == '96h' |
      gene_expression_df$`Hours` == '144h'), ]

###########################################################
###########################################################
################################################### A) ACTN
###########################################################
###########################################################

## Subset target and endogenous control for analysis
# 1) Subset target
gene_expression_df_subset_actn <- gene_expression_df[ which(
    gene_expression_df$`Target Gene Name` == gene  ), ]
# 2) Subset Endogenous Control
gene_expression_df_subset_actn <- gene_expression_df_subset_actn[ which(
    gene_expression_df_subset_actn$`Endogenous Control Name` == 'ACTN'  ), ]

# Reorder Sample, so Normoxia comes before Hypoxia in the graph
gene_expression_df_subset_actn$Sample <- factor(
  gene_expression_df_subset_actn$Sample, levels=c("WN_N", "WN_H", "WS_N", "WS_H"))

# Reorder Hours
gene_expression_df_subset_actn$Hours <- factor(
  gene_expression_df_subset_actn$Hours, levels=c("48h", "96h", "144h"))

#### Normoxia-Hypoxia Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.normoxia_hypoxia.actn <- gene_expression_df_subset_actn %>%
  group_by(`Cell_Line`) %>%
  t_test(RQ ~ Condition) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")

#### WN/WS Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.wn_ws.actn <- gene_expression_df_subset_actn %>%
  group_by(`Condition`) %>%
  t_test(RQ ~ `Sample`) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")

## Create Barplot
# Change the width of bars
barplot <- ggbarplot(gene_expression_df_subset_actn, 
                     x = "Hours", y = "RQ", 
                     fill = "Sample",
                     color = "Sample",
                     add = "mean_sd", 
                     ylab = paste0(gene, "/ACTN RQ"),
                     palette = c("#0412d4", "#f71105", "#0412d4", "#f71105"),
                     position = position_dodge())

# Add 10% spaces between the p-value labels and the plot border
barplot <- barplot + 
  geom_bracket(
    xmin = c("48h", "96h", "48h"), # xmin matches xmax: WN_N goes w/ WN_H
    xmax = c("96h", "144h", "144h"), 
    y.position = c(57, 67, 77),          # Same order as above
    label = c("***", "**", "**")         # Same order as above
  ) +
  scale_y_continuous(expand = expansion(mult = c(0, 0.1)))
barplot

###########################################################
###########################################################
################################################## B) GAPDH
###########################################################
###########################################################

## Subset target and endogenous control for analysis
# 1) Subset target
gene_expression_df_subset_gapdh <- gene_expression_df[ which(
    gene_expression_df$`Target Gene Name` == gene  ), ]
# 2) Subset Endogenous Control
gene_expression_df_subset_gapdh <- gene_expression_df_subset_gapdh[ which(
    gene_expression_df_subset_gapdh$`Endogenous Control Name` == 'GAPDH'  ), ]

# Reorder Sample, so Normoxia comes before Hypoxia in the graph
gene_expression_df_subset_gapdh$Sample <- factor(
  gene_expression_df_subset_gapdh$Sample, levels=c("WN_N", "WN_H", "WS_N", "WS_H"))

# Reorder Hours
gene_expression_df_subset_gapdh$Hours <- factor(
  gene_expression_df_subset_gapdh$Hours, levels=c("48h", "96h", "144h"))

#### Normoxia-Hypoxia Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.normoxia_hypoxia.gapdh <- gene_expression_df_subset_gapdh %>%
  group_by(`Cell_Line`) %>%
  t_test(RQ ~ `Condition`) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")

#### WN/WS Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.wn_ws.gapdh <- gene_expression_df_subset_gapdh %>%
  group_by(`Condition`) %>%
  t_test(RQ ~ `Sample`) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")

## Create Barplot
# Change the width of bars
barplot <- ggbarplot(gene_expression_df_subset_gapdh, 
                     x = "Hours", y = "RQ", 
                     fill = "Sample",
                     color = "Sample",
                     add = "mean_sd", 
                     ylab = paste0(gene, "/GAPDH RQ"),
                     palette = c("#0412d4", "#f71105", "#0412d4", "#f71105"),
                     position = position_dodge())

# Add 10% spaces between the p-value labels and the plot border
barplot <- barplot + 
  geom_bracket(
    xmin = c("48h", "96h", "48h"), # xmin matches xmax: WN_N goes w/ WN_H
    xmax = c("96h", "144h", "144h"), 
    y.position = c(95, 105, 115),          # Same order as above
    label = c("***", "**", "**")         # Same order as above
  ) +
  scale_y_continuous(expand = expansion(mult = c(0, 0.1)))
barplot

```

# 14.2) STC1

```{r}
library(rstatix)
library(readxl)
library(ggpubr)

gene="STC1"

directory <- "~/Desktop/Gepoliano/Hypoxia/Time Course/Results/"
excel_file <- "GeneExpressionDF.xlsx"
gene_expression_df <- read_excel(paste0(directory, excel_file))

# Subset Date Seeded "20220607"
gene_expression_df <- gene_expression_df[ which(
    gene_expression_df$Date_Seeded == "20220607"), ]

# Subset Number Experiment
gene_expression_df <- gene_expression_df[ which(
  gene_expression_df$Number_Given_to_Experiment == "0" |
    gene_expression_df$Number_Given_to_Experiment == "1" |
      gene_expression_df$Number_Given_to_Experiment == "2" |
      gene_expression_df$Number_Given_to_Experiment == "3"), ]

# Subset Time Point for both ACTN and GAPDH
gene_expression_df <- gene_expression_df[ which(
    gene_expression_df$`Hours` == '48h' |
      gene_expression_df$`Hours` == '96h' |
      gene_expression_df$`Hours` == '144h'), ]

###########################################################
###########################################################
################################################### A) ACTN
###########################################################
###########################################################

## Subset target and endogenous control for analysis
# 1) Subset target
gene_expression_df_subset_actn <- gene_expression_df[ which(
    gene_expression_df$`Target Gene Name` == gene  ), ]
# 2) Subset Endogenous Control
gene_expression_df_subset_actn <- gene_expression_df_subset_actn[ which(
    gene_expression_df_subset_actn$`Endogenous Control Name` == 'ACTN'  ), ]

# Reorder Sample, so Normoxia comes before Hypoxia in the graph
gene_expression_df_subset_actn$Sample <- factor(
  gene_expression_df_subset_actn$Sample, levels=c("WN_N", "WN_H", "WS_N", "WS_H"))

# Reorder Hours
gene_expression_df_subset_actn$Hours <- factor(
  gene_expression_df_subset_actn$Hours, levels=c("48h", "96h", "144h"))

#### Normoxia-Hypoxia Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.normoxia_hypoxia.actn <- gene_expression_df_subset_actn %>%
  group_by(`Cell_Line`) %>%
  t_test(RQ ~ Condition) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")

#### WN/WS Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.wn_ws.actn <- gene_expression_df_subset_actn %>%
  group_by(`Condition`) %>%
  t_test(RQ ~ `Sample`) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")

## Create Barplot
# Change the width of bars
barplot <- ggbarplot(gene_expression_df_subset_actn, 
                     x = "Hours", y = "RQ", 
                     fill = "Sample",
                     color = "Sample",
                     add = "mean_sd", 
                     ylab = paste0(gene, "/ACTN RQ"),
                     palette = c("#0412d4", "#f71105", "#0412d4", "#f71105"),
                     position = position_dodge())

# Add 10% spaces between the p-value labels and the plot border
barplot <- barplot + 
  geom_bracket(
    xmin = c("48h", "96h", "48h"), # xmin matches xmax: WN_N goes w/ WN_H
    xmax = c("96h", "144h", "144h"), 
    y.position = c(57, 67, 77),          # Same order as above
    label = c("***", "**", "**")         # Same order as above
  ) +
  scale_y_continuous(expand = expansion(mult = c(0, 0.1)))
barplot

###########################################################
###########################################################
################################################## B) GAPDH
###########################################################
###########################################################

## Subset target and endogenous control for analysis
# 1) Subset target
gene_expression_df_subset_gapdh <- gene_expression_df[ which(
    gene_expression_df$`Target Gene Name` == gene  ), ]
# 2) Subset Endogenous Control
gene_expression_df_subset_gapdh <- gene_expression_df_subset_gapdh[ which(
    gene_expression_df_subset_gapdh$`Endogenous Control Name` == 'GAPDH'  ), ]

# Reorder Sample, so Normoxia comes before Hypoxia in the graph
gene_expression_df_subset_gapdh$Sample <- factor(
  gene_expression_df_subset_gapdh$Sample, levels=c("WN_N", "WN_H", "WS_N", "WS_H"))

# Reorder Hours
gene_expression_df_subset_gapdh$Hours <- factor(
  gene_expression_df_subset_gapdh$Hours, levels=c("48h", "96h", "144h"))

#### Normoxia-Hypoxia Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.normoxia_hypoxia.gapdh <- gene_expression_df_subset_gapdh %>%
  group_by(`Cell_Line`) %>%
  t_test(RQ ~ `Condition`) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")

#### WN/WS Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.wn_ws.gapdh <- gene_expression_df_subset_gapdh %>%
  group_by(`Condition`) %>%
  t_test(RQ ~ `Sample`) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")

## Create Barplot
# Change the width of bars
barplot <- ggbarplot(gene_expression_df_subset_gapdh, 
                     x = "Hours", y = "RQ", 
                     fill = "Sample",
                     color = "Sample",
                     add = "mean_sd", 
                     ylab = paste0(gene, "/GAPDH RQ"),
                     palette = c("#0412d4", "#f71105", "#0412d4", "#f71105"),
                     position = position_dodge())

# Add 10% spaces between the p-value labels and the plot border
barplot <- barplot + 
  geom_bracket(
    xmin = c("48h", "96h", "48h"), # xmin matches xmax: WN_N goes w/ WN_H
    xmax = c("96h", "144h", "144h"), 
    y.position = c(95, 105, 115),          # Same order as above
    label = c("***", "**", "**")         # Same order as above
  ) +
  scale_y_continuous(expand = expansion(mult = c(0, 0.1)))
barplot

```

# 14.3) CD274

```{r}
library(rstatix)
library(readxl)
library(ggpubr)

gene="CD274"

directory <- "~/Desktop/Gepoliano/Hypoxia/Time Course/Results/"
excel_file <- "GeneExpressionDF.xlsx"
gene_expression_df <- read_excel(paste0(directory, excel_file))

# Subset Date Seeded "20220607"
gene_expression_df <- gene_expression_df[ which(
    gene_expression_df$Date_Seeded == "20220607"), ]

# Subset Number Experiment
gene_expression_df <- gene_expression_df[ which(
  gene_expression_df$Number_Given_to_Experiment == "0" |
    gene_expression_df$Number_Given_to_Experiment == "1" |
      gene_expression_df$Number_Given_to_Experiment == "2" |
      gene_expression_df$Number_Given_to_Experiment == "3"), ]

# Subset Time Point for both ACTN and GAPDH
gene_expression_df <- gene_expression_df[ which(
    gene_expression_df$`Hours` == '48h' |
      gene_expression_df$`Hours` == '96h' |
      gene_expression_df$`Hours` == '144h'), ]

###########################################################
###########################################################
################################################### A) ACTN
###########################################################
###########################################################

## Subset target and endogenous control for analysis
# 1) Subset target
gene_expression_df_subset_actn <- gene_expression_df[ which(
    gene_expression_df$`Target Gene Name` == gene  ), ]
# 2) Subset Endogenous Control
gene_expression_df_subset_actn <- gene_expression_df_subset_actn[ which(
    gene_expression_df_subset_actn$`Endogenous Control Name` == 'ACTN'  ), ]

# Reorder Sample, so Normoxia comes before Hypoxia in the graph
gene_expression_df_subset_actn$Sample <- factor(
  gene_expression_df_subset_actn$Sample, levels=c("WN_N", "WN_H", "WS_N", "WS_H"))

# Reorder Hours
gene_expression_df_subset_actn$Hours <- factor(
  gene_expression_df_subset_actn$Hours, levels=c("48h", "96h", "144h"))

#### Normoxia-Hypoxia Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.normoxia_hypoxia.actn <- gene_expression_df_subset_actn %>%
  group_by(`Cell_Line`) %>%
  t_test(RQ ~ Condition) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")

#### WN/WS Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.wn_ws.actn <- gene_expression_df_subset_actn %>%
  group_by(`Condition`) %>%
  t_test(RQ ~ `Sample`) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")

## Create Barplot
# Change the width of bars
barplot <- ggbarplot(gene_expression_df_subset_actn, 
                     x = "Hours", y = "RQ", 
                     fill = "Sample",
                     color = "Sample",
                     add = "mean_sd", 
                     ylab = paste0(gene, "/ACTN RQ"),
                     palette = c("#0412d4", "#f71105", "#0412d4", "#f71105"),
                     position = position_dodge())

# Add 10% spaces between the p-value labels and the plot border
barplot <- barplot + 
  geom_bracket(
    xmin = c("48h", "96h", "48h"), # xmin matches xmax: WN_N goes w/ WN_H
    xmax = c("96h", "144h", "144h"), 
    y.position = c(57, 67, 77),          # Same order as above
    label = c("***", "**", "**")         # Same order as above
  ) +
  scale_y_continuous(expand = expansion(mult = c(0, 0.1)))
barplot

###########################################################
###########################################################
################################################## B) GAPDH
###########################################################
###########################################################

## Subset target and endogenous control for analysis
# 1) Subset target
gene_expression_df_subset_gapdh <- gene_expression_df[ which(
    gene_expression_df$`Target Gene Name` == gene  ), ]
# 2) Subset Endogenous Control
gene_expression_df_subset_gapdh <- gene_expression_df_subset_gapdh[ which(
    gene_expression_df_subset_gapdh$`Endogenous Control Name` == 'GAPDH'  ), ]

# Reorder Sample, so Normoxia comes before Hypoxia in the graph
gene_expression_df_subset_gapdh$Sample <- factor(
  gene_expression_df_subset_gapdh$Sample, levels=c("WN_N", "WN_H", "WS_N", "WS_H"))

# Reorder Hours
gene_expression_df_subset_gapdh$Hours <- factor(
  gene_expression_df_subset_gapdh$Hours, levels=c("48h", "96h", "144h"))

#### Normoxia-Hypoxia Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.normoxia_hypoxia.gapdh <- gene_expression_df_subset_gapdh %>%
  group_by(`Cell_Line`) %>%
  t_test(RQ ~ `Condition`) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")

#### WN/WS Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.wn_ws.gapdh <- gene_expression_df_subset_gapdh %>%
  group_by(`Condition`) %>%
  t_test(RQ ~ `Sample`) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")

## Create Barplot
# Change the width of bars
barplot <- ggbarplot(gene_expression_df_subset_gapdh, 
                     x = "Hours", y = "RQ", 
                     fill = "Sample",
                     color = "Sample",
                     add = "mean_sd", 
                     ylab = paste0(gene, "/GAPDH RQ"),
                     palette = c("#0412d4", "#f71105", "#0412d4", "#f71105"),
                     position = position_dodge())

# Add 10% spaces between the p-value labels and the plot border
barplot <- barplot + 
  geom_bracket(
    xmin = c("48h", "96h", "48h"), # xmin matches xmax: WN_N goes w/ WN_H
    xmax = c("96h", "144h", "144h"), 
    y.position = c(95, 105, 115),          # Same order as above
    label = c("***", "**", "**")         # Same order as above
  ) +
  scale_y_continuous(expand = expansion(mult = c(0, 0.1)))
barplot

```

# 14.4) CSF1

```{r}
library(rstatix)
library(readxl)
library(ggpubr)

gene="CSF1"

directory <- "~/Desktop/Gepoliano/Hypoxia/Time Course/Results/"
excel_file <- "GeneExpressionDF.xlsx"
gene_expression_df <- read_excel(paste0(directory, excel_file))

# Subset Date Seeded "20220607"
gene_expression_df <- gene_expression_df[ which(
    gene_expression_df$Date_Seeded == "20220607"), ]

# Subset Number Experiment
gene_expression_df <- gene_expression_df[ which(
  gene_expression_df$Number_Given_to_Experiment == "0" |
    gene_expression_df$Number_Given_to_Experiment == "1" |
      gene_expression_df$Number_Given_to_Experiment == "2" |
      gene_expression_df$Number_Given_to_Experiment == "3"), ]

# Subset Time Point for both ACTN and GAPDH
gene_expression_df <- gene_expression_df[ which(
    gene_expression_df$`Hours` == '48h' |
      gene_expression_df$`Hours` == '96h' |
      gene_expression_df$`Hours` == '144h'), ]

###########################################################
###########################################################
################################################### A) ACTN
###########################################################
###########################################################

## Subset target and endogenous control for analysis
# 1) Subset target
gene_expression_df_subset_actn <- gene_expression_df[ which(
    gene_expression_df$`Target Gene Name` == gene  ), ]
# 2) Subset Endogenous Control
gene_expression_df_subset_actn <- gene_expression_df_subset_actn[ which(
    gene_expression_df_subset_actn$`Endogenous Control Name` == 'ACTN'  ), ]

# Reorder Sample, so Normoxia comes before Hypoxia in the graph
gene_expression_df_subset_actn$Sample <- factor(
  gene_expression_df_subset_actn$Sample, levels=c("WN_N", "WN_H", "WS_N", "WS_H"))

# Reorder Hours
gene_expression_df_subset_actn$Hours <- factor(
  gene_expression_df_subset_actn$Hours, levels=c("48h", "96h", "144h"))

#### Normoxia-Hypoxia Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.normoxia_hypoxia.actn <- gene_expression_df_subset_actn %>%
  group_by(`Cell_Line`) %>%
  t_test(RQ ~ Condition) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")

#### WN/WS Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.wn_ws.actn <- gene_expression_df_subset_actn %>%
  group_by(`Condition`) %>%
  t_test(RQ ~ `Sample`) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")

## Create Barplot
# Change the width of bars
barplot <- ggbarplot(gene_expression_df_subset_actn, 
                     x = "Hours", y = "RQ", 
                     fill = "Sample",
                     color = "Sample",
                     add = "mean_sd", 
                     ylab = paste0(gene, "/ACTN RQ"),
                     palette = c("#0412d4", "#f71105", "#0412d4", "#f71105"),
                     position = position_dodge())

# Add 10% spaces between the p-value labels and the plot border
barplot <- barplot + 
  geom_bracket(
    xmin = c("48h", "96h", "48h"), # xmin matches xmax: WN_N goes w/ WN_H
    xmax = c("96h", "144h", "144h"), 
    y.position = c(5, 6, 7),          # Same order as above
    label = c("***", "**", "**")         # Same order as above
  ) +
  scale_y_continuous(expand = expansion(mult = c(0, 0.1)))
barplot

###########################################################
###########################################################
################################################## B) GAPDH
###########################################################
###########################################################

## Subset target and endogenous control for analysis
# 1) Subset target
gene_expression_df_subset_gapdh <- gene_expression_df[ which(
    gene_expression_df$`Target Gene Name` == gene  ), ]
# 2) Subset Endogenous Control
gene_expression_df_subset_gapdh <- gene_expression_df_subset_gapdh[ which(
    gene_expression_df_subset_gapdh$`Endogenous Control Name` == 'GAPDH'  ), ]

# Reorder Sample, so Normoxia comes before Hypoxia in the graph
gene_expression_df_subset_gapdh$Sample <- factor(
  gene_expression_df_subset_gapdh$Sample, levels=c("WN_N", "WN_H", "WS_N", "WS_H"))

# Reorder Hours
gene_expression_df_subset_gapdh$Hours <- factor(
  gene_expression_df_subset_gapdh$Hours, levels=c("48h", "96h", "144h"))

#### Normoxia-Hypoxia Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.normoxia_hypoxia.gapdh <- gene_expression_df_subset_gapdh %>%
  group_by(`Cell_Line`) %>%
  t_test(RQ ~ `Condition`) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")

#### WN/WS Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.wn_ws.gapdh <- gene_expression_df_subset_gapdh %>%
  group_by(`Condition`) %>%
  t_test(RQ ~ `Sample`) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")

## Create Barplot
# Change the width of bars
barplot <- ggbarplot(gene_expression_df_subset_gapdh, 
                     x = "Hours", y = "RQ", 
                     fill = "Sample",
                     color = "Sample",
                     add = "mean_sd", 
                     ylab = paste0(gene, "/GAPDH RQ"),
                     palette = c("#0412d4", "#f71105", "#0412d4", "#f71105"),
                     position = position_dodge())

# Add 10% spaces between the p-value labels and the plot border
barplot <- barplot + 
  geom_bracket(
    xmin = c("48h", "96h", "48h"), # xmin matches xmax: WN_N goes w/ WN_H
    xmax = c("96h", "144h", "144h"), 
    y.position = c(10, 12, 14),          # Same order as above
    label = c("***", "**", "**")         # Same order as above
  ) +
  scale_y_continuous(expand = expansion(mult = c(0, 0.1)))
barplot

```

# 14.5) ASCL1

```{r}
library(rstatix)
library(readxl)
library(ggpubr)

gene="ASCL1"

directory <- "~/Desktop/Gepoliano/Hypoxia/Time Course/Results/"
excel_file <- "GeneExpressionDF.xlsx"
gene_expression_df <- read_excel(paste0(directory, excel_file))

# Subset Date Seeded "20220607"
gene_expression_df <- gene_expression_df[ which(
    gene_expression_df$Date_Seeded == "20220607"), ]

# Subset Number Experiment
gene_expression_df <- gene_expression_df[ which(
  gene_expression_df$Number_Given_to_Experiment == "0" |
    gene_expression_df$Number_Given_to_Experiment == "1" |
      gene_expression_df$Number_Given_to_Experiment == "2" |
      gene_expression_df$Number_Given_to_Experiment == "3"), ]

# Subset Time Point for both ACTN and GAPDH
gene_expression_df <- gene_expression_df[ which(
    gene_expression_df$`Hours` == '48h' |
      gene_expression_df$`Hours` == '96h' |
      gene_expression_df$`Hours` == '144h'), ]

###########################################################
###########################################################
################################################### A) ACTN
###########################################################
###########################################################

## Subset target and endogenous control for analysis
# 1) Subset target
gene_expression_df_subset_actn <- gene_expression_df[ which(
    gene_expression_df$`Target Gene Name` == gene  ), ]
# 2) Subset Endogenous Control
gene_expression_df_subset_actn <- gene_expression_df_subset_actn[ which(
    gene_expression_df_subset_actn$`Endogenous Control Name` == 'ACTN'  ), ]

# Reorder Sample, so Normoxia comes before Hypoxia in the graph
gene_expression_df_subset_actn$Sample <- factor(
  gene_expression_df_subset_actn$Sample, levels=c("WN_N", "WN_H", "WS_N", "WS_H"))

# Reorder Hours
gene_expression_df_subset_actn$Hours <- factor(
  gene_expression_df_subset_actn$Hours, levels=c("48h", "96h", "144h"))

#### Normoxia-Hypoxia Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.normoxia_hypoxia.actn <- gene_expression_df_subset_actn %>%
  group_by(`Cell_Line`) %>%
  t_test(RQ ~ Condition) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")

#### WN/WS Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.wn_ws.actn <- gene_expression_df_subset_actn %>%
  group_by(`Condition`) %>%
  t_test(RQ ~ `Sample`) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")

## Create Barplot
# Change the width of bars
barplot <- ggbarplot(gene_expression_df_subset_actn, 
                     x = "Hours", y = "RQ", 
                     fill = "Sample",
                     color = "Sample",
                     add = "mean_sd", 
                     ylab = paste0(gene, "/ACTN RQ"),
                     palette = c("#0412d4", "#f71105", "#0412d4", "#f71105"),
                     position = position_dodge())

# Add 10% spaces between the p-value labels and the plot border
barplot <- barplot + 
  geom_bracket(
    xmin = c("48h", "96h", "48h"), # xmin matches xmax: WN_N goes w/ WN_H
    xmax = c("96h", "144h", "144h"), 
    y.position = c(2, 3, 4),          # Same order as above
    label = c("***", "**", "**")         # Same order as above
  ) +
  scale_y_continuous(expand = expansion(mult = c(0, 0.1)))
barplot

###########################################################
###########################################################
################################################## B) GAPDH
###########################################################
###########################################################

## Subset target and endogenous control for analysis
# 1) Subset target
gene_expression_df_subset_gapdh <- gene_expression_df[ which(
    gene_expression_df$`Target Gene Name` == gene  ), ]
# 2) Subset Endogenous Control
gene_expression_df_subset_gapdh <- gene_expression_df_subset_gapdh[ which(
    gene_expression_df_subset_gapdh$`Endogenous Control Name` == 'GAPDH'  ), ]

# Reorder Sample, so Normoxia comes before Hypoxia in the graph
gene_expression_df_subset_gapdh$Sample <- factor(
  gene_expression_df_subset_gapdh$Sample, levels=c("WN_N", "WN_H", "WS_N", "WS_H"))

# Reorder Hours
gene_expression_df_subset_gapdh$Hours <- factor(
  gene_expression_df_subset_gapdh$Hours, levels=c("48h", "96h", "144h"))

#### Normoxia-Hypoxia Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.normoxia_hypoxia.gapdh <- gene_expression_df_subset_gapdh %>%
  group_by(`Cell_Line`) %>%
  t_test(RQ ~ `Condition`) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")

#### WN/WS Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.wn_ws.gapdh <- gene_expression_df_subset_gapdh %>%
  group_by(`Condition`) %>%
  t_test(RQ ~ `Sample`) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")

## Create Barplot
# Change the width of bars
barplot <- ggbarplot(gene_expression_df_subset_gapdh, 
                     x = "Hours", y = "RQ", 
                     fill = "Sample",
                     color = "Sample",
                     add = "mean_sd", 
                     ylab = paste0(gene, "/GAPDH RQ"),
                     palette = c("#0412d4", "#f71105", "#0412d4", "#f71105"),
                     position = position_dodge())

# Add 10% spaces between the p-value labels and the plot border
barplot <- barplot + 
  geom_bracket(
    xmin = c("48h", "96h", "48h"), # xmin matches xmax: WN_N goes w/ WN_H
    xmax = c("96h", "144h", "144h"), 
    y.position = c(3, 4, 5),          # Same order as above
    label = c("***", "**", "**")         # Same order as above
  ) +
  scale_y_continuous(expand = expansion(mult = c(0, 0.1)))
barplot

```

# 14.6) PHOX2B

```{r}
library(rstatix)
library(readxl)
library(ggpubr)

gene="PHOX2B"

directory <- "~/Desktop/Gepoliano/Hypoxia/Time Course/Results/"
excel_file <- "GeneExpressionDF.xlsx"
gene_expression_df <- read_excel(paste0(directory, excel_file))

# Subset Date Seeded "20220607"
gene_expression_df <- gene_expression_df[ which(
    gene_expression_df$Date_Seeded == "20220607"), ]

# Subset Number Experiment
gene_expression_df <- gene_expression_df[ which(
  gene_expression_df$Number_Given_to_Experiment == "0" |
    gene_expression_df$Number_Given_to_Experiment == "1" |
      gene_expression_df$Number_Given_to_Experiment == "2" |
      gene_expression_df$Number_Given_to_Experiment == "3"), ]

# Subset Time Point for both ACTN and GAPDH
gene_expression_df <- gene_expression_df[ which(
    gene_expression_df$`Hours` == '48h' |
      gene_expression_df$`Hours` == '96h' |
      gene_expression_df$`Hours` == '144h'), ]

###########################################################
###########################################################
################################################### A) ACTN
###########################################################
###########################################################

## Subset target and endogenous control for analysis
# 1) Subset target
gene_expression_df_subset_actn <- gene_expression_df[ which(
    gene_expression_df$`Target Gene Name` == gene  ), ]
# 2) Subset Endogenous Control
gene_expression_df_subset_actn <- gene_expression_df_subset_actn[ which(
    gene_expression_df_subset_actn$`Endogenous Control Name` == 'ACTN'  ), ]

# Reorder Sample, so Normoxia comes before Hypoxia in the graph
gene_expression_df_subset_actn$Sample <- factor(
  gene_expression_df_subset_actn$Sample, levels=c("WN_N", "WN_H", "WS_N", "WS_H"))

# Reorder Hours
gene_expression_df_subset_actn$Hours <- factor(
  gene_expression_df_subset_actn$Hours, levels=c("48h", "96h", "144h"))

#### Normoxia-Hypoxia Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.normoxia_hypoxia.actn <- gene_expression_df_subset_actn %>%
  group_by(`Cell_Line`) %>%
  t_test(RQ ~ Condition) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")

#### WN/WS Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.wn_ws.actn <- gene_expression_df_subset_actn %>%
  group_by(`Condition`) %>%
  t_test(RQ ~ `Sample`) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")

## Create Barplot
# Change the width of bars
barplot <- ggbarplot(gene_expression_df_subset_actn, 
                     x = "Hours", y = "RQ", 
                     fill = "Sample",
                     color = "Sample",
                     add = "mean_sd", 
                     ylab = paste0(gene, "/ACTN RQ"),
                     palette = c("#0412d4", "#f71105", "#0412d4", "#f71105"),
                     position = position_dodge())

# Add 10% spaces between the p-value labels and the plot border
barplot <- barplot + 
  geom_bracket(
    xmin = c("48h", "96h", "48h"), # xmin matches xmax: WN_N goes w/ WN_H
    xmax = c("96h", "144h", "144h"), 
    y.position = c(1.5, 2, 2.5),          # Same order as above
    label = c("***", "**", "**")         # Same order as above
  ) +
  scale_y_continuous(expand = expansion(mult = c(0, 0.1)))
barplot

###########################################################
###########################################################
################################################## B) GAPDH
###########################################################
###########################################################

## Subset target and endogenous control for analysis
# 1) Subset target
gene_expression_df_subset_gapdh <- gene_expression_df[ which(
    gene_expression_df$`Target Gene Name` == gene  ), ]
# 2) Subset Endogenous Control
gene_expression_df_subset_gapdh <- gene_expression_df_subset_gapdh[ which(
    gene_expression_df_subset_gapdh$`Endogenous Control Name` == 'GAPDH'  ), ]

# Reorder Sample, so Normoxia comes before Hypoxia in the graph
gene_expression_df_subset_gapdh$Sample <- factor(
  gene_expression_df_subset_gapdh$Sample, levels=c("WN_N", "WN_H", "WS_N", "WS_H"))

# Reorder Hours
gene_expression_df_subset_gapdh$Hours <- factor(
  gene_expression_df_subset_gapdh$Hours, levels=c("48h", "96h", "144h"))

#### Normoxia-Hypoxia Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.normoxia_hypoxia.gapdh <- gene_expression_df_subset_gapdh %>%
  group_by(`Cell_Line`) %>%
  t_test(RQ ~ `Condition`) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")

#### WN/WS Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.wn_ws.gapdh <- gene_expression_df_subset_gapdh %>%
  group_by(`Condition`) %>%
  t_test(RQ ~ `Sample`) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")

## Create Barplot
# Change the width of bars
barplot <- ggbarplot(gene_expression_df_subset_gapdh, 
                     x = "Hours", y = "RQ", 
                     fill = "Sample",
                     color = "Sample",
                     add = "mean_sd", 
                     ylab = paste0(gene, "/GAPDH RQ"),
                     palette = c("#0412d4", "#f71105", "#0412d4", "#f71105"),
                     position = position_dodge())

# Add 10% spaces between the p-value labels and the plot border
barplot <- barplot + 
  geom_bracket(
    xmin = c("48h", "96h", "48h"), # xmin matches xmax: WN_N goes w/ WN_H
    xmax = c("96h", "144h", "144h"), 
    y.position = c(2, 2.5, 3),          # Same order as above
    label = c("***", "**", "**")         # Same order as above
  ) +
  scale_y_continuous(expand = expansion(mult = c(0, 0.1)))
barplot

```

# 14.7) TET1

```{r}
library(rstatix)
library(readxl)
library(ggpubr)

gene="TET1"

directory <- "~/Desktop/Gepoliano/Hypoxia/Time Course/Results/"
excel_file <- "GeneExpressionDF.xlsx"
gene_expression_df <- read_excel(paste0(directory, excel_file))

# Subset Date Seeded "20220607"
gene_expression_df <- gene_expression_df[ which(
    gene_expression_df$Date_Seeded == "20220607"), ]

# Subset Number Experiment
gene_expression_df <- gene_expression_df[ which(
  gene_expression_df$Number_Given_to_Experiment == "0" |
    gene_expression_df$Number_Given_to_Experiment == "1" |
      gene_expression_df$Number_Given_to_Experiment == "2" |
      gene_expression_df$Number_Given_to_Experiment == "3"), ]

# Subset Time Point for both ACTN and GAPDH
gene_expression_df <- gene_expression_df[ which(
    gene_expression_df$`Hours` == '48h' |
      gene_expression_df$`Hours` == '96h' |
      gene_expression_df$`Hours` == '144h'), ]

###########################################################
###########################################################
################################################### A) ACTN
###########################################################
###########################################################

## Subset target and endogenous control for analysis
# 1) Subset target
gene_expression_df_subset_actn <- gene_expression_df[ which(
    gene_expression_df$`Target Gene Name` == gene  ), ]
# 2) Subset Endogenous Control
gene_expression_df_subset_actn <- gene_expression_df_subset_actn[ which(
    gene_expression_df_subset_actn$`Endogenous Control Name` == 'ACTN'  ), ]

# Reorder Sample, so Normoxia comes before Hypoxia in the graph
gene_expression_df_subset_actn$Sample <- factor(
  gene_expression_df_subset_actn$Sample, levels=c("WN_N", "WN_H", "WS_N", "WS_H"))

# Reorder Hours
gene_expression_df_subset_actn$Hours <- factor(
  gene_expression_df_subset_actn$Hours, levels=c("48h", "96h", "144h"))

#### Normoxia-Hypoxia Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.normoxia_hypoxia.actn <- gene_expression_df_subset_actn %>%
  group_by(`Cell_Line`) %>%
  t_test(RQ ~ Condition) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")

#### WN/WS Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.wn_ws.actn <- gene_expression_df_subset_actn %>%
  group_by(`Condition`) %>%
  t_test(RQ ~ `Sample`) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")

## Create Barplot
# Change the width of bars
barplot <- ggbarplot(gene_expression_df_subset_actn, 
                     x = "Hours", y = "RQ", 
                     fill = "Sample",
                     color = "Sample",
                     add = "mean_sd", 
                     ylab = paste0(gene, "/ACTN RQ"),
                     palette = c("#0412d4", "#f71105", "#0412d4", "#f71105"),
                     position = position_dodge())

# Add 10% spaces between the p-value labels and the plot border
barplot <- barplot + 
  geom_bracket(
    xmin = c("48h", "96h", "48h"), # xmin matches xmax: WN_N goes w/ WN_H
    xmax = c("96h", "144h", "144h"), 
    y.position = c(1.5, 2, 2.5),          # Same order as above
    label = c("***", "**", "**")         # Same order as above
  ) +
  scale_y_continuous(expand = expansion(mult = c(0, 0.1)))
barplot

###########################################################
###########################################################
################################################## B) GAPDH
###########################################################
###########################################################

## Subset target and endogenous control for analysis
# 1) Subset target
gene_expression_df_subset_gapdh <- gene_expression_df[ which(
    gene_expression_df$`Target Gene Name` == gene  ), ]
# 2) Subset Endogenous Control
gene_expression_df_subset_gapdh <- gene_expression_df_subset_gapdh[ which(
    gene_expression_df_subset_gapdh$`Endogenous Control Name` == 'GAPDH'  ), ]

# Reorder Sample, so Normoxia comes before Hypoxia in the graph
gene_expression_df_subset_gapdh$Sample <- factor(
  gene_expression_df_subset_gapdh$Sample, levels=c("WN_N", "WN_H", "WS_N", "WS_H"))

# Reorder Hours
gene_expression_df_subset_gapdh$Hours <- factor(
  gene_expression_df_subset_gapdh$Hours, levels=c("48h", "96h", "144h"))

#### Normoxia-Hypoxia Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.normoxia_hypoxia.gapdh <- gene_expression_df_subset_gapdh %>%
  group_by(`Cell_Line`) %>%
  t_test(RQ ~ `Condition`) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")

#### WN/WS Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.wn_ws.gapdh <- gene_expression_df_subset_gapdh %>%
  group_by(`Condition`) %>%
  t_test(RQ ~ `Sample`) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")

## Create Barplot
# Change the width of bars
barplot <- ggbarplot(gene_expression_df_subset_gapdh, 
                     x = "Hours", y = "RQ", 
                     fill = "Sample",
                     color = "Sample",
                     add = "mean_sd", 
                     ylab = paste0(gene, "/GAPDH RQ"),
                     palette = c("#0412d4", "#f71105", "#0412d4", "#f71105"),
                     position = position_dodge())

# Add 10% spaces between the p-value labels and the plot border
barplot <- barplot + 
  geom_bracket(
    xmin = c("48h", "96h", "48h"), # xmin matches xmax: WN_N goes w/ WN_H
    xmax = c("96h", "144h", "144h"), 
    y.position = c(2, 2.5, 3),          # Same order as above
    label = c("***", "**", "**")         # Same order as above
  ) +
  scale_y_continuous(expand = expansion(mult = c(0, 0.1)))
barplot

```

# 14.7) HIF1A

```{r}
library(rstatix)
library(readxl)
library(ggpubr)

gene="HIF1A"

directory <- "~/Desktop/Gepoliano/Hypoxia/Time Course/Results/"
excel_file <- "GeneExpressionDF.xlsx"
gene_expression_df <- read_excel(paste0(directory, excel_file))

# Subset Date Seeded "20220607"
gene_expression_df <- gene_expression_df[ which(
    gene_expression_df$Date_Seeded == "20220607"), ]

# Subset Number Experiment
gene_expression_df <- gene_expression_df[ which(
  gene_expression_df$Number_Given_to_Experiment == "0" |
    gene_expression_df$Number_Given_to_Experiment == "1" |
      gene_expression_df$Number_Given_to_Experiment == "2" |
      gene_expression_df$Number_Given_to_Experiment == "3"), ]

# Subset Time Point for both ACTN and GAPDH
gene_expression_df <- gene_expression_df[ which(
    gene_expression_df$`Hours` == '48h' |
      gene_expression_df$`Hours` == '96h' |
      gene_expression_df$`Hours` == '144h'), ]

###########################################################
###########################################################
################################################### A) ACTN
###########################################################
###########################################################

## Subset target and endogenous control for analysis
# 1) Subset target
gene_expression_df_subset_actn <- gene_expression_df[ which(
    gene_expression_df$`Target Gene Name` == gene  ), ]
# 2) Subset Endogenous Control
gene_expression_df_subset_actn <- gene_expression_df_subset_actn[ which(
    gene_expression_df_subset_actn$`Endogenous Control Name` == 'ACTN'  ), ]

# Reorder Sample, so Normoxia comes before Hypoxia in the graph
gene_expression_df_subset_actn$Sample <- factor(
  gene_expression_df_subset_actn$Sample, levels=c("WN_N", "WN_H", "WS_N", "WS_H"))

# Reorder Hours
gene_expression_df_subset_actn$Hours <- factor(
  gene_expression_df_subset_actn$Hours, levels=c("48h", "96h", "144h"))

#### Normoxia-Hypoxia Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.normoxia_hypoxia.actn <- gene_expression_df_subset_actn %>%
  group_by(`Cell_Line`) %>%
  t_test(RQ ~ Condition) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")

#### WN/WS Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.wn_ws.actn <- gene_expression_df_subset_actn %>%
  group_by(`Condition`) %>%
  t_test(RQ ~ `Sample`) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")

## Create Barplot
# Change the width of bars
barplot <- ggbarplot(gene_expression_df_subset_actn, 
                     x = "Hours", y = "RQ", 
                     fill = "Sample",
                     color = "Sample",
                     add = "mean_sd", 
                     ylab = paste0(gene, "/ACTN RQ"),
                     palette = c("#0412d4", "#f71105", "#0412d4", "#f71105"),
                     position = position_dodge())

# Add 10% spaces between the p-value labels and the plot border
barplot <- barplot + 
  geom_bracket(
    xmin = c("48h", "96h", "48h"), # xmin matches xmax: WN_N goes w/ WN_H
    xmax = c("96h", "144h", "144h"), 
    y.position = c(1.5, 2, 2.5),          # Same order as above
    label = c("***", "**", "**")         # Same order as above
  ) +
  scale_y_continuous(expand = expansion(mult = c(0, 0.1)))
barplot

###########################################################
###########################################################
################################################## B) GAPDH
###########################################################
###########################################################

## Subset target and endogenous control for analysis
# 1) Subset target
gene_expression_df_subset_gapdh <- gene_expression_df[ which(
    gene_expression_df$`Target Gene Name` == gene  ), ]
# 2) Subset Endogenous Control
gene_expression_df_subset_gapdh <- gene_expression_df_subset_gapdh[ which(
    gene_expression_df_subset_gapdh$`Endogenous Control Name` == 'GAPDH'  ), ]

# Reorder Sample, so Normoxia comes before Hypoxia in the graph
gene_expression_df_subset_gapdh$Sample <- factor(
  gene_expression_df_subset_gapdh$Sample, levels=c("WN_N", "WN_H", "WS_N", "WS_H"))

# Reorder Hours
gene_expression_df_subset_gapdh$Hours <- factor(
  gene_expression_df_subset_gapdh$Hours, levels=c("48h", "96h", "144h"))

#### Normoxia-Hypoxia Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.normoxia_hypoxia.gapdh <- gene_expression_df_subset_gapdh %>%
  group_by(`Cell_Line`) %>%
  t_test(RQ ~ `Condition`) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")

#### WN/WS Statistical Test ####
# stat.test is a DF with the padj and the Stars
stat.test.wn_ws.gapdh <- gene_expression_df_subset_gapdh %>%
  group_by(`Condition`) %>%
  t_test(RQ ~ `Sample`) %>%
  adjust_pvalue(method = "bonferroni") #%>%
  #add_significance("p.adj")

## Create Barplot
# Change the width of bars
barplot <- ggbarplot(gene_expression_df_subset_gapdh, 
                     x = "Hours", y = "RQ", 
                     fill = "Sample",
                     color = "Sample",
                     add = "mean_sd", 
                     ylab = paste0(gene, "/GAPDH RQ"),
                     palette = c("#0412d4", "#f71105", "#0412d4", "#f71105"),
                     position = position_dodge())

# Add 10% spaces between the p-value labels and the plot border
barplot <- barplot + 
  geom_bracket(
    xmin = c("48h", "96h", "48h"), # xmin matches xmax: WN_N goes w/ WN_H
    xmax = c("96h", "144h", "144h"), 
    y.position = c(3, 3.5, 4),          # Same order as above
    label = c("***", "**", "**")         # Same order as above
  ) +
  scale_y_continuous(expand = expansion(mult = c(0, 0.1)))
barplot

```

# 15) Barplot Ggplot

```{r}
directory <- "~/Desktop/Gepoliano/Hypoxia/Time Course/Results/"
excel_file <- "GeneExpressionDF.xlsx"
gene_expression_df <- read_excel(paste0(directory, excel_file))

# Subset Date Seeded "20220607"
gene_expression_df <- gene_expression_df[ which(
    gene_expression_df$Date_Seeded == "20220607"), ]

# Subset Number Experiment
gene_expression_df <- gene_expression_df[ which(
  gene_expression_df$Number_Given_to_Experiment == "0" |
    gene_expression_df$Number_Given_to_Experiment == "1" |
      gene_expression_df$Number_Given_to_Experiment == "2" |
      gene_expression_df$Number_Given_to_Experiment == "3"), ]

# Subset Time Point for both ACTN and GAPDH
gene_expression_df <- gene_expression_df[ which(
    gene_expression_df$`Hours` == '48h' |
      gene_expression_df$`Hours` == '96h' |
      gene_expression_df$`Hours` == '144h'), ]

# Target Gene
gene="ASCL1"

###########################################################
###########################################################
################################################### A) ACTN
###########################################################
###########################################################

## Subset target and endogenous control for analysis
# 1) Subset target
gene_expression_df_subset_actn <- gene_expression_df[ which(
    gene_expression_df$`Target Gene Name` == gene  ), ]
# 2) Subset Endogenous Control
gene_expression_df_subset_actn <- gene_expression_df_subset_actn[ which(
    gene_expression_df_subset_actn$`Endogenous Control Name` == 'ACTN'  ), ]

# Reorder Sample, so Normoxia comes before Hypoxia in the graph
gene_expression_df_subset_actn$Sample <- factor(
  gene_expression_df_subset_actn$Sample, levels=c("WN_N", "WN_H", "WS_N", "WS_H"))

# Reorder Hours: 48h, 96h, 144h
gene_expression_df_subset_actn$Hours <- factor(
  gene_expression_df_subset_actn$Hours, levels=c("48h", "96h", "144h"))

ggplot(gene_expression_df_subset_actn, 
       aes(fill=Sample, 
           y=RQ, 
           x=Hours)) + 
   ylab(paste0(gene,"/ACTN RQ")) +
  xlab("Time") +
  geom_bar(position="dodge", stat="identity")+
  scale_fill_manual(values = c("#0412d4", "#f71105", "#0412d4", "#f71105"), 
                    labels = c("WN Normoxia","WN Hypoxia",
                               "WS Normoxia","WS Hypoxia"))

###########################################################
###########################################################
################################################## B) GAPDH
###########################################################
###########################################################

## Subset target and endogenous control for analysis
# 1) Subset target
gene_expression_df_subset_gapdh <- gene_expression_df[ which(
    gene_expression_df$`Target Gene Name` == gene  ), ]
# 2) Subset Endogenous Control
gene_expression_df_subset_gapdh <- gene_expression_df_subset_gapdh[ which(
    gene_expression_df_subset_gapdh$`Endogenous Control Name` == 'GAPDH'  ), ]

# Reorder Sample, so Normoxia comes before Hypoxia in the graph
gene_expression_df_subset_gapdh$Sample <- factor(
  gene_expression_df_subset_gapdh$Sample, levels=c("WN_N", "WN_H", "WS_N", "WS_H"))

# Reorder Hours: 48h, 96h, 144h
gene_expression_df_subset_gapdh$Hours <- factor(
  gene_expression_df_subset_gapdh$Hours, levels=c("48h", "96h", "144h"))

ggplot(gene_expression_df_subset_gapdh, 
       aes(fill=Sample, 
           y=RQ,
           x=Hours)) + 
  ylab(paste0(gene,"/GAPDH RQ")) +
  xlab("Time") +
  geom_bar(position="dodge", stat="identity")+
  scale_fill_manual(values = c("#0412d4", "#f71105", "#0412d4", "#f71105"), 
                    labels = c("WN Normoxia","WN Hypoxia",
                               "WS Normoxia","WS Hypoxia"))
```
